---
title: "Melanocyte Paper Code Organized"
output: html_document
date: "2023-01-19"
---

###load Required Packages
```{r}
library(Seurat)
library(SeuratWrappers)
library(SCENIC)
library(GENIE3)
library(AUCell)
library(monocle3)
library(CellChat)
library(RColorBrewer)
library(ComplexHeatmap)
library(msigdbr)
library(circlize)
library(reshape2)
library(ggplot2)
library(stringr)
library(biomaRt)
library(fgsea)
library(dplyr)
library(survival)
library(survminer)
```

###Dataset Preprocessing and Analysis Pipelines
```{r D6 QC and Clustering}
D6.counts <- Read10X(data.dir = "/Users/fattahilab/Desktop/Mel_10X_FASTQs/d6/results/filtered_gene_bc_matrices/HG38-PLUS")
colnames(D6.counts) <- gsub("-1", "", colnames(D6.counts))
write.csv(D6.counts, "/Users/fattahilab/Desktop/Mel_10X_FASTQs/GEO Submission/Processed/D6.counts.csv")

D6_all <- CreateSeuratObject(D6.counts)

#QC Metrics
rb.genes <- rownames(D6_all)[grep("^RP[SL]",rownames(D6_all))]
D6_all <- PercentageFeatureSet(D6_all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(D6_all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
D6_all$novelty.score <- novelty$Score

Idents(D6_all) <- "orig.ident"
VlnPlot(D6_all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(D6_all, features = "percent.ribo")
VlnPlot(D6_all, features = "nCount_RNA") + geom_hline(yintercept = c(6000))
VlnPlot(D6_all, features = "nFeature_RNA") + geom_hline(yintercept = c(1500, 4500))
FeatureScatter(D6_all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")

#Remove low quality cells
D6_all <- subset(D6_all, subset = nCount_RNA > 6000 & nFeature_RNA > 1500 & nFeature_RNA < 4500)

D6_all <- NormalizeData(D6_all)
D6_all <- FindVariableFeatures(D6_all)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
D6_all <- CellCycleScoring(D6_all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
D6_all <- ScaleData(D6_all, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
D6_all <- RunPCA(D6_all)
ElbowPlot(D6_all, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(D6_all@reductions[["pca"]]@stdev[D6_all@reductions[["pca"]]@stdev > 2])
D6_all <- RunUMAP(D6_all, reduction = "pca", dims = 1:num.pcs)
D6_all <- FindNeighbors(D6_all, dims = 1:num.pcs)
D6_all <- FindClusters(D6_all, resolution = .2)

DimPlot(D6_all, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(D6_all, reduction = "umap", label = TRUE)

#verify clustering not by QC metrics
FeaturePlot(D6_all, reduction = "umap", features = "nFeature_RNA")
FeaturePlot(D6_all, reduction = "umap", features = "nCount_RNA", max.cutoff = 90000)
FeaturePlot(D6_all, reduction = "umap", features = "percent.ribo")
FeaturePlot(D6_all, reduction = "umap", features = "novelty.score")

#Run ALRA Gene Dropout Imputation
D6_all <- RunALRA(D6_all)
DefaultAssay(D6_all) <- "RNA"
```

```{r D15 QC and Clustering}
indir_D15.counts <- Read10X(data.dir = "/Users/fattahilab/Desktop/Mel_10X_FASTQs/D15_INDIR/results")
colnames(indir_D15.counts) <- gsub("-1", "", colnames(indir_D15.counts))
indir_D15.counts <- indir_D15.counts[,which(colSums(indir_D15.counts) > 5000)]
write.csv(indir_D15.counts, "/Users/fattahilab/Desktop/Mel_10X_FASTQs/GEO Submission/Processed/indir_D15.counts.csv")

indir_D15_all <- CreateSeuratObject(indir_D15.counts)

#QC Metrics
rb.genes <- rownames(indir_D15_all)[grep("^RP[SL]",rownames(indir_D15_all))]
indir_D15_all <- PercentageFeatureSet(indir_D15_all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(indir_D15_all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
indir_D15_all$novelty.score <- novelty$Score

Idents(indir_D15_all) <- "orig.ident"
VlnPlot(indir_D15_all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(indir_D15_all, features = "percent.ribo")
VlnPlot(indir_D15_all, features = "nCount_RNA") + geom_hline(yintercept = 6000)
VlnPlot(indir_D15_all, features = "nFeature_RNA") + geom_hline(yintercept = c(1500, 7500))
FeatureScatter(indir_D15_all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")

#Remove low quality cells
indir_D15_all <- subset(indir_D15_all, subset = nCount_RNA > 6000 & nFeature_RNA > 1500 & nFeature_RNA < 7500)

indir_D15_all <- NormalizeData(indir_D15_all)
indir_D15_all <- FindVariableFeatures(indir_D15_all)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
indir_D15_all <- CellCycleScoring(indir_D15_all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
indir_D15_all <- ScaleData(indir_D15_all, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D15_all <- RunPCA(indir_D15_all)
ElbowPlot(indir_D15_all, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D15_all@reductions[["pca"]]@stdev[indir_D15_all@reductions[["pca"]]@stdev > 2])
indir_D15_all <- RunUMAP(indir_D15_all, reduction = "pca", dims = 1:num.pcs)
indir_D15_all <- FindNeighbors(indir_D15_all, dims = 1:num.pcs)
indir_D15_all <- FindClusters(indir_D15_all, resolution = .2)

DimPlot(indir_D15_all, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(indir_D15_all, reduction = "umap", label = TRUE)

#verify clustering not by QC metrics
FeaturePlot(indir_D15_all, reduction = "umap", features = "nFeature_RNA")
FeaturePlot(indir_D15_all, reduction = "umap", features = "nCount_RNA", max.cutoff = 90000)
FeaturePlot(indir_D15_all, reduction = "umap", features = "percent.ribo")
FeaturePlot(indir_D15_all, reduction = "umap", features = "novelty.score")

#Run ALRA Gene Dropout Imputation
indir_D15_all <- RunALRA(indir_D15_all)
DefaultAssay(indir_D15_all) <- "RNA"
```

```{r D22 QC and Clustering}
indir_D22.counts <- Read10X(data.dir = "/Users/fattahilab/Desktop/Mel_10X_FASTQs/D22_INDIR/results")
colnames(indir_D22.counts) <- gsub("-1", "", colnames(indir_D22.counts))
indir_D22.counts <- indir_D22.counts[,which(colSums(indir_D22.counts) > 5000)]
write.csv(indir_D22.counts, "/Users/fattahilab/Desktop/Mel_10X_FASTQs/GEO Submission/Processed/indir_D22.counts.csv")

indir_D22_all <- CreateSeuratObject(indir_D22.counts)

#QC Metrics
rb.genes <- rownames(indir_D22_all)[grep("^RP[SL]",rownames(indir_D22_all))]
indir_D22_all <- PercentageFeatureSet(indir_D22_all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(indir_D22_all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
indir_D22_all$novelty.score <- novelty$Score

Idents(indir_D22_all) <- "orig.ident"
VlnPlot(indir_D22_all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(indir_D22_all, features = "percent.ribo")
VlnPlot(indir_D22_all, features = "nCount_RNA") + geom_hline(yintercept = 6000)
VlnPlot(indir_D22_all, features = "nFeature_RNA") + geom_hline(yintercept = c(1500, 6000))
FeatureScatter(indir_D22_all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")

#Remove low quality cells
indir_D22_all <- subset(indir_D22_all, subset = nCount_RNA > 6000 & nFeature_RNA > 1500 & nFeature_RNA < 6000)

indir_D22_all <- NormalizeData(indir_D22_all)
indir_D22_all <- FindVariableFeatures(indir_D22_all)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
indir_D22_all <- CellCycleScoring(indir_D22_all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
indir_D22_all <- ScaleData(indir_D22_all, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D22_all <- RunPCA(indir_D22_all)
ElbowPlot(indir_D22_all, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D22_all@reductions[["pca"]]@stdev[indir_D22_all@reductions[["pca"]]@stdev > 2])
indir_D22_all <- RunUMAP(indir_D22_all, reduction = "pca", dims = 1:num.pcs)
indir_D22_all <- FindNeighbors(indir_D22_all, dims = 1:num.pcs)
indir_D22_all <- FindClusters(indir_D22_all, resolution = .15)

DimPlot(indir_D22_all, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(indir_D22_all, reduction = "umap", label = TRUE)

#verify clustering not by QC metrics
FeaturePlot(indir_D22_all, reduction = "umap", features = "nFeature_RNA")
FeaturePlot(indir_D22_all, reduction = "umap", features = "nCount_RNA", max.cutoff = 90000)
FeaturePlot(indir_D22_all, reduction = "umap", features = "percent.ribo")
FeaturePlot(indir_D22_all, reduction = "umap", features = "novelty.score")
VlnPlot(indir_D22_all, features = "percent.ribo")

#Run ALRA Gene Dropout Imputation
indir_D22_all <- RunALRA(indir_D22_all)
DefaultAssay(indir_D22_all) <- "RNA"
```

```{r D30 QC and Clustering}
indir_D30.counts <- Read10X(data.dir = "/Users/fattahilab/Desktop/Mel_10X_FASTQs/d30_indir_output/results")
colnames(indir_D30.counts) <- gsub("-1", "", colnames(indir_D30.counts))
indir_D30.counts <- indir_D30.counts[,which(colSums(indir_D30.counts) > 5000)]
write.csv(indir_D30.counts, "/Users/fattahilab/Desktop/Mel_10X_FASTQs/GEO Submission/Processed/indir_D30.counts.csv")

indir_D30_all <- CreateSeuratObject(indir_D30.counts)

#QC Metrics
rb.genes <- rownames(indir_D30_all)[grep("^RP[SL]",rownames(indir_D30_all))]
indir_D30_all <- PercentageFeatureSet(indir_D30_all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(indir_D30_all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
indir_D30_all$novelty.score <- novelty$Score

Idents(indir_D30_all) <- "orig.ident"
VlnPlot(indir_D30_all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(indir_D30_all, features = "percent.ribo")
VlnPlot(indir_D30_all, features = "nCount_RNA") + geom_hline(yintercept = 6000)
VlnPlot(indir_D30_all, features = "nFeature_RNA") + geom_hline(yintercept = c(1500, 6000))
FeatureScatter(indir_D30_all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")

#Remove low quality cells
indir_D30_all <- subset(indir_D30_all, subset = nCount_RNA > 6000 & nFeature_RNA > 1500 & nFeature_RNA < 6000)

indir_D30_all <- NormalizeData(indir_D30_all)
indir_D30_all <- FindVariableFeatures(indir_D30_all)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
indir_D30_all <- CellCycleScoring(indir_D30_all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
indir_D30_all <- ScaleData(indir_D30_all, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D30_all <- RunPCA(indir_D30_all)
ElbowPlot(indir_D30_all, ndims = 30) + geom_hline(yintercept = 3)
num.pcs <- length(indir_D30_all@reductions[["pca"]]@stdev[indir_D30_all@reductions[["pca"]]@stdev > 3])
indir_D30_all <- RunUMAP(indir_D30_all, reduction = "pca", dims = 1:num.pcs)
indir_D30_all <- FindNeighbors(indir_D30_all, dims = 1:num.pcs)
indir_D30_all <- FindClusters(indir_D30_all, resolution = .1)

DimPlot(indir_D30_all, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(indir_D30_all, reduction = "umap", label = TRUE)

#verify clustering not by QC metrics
FeaturePlot(indir_D30_all, reduction = "umap", features = "nFeature_RNA")
FeaturePlot(indir_D30_all, reduction = "umap", features = "nCount_RNA", max.cutoff = 90000)
FeaturePlot(indir_D30_all, reduction = "umap", features = "percent.ribo")
FeaturePlot(indir_D30_all, reduction = "umap", features = "novelty.score")

#Run ALRA Gene Dropout Imputation
indir_D30_all <- RunALRA(indir_D30_all)
DefaultAssay(indir_D30_all) <- "RNA"
```

```{r D41 QC and Clustering}
indir_D41.counts <- Read10X(data.dir = "/Users/fattahilab/Desktop/Mel_10X_FASTQs/d41_indir_output/results")
colnames(indir_D41.counts) <- gsub("-1", "", colnames(indir_D41.counts))
indir_D41.counts <- indir_D41.counts[,which(colSums(indir_D41.counts) > 5000)]
write.csv(indir_D41.counts, "/Users/fattahilab/Desktop/Mel_10X_FASTQs/GEO Submission/Processed/indir_D41.counts.csv")

indir_D41_all <- CreateSeuratObject(indir_D41.counts)

#QC Metrics
rb.genes <- rownames(indir_D41_all)[grep("^RP[SL]",rownames(indir_D41_all))]
indir_D41_all <- PercentageFeatureSet(indir_D41_all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(indir_D41_all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
indir_D41_all$novelty.score <- novelty$Score

Idents(indir_D41_all) <- "orig.ident"
VlnPlot(indir_D41_all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(indir_D41_all, features = "percent.ribo")
VlnPlot(indir_D41_all, features = "nCount_RNA") + geom_hline(yintercept = 6000)
VlnPlot(indir_D41_all, features = "nFeature_RNA") + geom_hline(yintercept = c(1500, 6000))
FeatureScatter(indir_D41_all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")

#Remove low quality cells
indir_D41_all <- subset(indir_D41_all, subset = nCount_RNA > 6000 & nFeature_RNA > 1500 & nFeature_RNA < 6000)

indir_D41_all <- NormalizeData(indir_D41_all)
indir_D41_all <- FindVariableFeatures(indir_D41_all)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
indir_D41_all <- CellCycleScoring(indir_D41_all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
indir_D41_all <- ScaleData(indir_D41_all, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D41_all <- RunPCA(indir_D41_all)
ElbowPlot(indir_D41_all, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D41_all@reductions[["pca"]]@stdev[indir_D41_all@reductions[["pca"]]@stdev > 2])
indir_D41_all <- RunUMAP(indir_D41_all, reduction = "pca", dims = 1:num.pcs)
indir_D41_all <- FindNeighbors(indir_D41_all, dims = 1:num.pcs)
indir_D41_all <- FindClusters(indir_D41_all, resolution = .1)

DimPlot(indir_D41_all, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(indir_D41_all, reduction = "umap", label = TRUE)

#verify clustering not by QC metrics
FeaturePlot(indir_D41_all, reduction = "umap", features = "nFeature_RNA")
FeaturePlot(indir_D41_all, reduction = "umap", features = "nCount_RNA", max.cutoff = 90000)
FeaturePlot(indir_D41_all, reduction = "umap", features = "percent.ribo")
FeaturePlot(indir_D41_all, reduction = "umap", features = "novelty.score")

#Run ALRA Gene Dropout Imputation
indir_D41_all <- RunALRA(indir_D41_all)
DefaultAssay(indir_D41_all) <- "RNA"
```

```{r Primed D15 QC and Clustering}
dir_D15.counts <- Read10X(data.dir = "/Users/fattahilab/Desktop/Mel_10X_FASTQs/d15/results/filtered_gene_bc_matrices/HG38-PLUS")
colnames(dir_D15.counts) <- gsub("-1", "", colnames(dir_D15.counts))
write.csv(dir_D15.counts, "/Users/fattahilab/Desktop/Mel_10X_FASTQs/GEO Submission/Processed/dir_D15.counts.csv")

dir_D15_all <- CreateSeuratObject(dir_D15.counts)

#QC Metrics
rb.genes <- rownames(dir_D15_all)[grep("^RP[SL]",rownames(dir_D15_all))]
dir_D15_all <- PercentageFeatureSet(dir_D15_all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(dir_D15_all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
dir_D15_all$novelty.score <- novelty$Score

Idents(dir_D15_all) <- "orig_ident"
VlnPlot(dir_D15_all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(dir_D15_all, features = c("percent.mt", "percent.ribo"), ncol = 3)
VlnPlot(dir_D15_all, features = "nCount_RNA") + geom_hline(yintercept = 5000)
VlnPlot(dir_D15_all, features = "nFeature_RNA") + geom_hline(yintercept = c(1500, 6000))
FeatureScatter(dir_D15_all, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(dir_D15_all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
FeatureScatter(dir_D15_all, feature1 = "nFeature_RNA", feature2 = "percent.mt")

#Remove low quality cells
dir_D15_all <- subset(dir_D15_all, subset = nCount_RNA > 5000 & nFeature_RNA > 1500 & nFeature_RNA < 6000)

dir_D15_all <- NormalizeData(dir_D15_all)
dir_D15_all <- FindVariableFeatures(dir_D15_all)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
dir_D15_all <- CellCycleScoring(dir_D15_all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
dir_D15_all <- ScaleData(dir_D15_all, vars.to.regress = c("novelty.score", "S.Score", "G2M.Score", "percent.ribo"))
dir_D15_all <- RunPCA(dir_D15_all)
ElbowPlot(dir_D15_all, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(dir_D15_all@reductions[["pca"]]@stdev[dir_D15_all@reductions[["pca"]]@stdev > 2])
dir_D15_all <- RunUMAP(dir_D15_all, reduction = "pca", dims = 1:num.pcs)
dir_D15_all <- FindNeighbors(dir_D15_all, dims = 1:num.pcs)
dir_D15_all <- FindClusters(dir_D15_all, resolution = .2)
DimPlot(dir_D15_all, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(dir_D15_all, reduction = "umap", label = TRUE) + theme(legend.position = "none")

#verify clustering not by QC metrics
FeaturePlot(dir_D15_all, reduction = "umap", features = "nFeature_RNA")
FeaturePlot(dir_D15_all, reduction = "umap", features = "nCount_RNA", max.cutoff = 90000)
FeaturePlot(dir_D15_all, reduction = "umap", features = "percent.ribo")
FeaturePlot(dir_D15_all, reduction = "umap", features = "novelty.score")

#Run ALRA Gene Dropout Imputation
dir_D15_all <- RunALRA(dir_D15_all)
DefaultAssay(dir_D15_all) <- "RNA"
```

```{r T2 Mel QC and Clustering}
indir_D59.counts <- Read10X(data.dir = "/Users/fattahilab/Desktop/Mel_10X_FASTQs/d59_indir_output/results")
colnames(indir_D59.counts) <- gsub("-1", "", colnames(indir_D59.counts))
indir_D59.counts <- indir_D59.counts[,which(colSums(indir_D59.counts) > 5000)]
write.csv(indir_D59.counts, "/Users/fattahilab/Desktop/Mel_10X_FASTQs/GEO Submission/Processed/indir_D59.counts.csv")

indir_D59_all <- CreateSeuratObject(indir_D59.counts)

#QC Metrics
rb.genes <- rownames(indir_D59_all)[grep("^RP[SL]",rownames(indir_D59_all))]
indir_D59_all <- PercentageFeatureSet(indir_D59_all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(indir_D59_all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
indir_D59_all$novelty.score <- novelty$Score

Idents(indir_D59_all) <- "orig.ident"
VlnPlot(indir_D59_all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(indir_D59_all, features = "percent.ribo")
VlnPlot(indir_D59_all, features = "nCount_RNA") + geom_hline(yintercept = 8000)
VlnPlot(indir_D59_all, features = "nFeature_RNA") + geom_hline(yintercept = c(3000, 7000))
FeatureScatter(indir_D59_all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")

#Remove low quality cells
indir_D59_all <- subset(indir_D59_all, subset = nCount_RNA > 8000 & nFeature_RNA > 3000 & nFeature_RNA < 7000)

indir_D59_all <- NormalizeData(indir_D59_all)
indir_D59_all <- FindVariableFeatures(indir_D59_all)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
indir_D59_all <- CellCycleScoring(indir_D59_all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
indir_D59_all <- ScaleData(indir_D59_all, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D59_all <- RunPCA(indir_D59_all)
ElbowPlot(indir_D59_all, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D59_all@reductions[["pca"]]@stdev[indir_D59_all@reductions[["pca"]]@stdev > 2])
indir_D59_all <- RunUMAP(indir_D59_all, reduction = "pca", dims = 1:num.pcs)
indir_D59_all <- FindNeighbors(indir_D59_all, dims = 1:num.pcs)
indir_D59_all <- FindClusters(indir_D59_all, resolution = .2)
DimPlot(indir_D59_all, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(indir_D59_all, reduction = "umap", label = TRUE)

#verify clustering not by QC metrics
FeaturePlot(indir_D59_all, reduction = "umap", features = "nFeature_RNA")
FeaturePlot(indir_D59_all, reduction = "umap", features = "nCount_RNA", max.cutoff = 90000)
FeaturePlot(indir_D59_all, reduction = "umap", features = "percent.ribo")
FeaturePlot(indir_D59_all, reduction = "umap", features = "novelty.score")

#Run ALRA Gene Dropout Imputation
indir_D59_all <- RunALRA(indir_D59_all)
DefaultAssay(indir_D59_all) <- "RNA"
```

```{r T1 Mel QC and Clustering}
dir_D30.counts <- Read10X(data.dir = "/Users/fattahilab/Desktop/Mel_10X_FASTQs/d30_dir_output/results")
colnames(dir_D30.counts) <- gsub("-1", "", colnames(dir_D30.counts))
dir_D30.counts <- dir_D30.counts[,which(colSums(dir_D30.counts) > 5000)]
write.csv(dir_D30.counts, "/Users/fattahilab/Desktop/Mel_10X_FASTQs/GEO Submission/Processed/dir_D30.counts.csv")

dir_D30_all <- CreateSeuratObject(dir_D30.counts)

rb.genes <- rownames(dir_D30_all)[grep("^RP[SL]",rownames(dir_D30_all))]
dir_D30_all <- PercentageFeatureSet(dir_D30_all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(dir_D30_all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
dir_D30_all$novelty.score <- novelty$Score

Idents(dir_D30_all) <- "orig_ident"
VlnPlot(dir_D30_all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(dir_D30_all, features = "percent.ribo")
VlnPlot(dir_D30_all, features = "nCount_RNA") + geom_hline(yintercept = 7000)
VlnPlot(dir_D30_all, features = "nFeature_RNA") + geom_hline(yintercept = c(2000,6500))
FeatureScatter(dir_D30_all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")


dir_D30_all <- subset(dir_D30_all, subset = nCount_RNA > 7000 & nFeature_RNA > 2000 & nFeature_RNA < 6500)

dir_D30_all <- NormalizeData(dir_D30_all)
dir_D30_all <- FindVariableFeatures(dir_D30_all)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
dir_D30_all <- CellCycleScoring(dir_D30_all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
dir_D30_all <- ScaleData(dir_D30_all, vars.to.regress = c("novelty.score", "S.Score", "G2M.Score", "percent.ribo"))
dir_D30_all <- RunPCA(dir_D30_all)
ElbowPlot(dir_D30_all, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(dir_D30_all@reductions[["pca"]]@stdev[dir_D30_all@reductions[["pca"]]@stdev > 2])
dir_D30_all <- RunUMAP(dir_D30_all, reduction = "pca", dims = 1:num.pcs)
dir_D30_all <- FindNeighbors(dir_D30_all, dims = 1:num.pcs)
dir_D30_all <- FindClusters(dir_D30_all, resolution = .2)
DimPlot(dir_D30_all, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(dir_D30_all, reduction = "umap", label = TRUE) + theme(legend.position = "none")

#verify clustering not by QC metrics
FeaturePlot(dir_D30_all, reduction = "umap", features = "nFeature_RNA")
FeaturePlot(dir_D30_all, reduction = "umap", features = "nCount_RNA", max.cutoff = 90000)
FeaturePlot(dir_D30_all, reduction = "umap", features = "percent.ribo")
FeaturePlot(dir_D30_all, reduction = "umap", features = "novelty.score")

#Run ALRA Gene Dropout Imputation
dir_D30_all <- RunALRA(dir_D30_all)
DefaultAssay(dir_D30_all) <- "RNA"
```

###Annotation of Datasets
```{r Visualize Expression of CellType Markers}
datasets <- c("D6_all", "indir_D15_all", "indir_D22_all", "indir_D30_all", "indir_D41_all", "dir_D15_all", "indir_D59_all", "dir_D30_all")
#Cell.Type Identifiers
premigratory.NCC <- c("PAX3", "WNT1", "LMX1A")
migratory.NCC <- c("SOX10", "FOXD3", "MEF2C", "ETS1", "TFAP2B", "EDNRA", "NGFR", "SNAI2")
CNSprecursor <- c("PAX6", "SIX3", "FEZF2", "EMX2", "NEUROG2")
nonnueral.ectoderm <- c("GATA3", "DLX3", "EPCAM", "CDH1", "WNT6")
NCC <- c("SOX10", "FOXD3", "EDNRB", "ERBB3", "ITGA4")
mesenchymal <- c("TWIST1", "COL3A1", "PCOLCE", "PDGFRA", "PDGFRB")
melanoblast <- c("MITF", "SOX10", "KIT", "EDNRB", "SNAI2")
melanocyte <- c("MITF", "SOX10", "DCT", "TRPM1", "PMEL", "MLANA", "TYRP1", "TYR")
trigeminal.placode <- c("SIX1", "POU4F1", "ISL1")
SMP <- c("SOX10", "CAV1", "MITF")
Schwann.Cell <- c("GFRA1", "GFRA3", "S100B", "CD44", "MBP")

cell.type.annotations <- unique(c(premigratory.NCC, migratory.NCC, NCC, Schwann.Cell, CNSprecursor, nonnueral.ectoderm, trigeminal.placode, mesenchymal))

for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  plot <- DotPlot(seur_obj, features = cell.type.annotations, assay = "RNA", scale = F)+ ggtitle("Cell Type Annotation") + theme(axis.text.x = element_text(angle = 90, vjust = .5,hjust = 1, size = 8), axis.text.y = element_text(size = 8), axis.title.y = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 10), title = element_text(size = 18))+ scale_color_gradientn(colours = rev(brewer.pal(10, "Spectral")))
  plot(plot)
}
```

```{r Rename Cluster Assignments}
#D6
Idents(D6_all) <- "seurat_clusters"
cluster_levels <- c(2,0,3,1)
D6_all@meta.data$seurat_clusters <- factor(x = Idents(D6_all), levels = cluster_levels)

Idents(D6_all) <- "seurat_clusters"
new.cluster.ids <- c("S1 NCC", "Pre-migratory NCC", "CNS Precursor", "Non-Neural Ectoderm")
Idents(D6_all) <- plyr::mapvalues(x = Idents(D6_all), from = cluster_levels, to = new.cluster.ids)
D6_all$Cell.Type <- Idents(D6_all)

#D15
Idents(indir_D15_all) <- "seurat_clusters"
cluster_levels <- c(1,2,5,3,0,4)
indir_D15_all@meta.data$seurat_clusters <- factor(x = Idents(indir_D15_all), levels = cluster_levels)

Idents(indir_D15_all) <- "seurat_clusters"
new.cluster.ids <- c("S2 NCC", "S2 NCC", "CNS Precursor", "Non-Neural Ectoderm", "Placode", "Mesenchymal")
Idents(indir_D15_all) <- plyr::mapvalues(x = Idents(indir_D15_all), from = cluster_levels, to = new.cluster.ids)
indir_D15_all$Cell.Type <- Idents(indir_D15_all)

#D22
Idents(indir_D22_all) <- "seurat_clusters"
cluster_levels <- c(0,1,2,3)
indir_D22_all@meta.data$seurat_clusters <- factor(x = Idents(indir_D22_all), levels = cluster_levels)

Idents(indir_D22_all) <- "seurat_clusters"
new.cluster.ids <- c("S3 NCC", "S3 NCC", "Placode", "Mesenchymal")
Idents(indir_D22_all) <- plyr::mapvalues(x = Idents(indir_D22_all), from = cluster_levels, to = new.cluster.ids)
indir_D22_all$Cell.Type <- Idents(indir_D22_all)

#D30
Idents(indir_D30_all) <- "seurat_clusters"
cluster_levels <- c(0,1,2)
indir_D30_all@meta.data$seurat_clusters <- factor(x = Idents(indir_D30_all), levels = cluster_levels)


Idents(indir_D30_all) <- "seurat_clusters"
new.cluster.ids <- c("S4 NCC", "S4 NCC", "Mesenchymal")
Idents(indir_D30_all) <- plyr::mapvalues(x = Idents(indir_D30_all), from = cluster_levels, to = new.cluster.ids)
indir_D30_all$Cell.Type <- Idents(indir_D30_all)

#D41
Idents(indir_D41_all) <- "seurat_clusters"
cluster_levels <- c(0,1,2)
indir_D41_all@meta.data$seurat_clusters <- factor(x = Idents(indir_D41_all), levels = cluster_levels)

Idents(indir_D41_all) <- "seurat_clusters"
new.cluster.ids <- c("SCP", "Mature SCs", "Mesenchymal")
Idents(indir_D41_all) <- plyr::mapvalues(x = Idents(indir_D41_all), from = cluster_levels, to = new.cluster.ids)
indir_D41_all$Cell.Type <- Idents(indir_D41_all)

#Primed D15
Idents(dir_D15_all) <- "seurat_clusters"
cluster_levels <- c(0,3,2,1)
dir_D15_all@meta.data$seurat_clusters <- factor(x = Idents(dir_D15_all), levels = cluster_levels)

Idents(dir_D15_all) <- "seurat_clusters"
new.cluster.ids <- c("Primed S2 NCC", "CNS Precursor", "Non-Neural Ectoderm", "Mesenchymal")
Idents(dir_D15_all) <- plyr::mapvalues(x = Idents(dir_D15_all), from = cluster_levels, to = new.cluster.ids)
dir_D15_all$Cell.Type <- Idents(dir_D15_all)

#T2 Melanocytes
cluster_levels <- c(0,1)
Idents(indir_D59_all) <- factor(x = Idents(indir_D59_all), levels = cluster_levels)

Idents(indir_D59_all) <- "seurat_clusters"
new.cluster.ids <- c("T2 Melanocyte", "T2 Melanocyte")
Idents(indir_D59_all) <- plyr::mapvalues(x = Idents(indir_D59_all), from = cluster_levels, to = new.cluster.ids)
indir_D59_all$Cell.Type <- Idents(indir_D59_all)

#T1 Melanocytes
cluster_levels <- c(0,1)
Idents(dir_D30_all) <- factor(x = Idents(dir_D30_all), levels = cluster_levels)

Idents(dir_D30_all) <- "seurat_clusters"
new.cluster.ids <- c("T1 Melanocyte", "T1 Melanocyte")
Idents(dir_D30_all) <- plyr::mapvalues(x = Idents(dir_D30_all), from = cluster_levels, to = new.cluster.ids)
dir_D30_all$Cell.Type <- Idents(dir_D30_all)
```

###Subclustering NCCs
```{r D6 Subclustering NCCs}
Idents(D6_all) <- "Cell.Type"
DefaultAssay(D6_all) <- "RNA"
D6_cr <- subset(D6_all, idents = c("S1 NCC"))
D6_cr <- NormalizeData(D6_cr)
D6_cr <- FindVariableFeatures(D6_cr)
D6_cr <- ScaleData(D6_cr, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
D6_cr <- RunPCA(D6_cr)
ElbowPlot(D6_cr, ndims = 30) + geom_hline(yintercept = 2.75)
num.pcs <- length(D6_cr@reductions[["pca"]]@stdev[D6_cr@reductions[["pca"]]@stdev > 2.75])
D6_cr <- RunUMAP(D6_cr, reduction = "pca", dims = 1:num.pcs)
D6_cr <- FindNeighbors(D6_cr, dims = 1:num.pcs)
D6_cr <- FindClusters(D6_cr, resolution = .2)
DimPlot(D6_cr, reduction = "pca", label = TRUE)+NoLegend()
DimPlot(D6_cr, reduction = "umap", label = TRUE)+NoLegend()

#Annotate
Idents(D6_cr) <- "seurat_clusters"
cluster_levels <- c(0)
D6_cr@meta.data$seurat_clusters <- factor(x = Idents(D6_cr), levels = cluster_levels)

Idents(D6_cr) <- "seurat_clusters"
new.cluster.ids <- c("S1 NCC")
Idents(D6_cr) <- plyr::mapvalues(x = Idents(D6_cr), from = cluster_levels, to = new.cluster.ids)
D6_cr$Subtype <- Idents(D6_cr)
```

```{r D15 Subclustering NCCs}
Idents(indir_D15_all) <- "Cell.Type"
DefaultAssay(indir_D15_all) <- "RNA"
indir_D15_cr <- subset(indir_D15_all, idents = c("S2 NCC"))
indir_D15_cr <- NormalizeData(indir_D15_cr)
indir_D15_cr <- FindVariableFeatures(indir_D15_cr)
indir_D15_cr <- ScaleData(indir_D15_cr, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D15_cr <- RunPCA(indir_D15_cr)
ElbowPlot(indir_D15_cr, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D15_cr@reductions[["pca"]]@stdev[indir_D15_cr@reductions[["pca"]]@stdev > 2])
indir_D15_cr <- RunUMAP(indir_D15_cr, reduction = "pca", dims = 1:num.pcs)
indir_D15_cr <- FindNeighbors(indir_D15_cr, dims = 1:num.pcs)
indir_D15_cr <- FindClusters(indir_D15_cr, resolution = .2)
DimPlot(indir_D15_cr, reduction = "pca", label = TRUE)+NoLegend()
DimPlot(indir_D15_cr, reduction = "umap", label = TRUE)+NoLegend()
VlnPlot(indir_D15_cr, features = "SIX1")

###Clusters 2 is high for  trigeminal markers, remove
indir_D15_cr <- subset(indir_D15_cr, idents = 2, invert = TRUE)
indir_D15_cr <- NormalizeData(indir_D15_cr)
indir_D15_cr <- FindVariableFeatures(indir_D15_cr)
indir_D15_cr <- ScaleData(indir_D15_cr, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D15_cr <- RunPCA(indir_D15_cr)
ElbowPlot(indir_D15_cr, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D15_cr@reductions[["pca"]]@stdev[indir_D15_cr@reductions[["pca"]]@stdev > 2])
indir_D15_cr <- RunUMAP(indir_D15_cr, reduction = "pca", dims = 1:num.pcs)
indir_D15_cr <- FindNeighbors(indir_D15_cr, dims = 1:num.pcs)
indir_D15_cr <- FindClusters(indir_D15_cr, resolution = .2)
DimPlot(indir_D15_cr, reduction = "pca", label = TRUE)+NoLegend()
DimPlot(indir_D15_cr, reduction = "umap", label = TRUE)+NoLegend()

#Annotate
Idents(indir_D15_cr) <- "seurat_clusters"
cluster_levels <- c(0,1)
indir_D15_cr@meta.data$seurat_clusters <- factor(x = Idents(indir_D15_cr), levels = cluster_levels)

Idents(indir_D15_cr) <- "seurat_clusters"
new.cluster.ids <- c("S2 NCC1", "S2 NCC2")
Idents(indir_D15_cr) <- plyr::mapvalues(x = Idents(indir_D15_cr), from = cluster_levels, to = new.cluster.ids)
indir_D15_cr$Subtype <- Idents(indir_D15_cr)
```

```{r D22 Subclustering NCCs}
Idents(indir_D22_all) <- "Cell.Type"
DefaultAssay(indir_D22_all) <- "RNA"
indir_D22_cr <- subset(indir_D22_all, idents = c("S3 NCC"))
indir_D22_cr <- NormalizeData(indir_D22_cr)
indir_D22_cr <- FindVariableFeatures(indir_D22_cr)
indir_D22_cr <- ScaleData(indir_D22_cr, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D22_cr <- RunPCA(indir_D22_cr)
ElbowPlot(indir_D22_cr, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D22_cr@reductions[["pca"]]@stdev[indir_D22_cr@reductions[["pca"]]@stdev > 2])
indir_D22_cr <- RunUMAP(indir_D22_cr, reduction = "pca", dims = 1:num.pcs)
indir_D22_cr <- FindNeighbors(indir_D22_cr, dims = 1:num.pcs)
indir_D22_cr <- FindClusters(indir_D22_cr, resolution = .1)
DimPlot(indir_D22_cr, reduction = "pca", label = TRUE)+NoLegend()
DimPlot(indir_D22_cr, reduction = "umap", label = TRUE)+NoLegend()
VlnPlot(indir_D22_cr, features = "SIX1")

###Cluster 2 trigeminal, cluster 4 mesenchymal
indir_D22_cr <- subset(indir_D22_cr, idents = c(2,4), invert = TRUE)
indir_D22_cr <- NormalizeData(indir_D22_cr)
indir_D22_cr <- FindVariableFeatures(indir_D22_cr)
indir_D22_cr <- ScaleData(indir_D22_cr, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D22_cr <- RunPCA(indir_D22_cr)
ElbowPlot(indir_D22_cr, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D22_cr@reductions[["pca"]]@stdev[indir_D22_cr@reductions[["pca"]]@stdev > 2])
indir_D22_cr <- RunUMAP(indir_D22_cr, reduction = "pca", dims = 1:num.pcs)
indir_D22_cr <- FindNeighbors(indir_D22_cr, dims = 1:num.pcs)
indir_D22_cr <- FindClusters(indir_D22_cr, resolution = .15)
DimPlot(indir_D22_cr, reduction = "pca", label = TRUE)+NoLegend()
DimPlot(indir_D22_cr, reduction = "umap", label = TRUE)+NoLegend()

#Annotate
Idents(indir_D22_cr) <- "seurat_clusters"
cluster_levels <- c(0,1,3,2)
indir_D22_cr@meta.data$seurat_clusters <- factor(x = Idents(indir_D22_cr), levels = cluster_levels)

Idents(indir_D22_cr) <- "seurat_clusters"
new.cluster.ids <- c("S3 NCC1", "S3 NCC2", "S3 NCC3", "S3 NCC1")
Idents(indir_D22_cr) <- plyr::mapvalues(x = Idents(indir_D22_cr), from = cluster_levels, to = new.cluster.ids)
indir_D22_cr$Subtype <- Idents(indir_D22_cr)
```

```{r D30 Subclustering NCCs}
Idents(indir_D30_all) <- "Cell.Type"
DefaultAssay(indir_D30_all) <- "RNA"
indir_D30_cr <- subset(indir_D30_all, idents = c("S4 NCC"))
indir_D30_cr <- NormalizeData(indir_D30_cr)
indir_D30_cr <- FindVariableFeatures(indir_D30_cr)
indir_D30_cr <- ScaleData(indir_D30_cr, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D30_cr <- RunPCA(indir_D30_cr)
ElbowPlot(indir_D30_cr, ndims = 30) + geom_hline(yintercept = 3)
num.pcs <- length(indir_D30_cr@reductions[["pca"]]@stdev[indir_D30_cr@reductions[["pca"]]@stdev > 3])
indir_D30_cr <- RunUMAP(indir_D30_cr, reduction = "pca", dims = 1:num.pcs)
indir_D30_cr <- FindNeighbors(indir_D30_cr, dims = 1:num.pcs)
indir_D30_cr <- FindClusters(indir_D30_cr, resolution = .2)
DimPlot(indir_D30_cr, reduction = "pca", label = TRUE)+NoLegend()
DimPlot(indir_D30_cr, reduction = "umap", label = TRUE)+NoLegend()

#Annotate
Idents(indir_D30_cr) <- "seurat_clusters"
cluster_levels <- c(0,2,1)
indir_D30_cr@meta.data$seurat_clusters <- factor(x = Idents(indir_D30_cr), levels = cluster_levels)

Idents(indir_D30_cr) <- "seurat_clusters"
new.cluster.ids <- c("S4 NCC1", "S4 NCC2", "S4 NCC3")
Idents(indir_D30_cr) <- plyr::mapvalues(x = Idents(indir_D30_cr), from = cluster_levels, to = new.cluster.ids)
indir_D30_cr$Subtype <- Idents(indir_D30_cr)
```

```{r D41 Subclustering NCCs}
Idents(indir_D41_all) <- "Cell.Type"
DefaultAssay(indir_D41_all) <- "RNA"
indir_D41_cr <- subset(indir_D41_all, idents = c("S5 NCC"))
indir_D41_cr <- NormalizeData(indir_D41_cr)
indir_D41_cr <- FindVariableFeatures(indir_D41_cr)
indir_D41_cr <- ScaleData(indir_D41_cr, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
indir_D41_cr <- RunPCA(indir_D41_cr)
ElbowPlot(indir_D41_cr, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D41_cr@reductions[["pca"]]@stdev[indir_D41_cr@reductions[["pca"]]@stdev > 2])
indir_D41_cr <- RunUMAP(indir_D41_cr, reduction = "pca", dims = 1:num.pcs)
indir_D41_cr <- FindNeighbors(indir_D41_cr, dims = 1:num.pcs)
indir_D41_cr <- FindClusters(indir_D41_cr, resolution = .15)
DimPlot(indir_D41_cr, reduction = "pca", label = TRUE)+NoLegend()
DimPlot(indir_D41_cr, reduction = "umap", label = TRUE)+NoLegend()

#Annotate (Based on comparison to Majd. LP Dataset Fig S1F)
Idents(indir_D41_cr) <- "seurat_clusters"
cluster_levels <- c(0,1,2)
indir_D41_cr@meta.data$seurat_clusters <- factor(x = Idents(indir_D41_cr), levels = cluster_levels)

Idents(indir_D41_cr) <- "seurat_clusters"
new.cluster.ids <- c("Early SCs", "SCPs", "SCPDs")
Idents(indir_D41_cr) <- plyr::mapvalues(x = Idents(indir_D41_cr), from = cluster_levels, to = new.cluster.ids)
indir_D41_cr$Subtype <- Idents(indir_D41_cr)
```

```{r Primed D15 Subclustering NCCs}
Idents(dir_D15_all) <- "Cell.Type"
DefaultAssay(dir_D15_all) <- "RNA"
dir_D15_cr <- subset(dir_D15_all, idents = c("Primed Early NCC"))
dir_D15_cr <- NormalizeData(dir_D15_cr)
dir_D15_cr <- FindVariableFeatures(dir_D15_cr)
dir_D15_cr <- ScaleData(dir_D15_cr, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
dir_D15_cr <- RunPCA(dir_D15_cr)
ElbowPlot(dir_D15_cr, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(dir_D15_cr@reductions[["pca"]]@stdev[dir_D15_cr@reductions[["pca"]]@stdev > 2])
dir_D15_cr <- RunUMAP(dir_D15_cr, reduction = "pca", dims = 1:num.pcs)
dir_D15_cr <- FindNeighbors(dir_D15_cr, dims = 1:num.pcs)
dir_D15_cr <- FindClusters(dir_D15_cr, resolution = .3)
DimPlot(dir_D15_cr, reduction = "pca", label = TRUE)+NoLegend()
DimPlot(dir_D15_cr, reduction = "umap", label = TRUE)+NoLegend()

###Clusters 4 is high for  mesenchymal markers, remove
dir_D15_cr <- subset(dir_D15_cr, idents = 4, invert = TRUE)
dir_D15_cr <- NormalizeData(dir_D15_cr)
dir_D15_cr <- FindVariableFeatures(dir_D15_cr)
dir_D15_cr <- ScaleData(dir_D15_cr, vars.to.regress = c("S.Score", "G2M.Score", "novelty.score", "percent.ribo"))
dir_D15_cr <- RunPCA(dir_D15_cr)
ElbowPlot(dir_D15_cr, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(dir_D15_cr@reductions[["pca"]]@stdev[dir_D15_cr@reductions[["pca"]]@stdev > 2])
dir_D15_cr <- RunUMAP(dir_D15_cr, reduction = "pca", dims = 1:num.pcs)
dir_D15_cr <- FindNeighbors(dir_D15_cr, dims = 1:num.pcs)
dir_D15_cr <- FindClusters(dir_D15_cr, resolution = .3)
DimPlot(dir_D15_cr, reduction = "pca", label = TRUE)+NoLegend()
DimPlot(dir_D15_cr, reduction = "umap", label = TRUE)+NoLegend()

#Annotate
Idents(dir_D15_cr) <- "seurat_clusters"
cluster_levels <- c(0,1)
dir_D15_cr@meta.data$seurat_clusters <- factor(x = Idents(dir_D15_cr), levels = cluster_levels)

Idents(dir_D15_cr) <- "seurat_clusters"
new.cluster.ids <- c("Primed S2 NCC1", "Primed S2 NCC2")
Idents(dir_D15_cr) <- plyr::mapvalues(x = Idents(dir_D15_cr), from = cluster_levels, to = new.cluster.ids)
dir_D15_cr$Subtype <- Idents(dir_D15_cr)
```

```{r Merge T1 and T2 Melanocytes}
DefaultAssay(dir_D30_all) <- "RNA"
DefaultAssay(indir_D59_all) <- "RNA"
mature.melanocytes <- merge(dir_D30_all, indir_D59_all)
mature.melanocytes <- NormalizeData(mature.melanocytes)
mature.melanocytes <- FindVariableFeatures(mature.melanocytes, nfeatures = 3000)
mature.melanocytes <- ScaleData(mature.melanocytes, vars.to.regress = c("novelty.score", "S.Score", "G2M.Score", "percent.ribo"))
mature.melanocytes <- RunPCA(mature.melanocytes)
ElbowPlot(mature.melanocytes, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(mature.melanocytes@reductions[["pca"]]@stdev[mature.melanocytes@reductions[["pca"]]@stdev > 2])
mature.melanocytes <- RunUMAP(mature.melanocytes, reduction = "pca", dims = 1:num.pcs)
mature.melanocytes <- FindNeighbors(mature.melanocytes, dims = 1:num.pcs)
mature.melanocytes <- FindClusters(mature.melanocytes, resolution = .1)

DimPlot(mature.melanocytes, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(mature.melanocytes, reduction = "umap", label = TRUE) + theme(legend.position = "none")
```

###Running SCENIC
```{r Scenic Pipeline}
library(SCENIC)
setwd(dir = "/Users/fattahilab/Desktop/Manuscript Figures")
datasets <- c("D6_all", "indir_D15_cr", "indir_D22_cr", "indir_D30_cr", "indir_D41_cr")
for(i in 1:length(datasets)){
  dir.create(datasets[i])
  seur_obj <- get(datasets[i])
  
  #get counts matrix and metadata for input
  seur_obj.exprMat <- as.matrix(GetAssayData(seur_obj, assay = "alra", slot = "data")) # normalized data matrix
  seur_obj.cellInfo <- seur_obj@meta.data
  seur_obj.cellInfo <- data.frame(seur_obj.cellInfo)
  cbind(table(seur_obj.cellInfo$Subtype))
  dir.create(paste0(datasets[1], "/int"))
  saveRDS(seur_obj.cellInfo, file=paste0(datasets[i], "/int/", datasets[i], ".cellInfo.Rds"))

  org <- "hgnc" # or mgi, or dmel
  dbDir <- "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/SCENIC/cisTarget_databases" # RcisTarget databases location
  myDatasetTitle <- paste0("SCENIC", datasets[i]) # choose a name for your analysis
  data(defaultDbNames)
  dbs <- defaultDbNames[[org]]
  seur_obj.scenicOptions <- initializeScenic(org=org, dbDir=dbDir, dbs=dbs, datasetTitle=myDatasetTitle, nCores=10) 

  # Modify if needed
  seur_obj.scenicOptions@inputDatasetInfo$cellInfo <- paste0(datasets[i], "/int/", datasets[i], ".cellInfo.Rds")
  #seur_obj.scenicOptions@inputDatasetInfo$colVars <- paste0(datasets[i], "/int/", datasets[i], ".colVars.Rds")
  # Databases:
  # seur_obj.scenicOptions@settings$dbs <- c("mm9-5kb-mc8nr"="mm9-tss-centered-5kb-10species.mc8nr.feather")
  # seur_obj.scenicOptions@settings$db_mcVersion <- "v8"

  # Save to use at a later time...
  saveRDS(seur_obj.scenicOptions, file=paste0(datasets[i], "/int/", datasets[i], ".scenicOptions.RDS"))

  # (Adjust minimum values according to your dataset)
  seur_obj.genesKept <- geneFiltering(seur_obj.exprMat, scenicOptions=seur_obj.scenicOptions,
                           minCountsPerGene=3*.01*ncol(seur_obj.exprMat),
                           minSamples=ncol(seur_obj.exprMat)*.01)

  seur_obj.exprMat_filtered <- seur_obj.exprMat[seur_obj.genesKept, ]
  dim(seur_obj.exprMat)

  runCorrelation(seur_obj.exprMat_filtered, seur_obj.scenicOptions)

  # Optional: add log (if it is not logged/normalized already)
  #exprMat_filtered <- log2(exprMat_filtered+1) 

  # Run GENIE3
  runGenie3(seur_obj.exprMat_filtered, seur_obj.scenicOptions)

  seur_obj.scenicOptions@settings$verbose <- TRUE
  seur_obj.scenicOptions@settings$nCores <- 10
  seur_obj.scenicOptions@settings$seed <- 123

  seur_obj.scenicOptions <- runSCENIC_1_coexNetwork2modules(seur_obj.scenicOptions)
  seur_obj.scenicOptions <- runSCENIC_2_createRegulons(seur_obj.scenicOptions)
  seur_obj.scenicOptions <- runSCENIC_3_scoreCells(seur_obj.scenicOptions, seur_obj.exprMat, skipHeatmap = TRUE)

  saveRDS(seur_obj.scenicOptions, file=paste0(datasets[i], "/int/", datasets[i], ".scenicOptions.RDS")) # To save status
}
```

###Running CellChat
```{r CellChat Pipeline}
library(CellChat)
datasets <- c("D6_all", "indir_D15_cr", "indir_D22_cr", "indir_D30_cr")
cluster.levels <- c("Cell.Type", "Subtype", "Subtype", "Subtype")
for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  Idents(seur_obj) <- cluster.levels[i]
  data.input <- GetAssayData(seur_obj, assay = "alra", slot = "data") # normalized data matrix
  full.meta <- seur_obj@meta.data

  cellchat_obj <- createCellChat(object = data.input, meta = full.meta, group.by = cluster.levels[i])

  CellChatDB <- CellChatDB.human # use CellChatDB.mouse if running on mouse data

  # use a subset of CellChatDB for cell-cell communication analysis
  #CellChatDB.use <- subsetDB(CellChatDB, search = "Secreted Signaling") # use Secreted Signaling
  # use all CellChatDB for cell-cell communication analysis
  CellChatDB.use <- CellChatDB # simply use the default CellChatDB

  # set the used database in the object
  cellchat_obj@DB <- CellChatDB.use

  # subset the expression data of signaling genes for saving computation cost
  cellchat_obj <- subsetData(cellchat_obj) # This step is necessary even if using the whole database
  future::plan("multiprocess", workers = 4) # do parallel
  #> Warning: [ONE-TIME WARNING] Forked processing ('multicore') is disabled
  #> in future (>= 1.13.0) when running R from RStudio, because it is
  #> considered unstable. Because of this, plan("multicore") will fall
  #> back to plan("sequential"), and plan("multiprocess") will fall back to
  #> plan("multisession") - not plan("multicore") as in the past. For more details,
  #> how to control forked processing or not, and how to silence this warning in
  #> future R sessions, see ?future::supportsMulticore
  cellchat_obj <- identifyOverExpressedGenes(cellchat_obj)
  cellchat_obj <- identifyOverExpressedInteractions(cellchat_obj)
  # project gene expression data onto PPI network (optional)
  cellchat_obj <- projectData(cellchat_obj, PPI.human)


  cellchat_obj <- computeCommunProb(cellchat_obj)
  # Filter out the cell-cell communication if there are only few number of cells in certain cell groups
  cellchat_obj <- filterCommunication(cellchat_obj, min.cells = 10)

  cellchat_obj <- computeCommunProbPathway(cellchat_obj)
  D6_df.net <- subsetCommunication(cellchat_obj)
  cellchat_obj <- aggregateNet(cellchat_obj)
  saveRDS(cellchat_obj, file = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/", datasets[i], "_cellchat.RDS"))
}
```

###Running Monocle3
```{r Monocle Pipeline}
library(monocle3)
datasets <- c("D6_all", "indir_D15_cr", "indir_D22_cr", "indir_D30_cr")
dims.number <- c(45, 23, 13, 23)

for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  exprs <- GetAssayData(object = seur_obj, assay = "alra")
  cells <- colnames(GetAssayData(object = seur_obj))
  exprs_mat <- exprs[,match(cells, colnames(exprs))]
  pheno.data = seur_obj@meta.data
  genes <- data.frame(gene_short_name = rownames(exprs_mat))
  rownames(genes) <- rownames(exprs_mat)

  # Make CellDataSet object
  cds <- new_cell_data_set(as(exprs_mat,"sparseMatrix"),
                         cell_metadata = pheno.data,
                         gene_metadata = genes)


  ## Step 1: Normalize and pre-process the data
  cds <- preprocess_cds(cds, num_dim = dims.number[i], norm_method = "none")
  plot_pc_variance_explained(cds)

  ## Step 2: Remove batch effects with cell alignment
  cds <- align_cds(cds, alignment_group = "Phase")

  ## Step 3: Reduce the dimensions using UMAP
  cds <- reduce_dimension(cds)

  ## Step 4: Cluster the cells
  cds <- cluster_cells(cds, resolution = .0004)

  ## Step 5: Learn a graph
  cds <- learn_graph(cds)

  ## Step 6: Order cells
  cds <- order_cells(cds)
  saveRDS(cds, file = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/", datasets[i], "_monocle3.RDS"))
}
```

```{r Color Palettes}
D6_celltype_cols <- c("#DB9D85FF", "#C7A76CFF", "gray10", "gray33")
indir_D15_celltype_cols <- c("#ABB065FF", "gray10", "gray33", "gray57", "gray80")
indir_D22_celltype_cols <- c("#86B875FF", "gray57", "gray80")
indir_D30_celltype_cols <- c("#5CBD92FF", "gray80")
indir_D41_celltype_cols <- c("#39BEB1FF", "#E495A5FF", "gray80")
indir_D59_celltype_cols <- c("#E093C3FF")
dir_D15_celltype_cols <- c("#4CB9CCFF", "gray10", "gray33", "gray80")
dir_D30_celltype_cols <- c("#7DB0DDFF")

D6_subtype_cols <- c("#089392FF")
indir_D15_subtype_cols <- c("#0CA291FF", "#CBD98EFF")
indir_D22_subtype_cols <- c("#33B08DFF", "#EAE29CFF", "#EABC76FF")
indir_D30_subtype_cols <- c("#59BD86FF", "#E57F6CFF", "#E9A96CFF")
indir_D41_subtype_cols <- c("#82C782FF", "#DF6974FF", "#CF597EFF")
dir_D15_subtype_cols <- c("#0CA291FF", "#CBD98EFF")

paths <- c("#7EAEDC","#A2D1FF", "#FFD1FF","#D1A2D1")
```

###Figure1
```{r Figure 1C}
datasets <- c("D6_all", "indir_D15_all", "indir_D22_all", "indir_D30_all", "indir_D41_all")
color.palettes <- c("D6_celltype_cols", "indir_D15_celltype_cols", "indir_D22_celltype_cols", "indir_D30_celltype_cols", "indir_D41_celltype_cols")
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures/"

#UMAPs
for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  cols <- get(color.palettes[i])
  DimPlot(seur_obj, reduction = "umap", group.by = "Cell.Type", pt.size = 2, cols = cols) + theme_void() +theme(legend.position = "none")+ggtitle("")
  ggsave(paste0(outputDir, "/UMAP.", datasets[i], ".Cell.Type.jpeg"), width = 4, height = 4, dpi = 320)
}

#SOX10 Expression
for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  DotPlot(seur_obj, features = "SOX10", assay = "alra", scale = F, scale.min = 0, scale.max = 100) + theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust = 1, size = 18), axis.text.y = element_text(size = 18), axis.title.y = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), title = element_text(size = 18))+ scale_color_gradientn(colours = rev(brewer.pal(10, "Spectral")))
  ggsave(paste0(outputDir, "/DotPlot.", datasets[i], ".SOX10.pdf"), width = 6, height = 4, dpi = 320)
}

#CellType Proportion Plots
for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  cols <- get(color.palettes[i])
  ggplot(seur_obj@meta.data, aes(x = sample, fill = factor(seur_obj$Cell.Type, levels=rev(levels(seur_obj$Cell.Type))))) + geom_bar(position = "fill") +theme_minimal()  + theme(axis.text.x = element_blank(), axis.title.x = element_text(size = 15),legend.text = element_text(size = 15), legend.title = element_text(size = 15), axis.text.y.left = element_text(size = 15, colour = "black"), axis.title.y = element_text(size = 15)) + ylab("Proportion") + scale_fill_manual(values = rev(cols)) +xlab("") +NoLegend()
  ggsave(paste0(outputDir, "/BarGraph.",datasets[i], "cell.type.proportions" ,".pdf"), width = 2, height = 4, dpi = 320, limitsize = FALSE)
}
```

```{r Figure 1D}
datasets <- c("D6_cr", "indir_D15_cr", "indir_D22_cr", "indir_D30_cr", "indir_D41_cr")
color.palettes <- c("D6_subtype_cols", "indir_D15_subtype_cols", "indir_D22_subtype_cols", "indir_D30_subtype_cols", "indir_D41_subtype_cols")
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures/"

#UMAPs
for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  cols <- get(color.palettes[i])
  DimPlot(seur_obj, reduction = "umap", group.by = "Subtype", pt.size = 2, cols = cols) + theme_void() +theme(legend.position = "none")+ggtitle("")
  ggsave(paste0(outputDir, "/UMAP.", datasets[i], ".Subtype.jpeg"), width = 4, height = 4, dpi = 320)
}
```

```{r Figures 1E-G and S1G-H DotPlots}
Uncommited <- c("SOX10", "SOX2", "FOXD3", "ETS1", "SNAI1", "SNAI2", "TFAP2A", "TFAP2B")
Neuronal <- c("POU4F1", "NEUROG1", "ASCL1", "PHOX2A", "PHOX2B", "HAND2")
Melanogenic <- c("MITF", "PAX3", "MEF2C")
Mesenchymal <- c("SOX9","TWIST1", "PRRX1")
Glial <- c("EGR2", "MSTN", "SOX8", "POU3F1", "NFATC4")
all.lineal.markers <- c(Uncommited, Neuronal, Melanogenic, Glial, Mesenchymal)

all.NCCs <- NormalizeData(all.NCCs)
all.NCCs <- FindVariableFeatures(all.NCCs)
all.NCCs <- ScaleData(all.NCCs, vars.to.regress = c("novelty.score", "S.Score", "G2M.Score", "percent.ribo"))

DotPlot(all.NCCs, features = rev(all.lineal.markers), assay = "alra", group.by = "Subtype", scale = F) + theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust = 1, size = 18), axis.text.y = element_text(size = 18), axis.title.y = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), title = element_text(size = 18))+ ggtitle("NCC Lineal Markers")+coord_flip()+ scale_color_gradientn(colours = rev(brewer.pal(10, "Spectral")))
ggsave(paste0(outputDir, "/NCC_lineal_markers_dotplot.pdf"), width = 7, height = 8, dpi = 320)
```

```{r Figures 1E-G and S1G-H TF Activity Heatmaps}
datasets <- c("D6_all", "indir_D15_cr", "indir_D22_cr", "indir_D30_cr", "indir_D41_cr")
dataset.groupings <- c(23,27,29,27,26)
merged.regulons <- data.frame()
for(i in 1:length(datasets)){
  setwd(dir = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/SCENIC/SCENIC Datasets/", datasets[i], "/"))
  scenicOptions <- readRDS(paste0("int/", datasets[i], ".scenicOptions.RDS"))
  cellInfo <- readRDS(paste0("int/", datasets[i], ".cellInfo.Rds"))
  tSNE_scenic <- readRDS(tsneFileName(scenicOptions))
  aucell_regulonAUC <- loadInt(scenicOptions, "aucell_regulonAUC")
  aucell_regulonAUC <- aucell_regulonAUC[onlyNonDuplicatedExtended(rownames(aucell_regulonAUC)),]
  regulonActivity_byGrouping <- sapply(split(rownames(cellInfo), cellInfo[,dataset.groupings[i]]),
                                      function(cells) rowMeans(getAUC(aucell_regulonAUC)[,cells]))
  rownames(regulonActivity_byGrouping) <- sub(" .*", "", rownames(regulonActivity_byGrouping))
  merged.regulons <- merge(merged.regulons, regulonActivity_byGrouping, by = 0, all = TRUE)
  rownames(merged.regulons) <- merged.regulons$Row.names
  merged.regulons <- merged.regulons[,-1]
}

merged.regulons[is.na(merged.regulons)] <- 0
merged.regulons <- as.matrix(merged.regulons)
merged.regulons.all <- merged.regulons[,c(2,5:12,14,13,15)]

NCC <- c("SOX10", "SOX2", "FOXD3", "ETS1", "SNAI1", "SNAI2", "TFAP2A", "TFAP2B")
Neuronal <- c("POU4F1", "NEUROG1", "ASCL1", "PHOX2A", "PHOX2B", "HAND2")
Melanogenic <- c("MITF", "PAX3", "MEF2C")
Mesenchymal <- c("SOX9","TWIST1", "PRRX1")
Glial <- c("EGR2", "MSTN", "SOX8", "POU3F1", "NFATC4")


TFs <- c(NCC, Neuronal, Melanogenic, Glial, Mesenchymal)
TFs.extended <- gsub("$", "_extended", TFs)
Lineal.regulons <- c()
for(i in 1:length(TFs)){
  Lineal.regulons <- append(Lineal.regulons, TFs[i])
  Lineal.regulons <- append(Lineal.regulons, TFs.extended[i])
}

merged.regulons.all.sub <- merged.regulons.all[rownames(merged.regulons.all) %in% Lineal.regulons,]
merged.regulons.all.sub <- merged.regulons.all.sub[match(Lineal.regulons, rownames(merged.regulons.all.sub)),] 
merged.regulons.all.sub[is.na(merged.regulons.all.sub)] <- 0
rownames(merged.regulons.all.sub) <- Lineal.regulons
ht <- Heatmap(merged.regulons.all.sub, name="Regulon activity", cluster_rows = F, cluster_columns = F, col = rev(brewer.pal(10, "Spectral")))
pdf("/Users/fattahilab/Desktop/Manuscript Figures/NCC_lineal_TFactivity_heatmap.pdf", width=10, height=15)
draw(ht)
dev.off()
```

```{r Figure S1A}
DefaultAssay(D6_all) <- "RNA"
DefaultAssay(indir_D15_all) <- "RNA"
DefaultAssay(indir_D22_all) <- "RNA"
DefaultAssay(indir_D30_all) <- "RNA"
DefaultAssay(indir_D41_all) <- "RNA"
all.progenitors <- merge(D6_all, c(indir_D15_all, indir_D22_all, indir_D30_all, indir_D41_all))
all.progenitors <- NormalizeData(all.progenitors)
Idents(all.progenitors) <- "Cell.Type"
cluster_levels <- c("Pre-migratory NCC", "S1 NCC", "S2 NCC", "S3 NCC", "S4 NCC", "S5 NCC", "Mature SCs", "CNS Precursor", "Non-Neural Ectoderm", "Placode", "Mesenchymal")
all.progenitors$Cell.Type <- factor(x = Idents(all.progenitors), levels = cluster_levels)

#Cell.Type Identifiers
premigratory.NCC <- c("PAX3", "WNT1", "LMX1A")
migratory.NCC <- c("SOX10", "FOXD3", "MEF2C", "ETS1", "TFAP2B", "EDNRA", "NGFR", "SNAI2")
CNSprecursor <- c("PAX6", "SIX3", "FEZF2", "EMX2", "NEUROG2")
nonnueral.ectoderm <- c("GATA3", "DLX3", "EPCAM", "CDH1", "WNT6")
NCC <- c("SOX10", "FOXD3", "EDNRB", "ERBB3", "ITGA4")
mesenchymal <- c("TWIST1", "COL3A1", "PCOLCE", "PDGFRA", "PDGFRB")
melanoblast <- c("MITF", "SOX10", "KIT", "EDNRB", "SNAI2")
melanocyte <- c("MITF", "SOX10", "DCT", "TRPM1", "PMEL", "MLANA", "TYRP1", "TYR")
trigeminal.placode <- c("SIX1", "POU4F1", "ISL1")
SMP <- c("SOX10", "CAV1", "MITF")
Schwann.Cell <- c("GFRA1", "GFRA3", "S100B", "CD44", "MBP")
cell.type.annotations <- unique(c(premigratory.NCC, migratory.NCC, NCC, Schwann.Cell, CNSprecursor, nonnueral.ectoderm, trigeminal.placode, mesenchymal))

DotPlot(all.progenitors, features = cell.type.annotations, assay = "alra", scale = F, group.by = "Cell.Type") + theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust = 1, size = 18), axis.text.y = element_text(size = 18), axis.title.y = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), title = element_text(size = 18))+ scale_color_gradientn(colours = brewer.pal(10, "Reds"))
ggsave(paste0(outputDir, "/DotPlot.","Cell.Type.Annotation" ,".pdf"), width = 14, height = 4, dpi = 320, limitsize = FALSE)
```

```{r Figure S1B}
D6_all.cds <- readRDS(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/D6_all_monocle3.RDS")
plot_cells(D6_all.cds, color_cells_by = "Cell.Type",cell_size = 2, label_branch_points = F, label_roots = T, label_cell_groups = F, label_leaves = T, trajectory_graph_color = "black", trajectory_graph_segment_size = 1.5)+theme_void()+ scale_color_manual(values=D6_celltype_cols)+NoLegend()
ggsave(paste0(outputDir, "/D6_Cell.Type_monolce3.jpeg"), width = 4, height = 4, dpi = 320)
```

```{r Figure S1C}
D6_cellchat <- readRDS(cds, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/D6_cellchat.rds")
# Compute the network centrality scores
D6_cellchat <- netAnalysis_computeCentrality(D6_cellchat, slot.name = "netP") # the slot 'netP' means the inferred intercellular communication network of signaling pathways
# Visualize the computed centrality scores using heatmap, allowing ready identification of major signaling roles of cell groups
netAnalysis_signalingRole_heatmap(D6_cellchat, signaling = c("FGF", "BMP", "ncWNT", "WNT"), width = 4, height = 2.5, font.size = 10, color.use = D6_celltype_cols, color.heatmap = "YlOrRd", pattern = "incoming")

ht <- netAnalysis_signalingRole_heatmap(D6_cellchat, signaling = c("FGF", "BMP", "ncWNT", "WNT"), width = 4, height = 2.5, font.size = 10, color.use = D6_celltype_cols, color.heatmap = "YlOrRd", pattern = "incoming")
pdf("/Users/fattahilab/Desktop/Manuscript Figures/D6_incoming_signaling_heatmap.pdf", width=10, height=15)
draw(ht)
dev.off()
```

```{r Figure S1D}
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures/"
GO_gene_sets_human = msigdbr(species = "Homo sapiens", category = "C5")
GO_gene_sets_human = split(x = GO_gene_sets_human$gene_symbol, f = GO_gene_sets_human$gs_name)
GO_WNT <- GO_gene_sets_human[str_detect(names(GO_gene_sets_human), "WNT")]
GO_FGF <- GO_gene_sets_human[str_detect(names(GO_gene_sets_human), "FIBROBLAST")]
GO_BMP <- GO_gene_sets_human[str_detect(names(GO_gene_sets_human), "BMP")]

neg.regulation.pathways <- c(GO_FGF[c(11)], GO_BMP[c(2)], GO_WNT[c(8,7)])

for(i in 1:length(neg.regulation.pathways)){
  D6_all <- AddModuleScore(D6_all, features = neg.regulation.pathways[i], assay = "alra", name = names(neg.regulation.pathways)[i])
}
regulation.pathways.names <- names(neg.regulation.pathways)
regulation.pathways.names <- sub("$", "1", regulation.pathways.names)
DotPlot(D6_all, features = rev(regulation.pathways.names), group.by = "Cell.Type", assay = "alra", scale = FALSE, scale.min = 0) + theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust = 1), axis.title.x = element_blank(), axis.title.y = element_blank(), axis.text.y = element_text(size = 4)) +coord_flip() +scale_color_gradient2(low = "blue", high = "red", mid = "snow", midpoint = 0)
ggsave(paste0(outputDir, "/DotPlot", "Neg.Regulation.Scores", ".pdf"), width = 7, height = 5, dpi = 320)
```

```{r Figure S1F}
homa.D38 <- readRDS(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/S1_SCs.rds")
source("/Users/fattahilab/Documents/Single Cell Database Repository/SynologyDrive/Useful R Things/Functions/Ryans_Functions.R")

Idents(indir_D41_cr) <- "Subtype"
ESCs.cells <- WhichCells(indir_D41_cr, idents = "Early SCs")
SCP.cells <- WhichCells(indir_D41_cr, idents = "SCPs")
SCPD.cells <- WhichCells(indir_D41_cr, idents = "SCPDs")

Idents(indir_D41_all) <- "Cell.Type"
Mes.cells <- WhichCells(indir_D41_all, idents = "Mesenchymal")
SC.cells <- WhichCells(indir_D41_all, idents = "Mature SCs")

Idents(indir_D41_all) <- "Subtype"
Idents(object = indir_D41_all, cells = ESCs.cells) <- "Early SCs"
Idents(object = indir_D41_all, cells = SCP.cells) <- "SCPs"
Idents(object = indir_D41_all, cells = SCPD.cells) <- "SCPDs"
Idents(object = indir_D41_all, cells = Mes.cells) <- "Mesenchymal"
Idents(object = indir_D41_all, cells = SC.cells) <- "Mature SCs"

indir_D41_all <- AddMetaData(indir_D41_all, col.name = "Subtype", metadata = Idents(indir_D41_all))
levels <- c("Early SCs", "SCPs", "SCPDs", "Mature SCs", "Mesenchymal")
indir_D41_all@meta.data$Subtype <- factor(x = Idents(indir_D41_all), levels = levels)

Idents(indir_D41_all) <- "Subtype"
indir_D41_SCs <- subset(indir_D41_all, idents = "Mesenchymal", invert = T)

DefaultAssay(homa.D38) <- "RNA"
DefaultAssay(indir_D41_SCs) <- "RNA"
homa.D38$Day <- "D38"
indir_D41_SCs$Day <- "D41"
SCPs.merged <- merge(homa.D38, indir_D41_SCs)
homa.D38.all.genes <- subset(SCPs.merged, subset = Day == "D38")
indir_D41_SCs.all.genes <- subset(SCPs.merged, subset = Day == "D41")
homa.D38.all.genes <- NormalizeData(homa.D38.all.genes)
homa.D38.all.genes <- FindVariableFeatures(homa.D38.all.genes, selection.method = "vst", nfeature = 3000)
homa.D38.all.genes$Subtype <- homa.D38$seurat_clusters
indir_D41_SCs.all.genes <- NormalizeData(indir_D41_SCs.all.genes)
indir_D41_SCs.all.genes <- FindVariableFeatures(indir_D41_SCs.all.genes, selection.method = "vst", nfeature = 3000)
all.var.genes <- unique(c(VariableFeatures(homa.D38.all.genes), VariableFeatures(indir_D41_SCs.all.genes)))

Idents(homa.D38.all.genes) <- "Subtype"
cluster_levels <- c("S1 Early SCs", "S1 Mature SCs", "S1 SCPs", "S1 SCPDs")
homa.D38.all.genes@meta.data$Subtype <- factor(x = Idents(homa.D38.all.genes), levels = cluster_levels)
Idents(homa.D38.all.genes) <- "Subtype"
new.cluster.ids <- c("Early SCs", "Mature SCs", "SCPs", "SCPDs")
Idents(homa.D38.all.genes) <- plyr::mapvalues(x = Idents(homa.D38.all.genes), from = cluster_levels, to = new.cluster.ids)
homa.D38.all.genes$Subtype <- Idents(homa.D38.all.genes)

homa.D38.all.genes <- NormalizeData(homa.D38.all.genes)
homa.D38.all.genes <- FindVariableFeatures(homa.D38.all.genes)
homa.D38.all.genes <- ScaleData(homa.D38.all.genes, vars.to.regress = c("novelty.score", "S.Score", "G2M.Score", "percent.ribo"))
homa.D38.all.genes <- RunPCA(homa.D38.all.genes)
ElbowPlot(homa.D38.all.genes, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(homa.D38.all.genes@reductions[["pca"]]@stdev[homa.D38.all.genes@reductions[["pca"]]@stdev > 2])
homa.D38.all.genes <- RunUMAP(homa.D38.all.genes, reduction = "pca", dims = 1:num.pcs)
homa.D38.all.genes <- FindNeighbors(homa.D38.all.genes, dims = 1:num.pcs)
homa.D38.all.genes <- FindClusters(homa.D38.all.genes, resolution = .1)
DimPlot(homa.D38.all.genes, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(homa.D38.all.genes, reduction = "umap", label = TRUE) + theme(legend.position = "none")
DimPlot(homa.D38.all.genes, reduction = "umap", group.by = "Subtype", pt.size = 1) + theme(legend.text = element_text(size = 20), title = element_blank())+theme_void()

indir_D41_SCs.all.genes <- NormalizeData(indir_D41_SCs.all.genes)
indir_D41_SCs.all.genes <- FindVariableFeatures(indir_D41_SCs.all.genes)
indir_D41_SCs.all.genes <- ScaleData(indir_D41_SCs.all.genes, vars.to.regress = c("novelty.score", "S.Score", "G2M.Score", "percent.ribo"))
indir_D41_SCs.all.genes <- RunPCA(indir_D41_SCs.all.genes)
ElbowPlot(indir_D41_SCs.all.genes, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(indir_D41_SCs.all.genes@reductions[["pca"]]@stdev[indir_D41_SCs.all.genes@reductions[["pca"]]@stdev > 2])
indir_D41_SCs.all.genes <- RunUMAP(indir_D41_SCs.all.genes, reduction = "pca", dims = 1:num.pcs)
indir_D41_SCs.all.genes <- FindNeighbors(indir_D41_SCs.all.genes, dims = 1:num.pcs)
indir_D41_SCs.all.genes <- FindClusters(indir_D41_SCs.all.genes, resolution = .1)
DimPlot(indir_D41_SCs.all.genes, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(indir_D41_SCs.all.genes, reduction = "umap", label = TRUE) + theme(legend.position = "none")
Idents(indir_D41_SCs.all.genes) <- "Subtype"
levels <- c("Early SCs", "SCPs", "SCPDs", "Mature SCs")
indir_D41_SCs.all.genes@meta.data$Subtype <- factor(x = Idents(indir_D41_SCs.all.genes), levels = levels)
DimPlot(indir_D41_SCs.all.genes, reduction = "umap", group.by = "Subtype", pt.size = 1, cols = c(indir_D41_subtype_cols, indir_D41_celltype_cols[c(2)])) + theme(legend.text = element_text(size = 20), title = element_blank())+theme_void()

swne.projection(homa.D38.all.genes, indir_D41_SCs.all.genes, genes.embed = c("MITF", "SOX10"), reference.group.by="Subtype", query.group.by="Subtype", genes.use = all.var.genes, reference.colors = c("#7D9D35", "#D38B24", "#6671AD", "#E1CB1B"), query.colors = c(indir_D41_subtype_cols, indir_D41_celltype_cols[c(2)]))
```

```{r Figure S1I}
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures"
DefaultAssay(indir_D15_cr) <- "alra"
DefaultAssay(indir_D22_cr) <- "alra"
DefaultAssay(indir_D30_cr) <- "alra"
DefaultAssay(indir_D41_cr) <- "alra"

sox8.max <- max(max(indir_D15_cr@assays$alra[which(rownames(indir_D15_cr) == "SOX8"),]), max(indir_D22_cr@assays$alra[which(rownames(indir_D22_cr) == "SOX8"),]), max(indir_D30_cr@assays$alra[which(rownames(indir_D30_cr) == "SOX8"),]), max(indir_D41_cr@assays$alra[which(rownames(indir_D41_cr) == "SOX8"),]))
itga4.max <- max(max(indir_D15_cr@assays$alra[which(rownames(indir_D15_cr) == "ITGA4"),]), max(indir_D22_cr@assays$alra[which(rownames(indir_D22_cr) == "ITGA4"),]), max(indir_D30_cr@assays$alra[which(rownames(indir_D30_cr) == "ITGA4"),]), max(indir_D41_cr@assays$alra[which(rownames(indir_D41_cr) == "ITGA4"),]))
serpine2.max <- max(max(indir_D15_cr@assays$alra[which(rownames(indir_D15_cr) == "SERPINE2"),]), max(indir_D22_cr@assays$alra[which(rownames(indir_D22_cr) == "SERPINE2"),]), max(indir_D30_cr@assays$alra[which(rownames(indir_D30_cr) == "SERPINE2"),]), max(indir_D41_cr@assays$alra[which(rownames(indir_D41_cr) == "SERPINE2"),]))

datasets <- c("indir_D15_cr", "indir_D22_cr", "indir_D30_cr", "indir_D41_cr")
hub.markers <- c("SOX8", "ITGA4", "SERPINE2")
y.max <- c(sox8.max, itga4.max, serpine2.max)
for(i in 1:length(datasets)){
   seur_obj <- get(datasets[i])
   for(x in 1:length(hub.markers)){
     FeaturePlot(seur_obj, features = hub.markers[x], pt.size = 2, order = TRUE)+ scale_color_gradientn(colours = rev(brewer.pal(10, "Spectral")), limits = c(0,y.max[x]))+theme_void()+ggtitle("")+theme(legend.position = "none")
     ggsave(paste0(outputDir, "/",datasets[i], hub.markers[x], ".jpeg"), width = 4, height = 4.5, dpi = 320)
   }
}

for(x in 1:length(hub.markers)){
  FeaturePlot(indir_D15_cr, features = hub.markers[x], pt.size = 2, order = TRUE)+ scale_color_gradientn(colours = rev(brewer.pal(10, "Spectral")), limits = c(0,y.max[x]))+theme_void()+ggtitle("")+theme(legend.position = "right")
  ggsave(paste0(outputDir, "/legend.", hub.markers[x], ".jpeg"), width = 4, height = 4.5, dpi = 320)
}
```

```{r FIgure S1J}
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures"
sc <- import("scanpy")
adata <- sc$read_h5ad("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/GSE201257_adata_processed.h5ad")

ademayko <- LoadH5Seurat("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/GSE201257_adata_QC_filtered.h5seurat")
adameyko.meta <- adata$obs
cells.to.keep <- rownames(adameyko.meta)
adameyko.sub <- subset(ademayko, cells = cells.to.keep)

adameyko.sub <- AddMetaData(adameyko.sub,  adata$obs)
adameyko.sub <- NormalizeData(adameyko.sub)

embedding <- adata$obsm["X_umap"]
rownames(embedding) <- adata$obs_names$to_list()
colnames(embedding) <- c("umap_1", "umap_2")
adameyko.sub[["umap"]] <- CreateDimReducObject(embedding, key = "umap_")
DimPlot(adameyko.sub, reduction = "umap", group.by = "leiden", label = T) + theme(legend.position = "none")
FeaturePlot(adameyko.sub, features = "Mal", order = T, pt.size = 2)

Idents(adameyko.sub) <- "leiden"
cluster_levels <- c(2,6,7,9,3,4,11,14,19,1,17,12,10,22,0,21,5,28,18,29,20,16,26,8,15,24,27,13,23,25)
adameyko.sub@meta.data$leiden <- factor(x = Idents(adameyko.sub), levels = cluster_levels)

Idents(adameyko.sub) <- "leiden"
cluster_levels <- c(2,6,7,9,3,4,11,14,19,1,17,12,10,22,0,21,5,28,18,29,20,16,26,8,15,24,27,13,23,25)
new.cluster.ids <- c( "Cranial NCC", "Trunk NCC", "Trunk NCC", "Trunk NCC", "Hub", "Hub", "Hub", "Hub", "Hub", "Mesenchyme", "Mesenchyme", "Sensory Neuron", "Sensory Neuron", "Sensory Neuron", "Sympathetic Neuron", "Sympathetic Neuron", "ChC", "ChC", "Enteric Glia", "Enteric Neuron", "Enteric Neuron", "Satellite Glia", "Schwann Cell", "Schwann Cell", "Schwann Cell", "Schwann Cell", "Schwann Cell", "Endoneurial Fibroblast", "Endoneurial Fibroblast", "Melanocyte")
Idents(adameyko.sub) <- plyr::mapvalues(x = Idents(adameyko.sub), from = cluster_levels, to = new.cluster.ids)
adameyko.sub$Cell.Types <- Idents(adameyko.sub)
Idents(adameyko.sub) <- "Cell.Types"

oTab = utils_loadObject("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Analysis Tools/human_mouse_genes_Jul_24_2018.rda")
oTab <- as.data.frame(as.matrix(oTab[,1:2]))
oTab[nrow(oTab) + 1,] = c("SOX8", "Sox8")

S4NCC2.markers<-FindMarkers(indir_D30_cr, ident.1 = "S4 NCC2", assay = "RNA")
S4NCC2.markers$human <- rownames(S4NCC2.markers)
S4NCC2.markers.homologues <- merge(S4NCC2.markers, oTab, by = "human")
S4NCC2.markers.homologues <- S4NCC2.markers.homologues[order(S4NCC2.markers.homologues$p_val_adj, decreasing = F),]
adameyko.sub <- AddModuleScore(adameyko.sub, features = list(S4NCC2.markers.homologues$mouse[1:100]), name = "S4NCC2.markers.score")
FeaturePlot(adameyko.sub, features = "S4NCC2.markers.score1", pt.size = 1, order = TRUE)+ scale_color_gradientn(colours = rev(brewer.pal(10, "Spectral")))+theme_void()+ggtitle("S4 NCC2 Score")+theme(plot.title = element_text(hjust = .5, face = "bold", size = 25), legend.position = "top", legend.text = element_text(size = 10, angle = 90, hjust = 1))
ggsave(paste0(outputDir, "/adameyko.S4NCC2.Score.jpeg"), width = 4, height = 5, dpi = 320)
```

###Figure 2
```{r Figure 2A}
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures"
indir_D15_cr.cds <- readRDS(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/indir_D15_cr_monocle3.RDS")
indir_D22_cr.cds <- readRDS(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/indir_D22_cr_monocle3.RDS")
indir_D30_cr.cds <- readRDS(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/indir_D30_cr_monocle3.RDS")

datasets <- c("indir_D15_cr.cds", "indir_D22_cr.cds", "indir_D30_cr.cds")
color.palettes <- c("indir_D15_subtype_cols", "indir_D22_subtype_cols", "indir_D30_subtype_cols")
for(i in 1:length(datasets)){
  cds <- get(datasets[i])
  cols <- get(color.palettes[i])
  plot_cells(cds, color_cells_by = "Subtype",cell_size = 2, label_branch_points = F, label_roots = T, label_cell_groups = F, label_leaves = T, trajectory_graph_color = "black", trajectory_graph_segment_size = 1.5)+theme_void()+
  scale_color_manual(values=cols)+NoLegend()
  ggsave(paste0(outputDir, "/", datasets[i], "_Subtype_monocle3.jpeg"), width = 4, height = 4, dpi = 320)
  plot_cells(cds, genes = "MITF",cell_size = 2, label_branch_points = F, label_roots = T, label_cell_groups = F, label_leaves = T, trajectory_graph_color = "black", trajectory_graph_segment_size = 1.5)+theme_void()+NoLegend()
  ggsave(paste0(outputDir, "/", datasets[i], "_MITF_monocle3.jpeg"), width = 4, height = 4, dpi = 320)
  plot_cells(cds, genes = "MITF",cell_size = 2, label_branch_points = F, label_roots = T, label_cell_groups = F, label_leaves = T, trajectory_graph_color = "black", trajectory_graph_segment_size = 1.5)+theme_void()+theme(legend.position = "bottom")
  ggsave(paste0(outputDir, "/", datasets[i], "_MITF_legend_monocle3.pdf"), width = 4, height = 4, dpi = 320)
}
```

```{r Figure 2B}
swne.projection(indir_D22_cr.shared, indir_D15_cr.shared, genes.embed = c("MITF", "SOX10"), reference.group.by="Subtype", query.group.by="Subtype", reference.colors =  indir_D22_subtype_cols, query.colors = indir_D15_subtype_cols)
swne.projection(indir_D22_cr.shared, indir_D30_cr.shared, genes.embed = c("MITF", "SOX10"), reference.group.by="Subtype", query.group.by="Subtype", reference.colors = indir_D22_subtype_cols, query.colors = indir_D30_subtype_cols)
```

```{r Figure 2E and S2C TF Activity Module Idenitfication and Heatmap}
#set up TF activity matrix of only the uncommitted and melanogenic clusters
datasets <- c("indir_D15_cr", "indir_D22_cr", "indir_D30_cr")
dataset.groupings <- c(27,29,27)
merged.regulons <- data.frame()
for(i in 1:length(datasets)){
  setwd(dir = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/SCENIC/SCENIC Datasets/", datasets[i], "/"))
  scenicOptions <- readRDS(paste0("int/", datasets[i], ".scenicOptions.RDS"))
  cellInfo <- readRDS(paste0("int/", datasets[i], ".cellInfo.Rds"))
  tSNE_scenic <- readRDS(tsneFileName(scenicOptions))
  aucell_regulonAUC <- loadInt(scenicOptions, "aucell_regulonAUC")
  aucell_regulonAUC <- aucell_regulonAUC[onlyNonDuplicatedExtended(rownames(aucell_regulonAUC)),]
  regulonActivity_byGrouping <- sapply(split(rownames(cellInfo), cellInfo[,dataset.groupings[i]]),
                                      function(cells) rowMeans(getAUC(aucell_regulonAUC)[,cells]))
  rownames(regulonActivity_byGrouping) <- sub(" .*", "", rownames(regulonActivity_byGrouping))
  merged.regulons <- merge(merged.regulons, regulonActivity_byGrouping, by = 0, all = TRUE)
  rownames(merged.regulons) <- merged.regulons$Row.names
  merged.regulons <- merged.regulons[,-1]
}
merged.regulons <- merged.regulons[,-7]

Regulon.grouping.list <- vector(mode = "list", length = 0)
Regulon.grouping.list$Module1 <- rownames(subset(merged.regulons, subset = rowSums(merged.regulons) > 0 & merged.regulons[,2] > merged.regulons[,1] & merged.regulons[,4] > merged.regulons[,3] & merged.regulons[,4] > merged.regulons[,2] & merged.regulons[,4] > merged.regulons[,5] & merged.regulons[,4] > merged.regulons[,7] & merged.regulons[,4] - merged.regulons[,5] > .05))
Regulon.grouping.list$Module2 <- rownames(subset(merged.regulons, subset = rowSums(merged.regulons[,1:5]) > 0 & is.na(merged.regulons[,6]) & merged.regulons[,2] > merged.regulons[,1] & merged.regulons[,4] > merged.regulons[,3] & merged.regulons[,4] > merged.regulons[,2] & merged.regulons[,4] > merged.regulons[,5] & merged.regulons[,4] - merged.regulons[,5] > .05))
Regulon.grouping.list$Module3 <- rownames(subset(merged.regulons, subset = rowSums(merged.regulons[,3:5]) > 0 & is.na(merged.regulons[,1]) & is.na(merged.regulons[,6]) & merged.regulons[,4] > merged.regulons[,3] & merged.regulons[,4] > merged.regulons[,5] & merged.regulons[,4] - merged.regulons[,5] > .05))
Regulon.grouping.list$Module4 <- rownames(subset(merged.regulons, subset = rowSums(merged.regulons[,3:7]) > 0 & is.na(merged.regulons[,1]) & merged.regulons[,4] > merged.regulons[,3] & merged.regulons[,4] > merged.regulons[,5] & merged.regulons[,5] > merged.regulons[,7] & merged.regulons[,4] - merged.regulons[,5] > .05))
Regulon.grouping.list$Module5 <- rownames(subset(merged.regulons, subset = rowSums(merged.regulons) > 0 & merged.regulons[,5] > merged.regulons[,4] & merged.regulons[,7] > merged.regulons[,4] & merged.regulons[,7] - merged.regulons[,4] > .05))
Regulon.grouping.list$Module6 <- rownames(subset(merged.regulons, subset = rowSums(merged.regulons[,3:7]) > 0 & is.na(merged.regulons[,1]) & merged.regulons[,5] > merged.regulons[,4] & merged.regulons[,5] > merged.regulons[,3] & merged.regulons[,7] > merged.regulons[,4] & merged.regulons[,7] - merged.regulons[,4] > .05))
Regulon.grouping.list$Module7 <- rownames(subset(merged.regulons, subset = rowSums(merged.regulons[,6:7]) > 0 & is.na(merged.regulons[,1]) & is.na(merged.regulons[,3]) & merged.regulons[,7] > merged.regulons[,6] & merged.regulons[,7] - merged.regulons[,6] > .05))
Regulon.grouping.list$Module8 <- rownames(subset(merged.regulons, subset = rowSums(merged.regulons[,c(1:2,6:7)]) > 0 & is.na(merged.regulons[,3]) & merged.regulons[,7] > merged.regulons[,6] & merged.regulons[,7] - merged.regulons[,6] > .05 & merged.regulons[,1] - merged.regulons[,2] > -.05))

merged.regulons[is.na(merged.regulons)] <- 0
merged.regulons.NCC.Mel.Paths <- merged.regulons[,c(1,2,4,1,3,4,1,3,5,7,1,3,6,7)]
merged.regulons_grouped <- merged.regulons.NCC.Mel.Paths[match(unlist(Regulon.grouping.list), rownames(merged.regulons.NCC.Mel.Paths)),]
rownames(merged.regulons_grouped) <- gsub("_extended", "*", rownames(merged.regulons_grouped))
colnames(merged.regulons_grouped) <- c("1S2 NCC1", "1S2 NCC2", "1S3 NCC2", "2S2 NCC1", "2S3 NCC1", "2S3 NCC2", "3S2 NCC1", "3S3 NCC1", "3S3 NCC3", "3S4 NCC3", "4S2 NCC1", "4S3 NCC1", "4S4 NCC1", "4S4 NCC3")
Heatmap(merged.regulons_grouped, name="Regulon activity", cluster_rows = F, cluster_columns = F, col = colorRamp2(c(0,.81), c("white", "red")))
ht <- Heatmap(merged.regulons_grouped, name="Regulon activity", cluster_rows = F, cluster_columns = F, col = colorRamp2(c(0,.81), c("white", "red")))
pdf("/Users/fattahilab/Desktop/Manuscript Figures/TF.activity.modules.heatmap.pdf", width=6, height=16)
draw(ht)
dev.off()
```

```{r Figure 2D TF Module Line Plots}
merged.regulons.NCC.Mel.Paths <- t(merged.regulons.NCC.Mel.Paths)
for(i in 1:length(Regulon.grouping.list)){
  merged.regulons.NCC.Mels.sub <- merged.regulons.NCC.Mel.Paths[,colnames(merged.regulons.NCC.Mel.Paths) %in% Regulon.grouping.list[[i]]]
  merged.regulons.NCC.Mels.temp <- as.data.frame(merged.regulons.NCC.Mels.sub)
  merged.regulons.NCC.Mels.temp$Average <- rowMeans(merged.regulons.NCC.Mels.temp)
  merged.regulons.NCC.Mels.temp$Step <- c("D15.NCC","D15.Mel","D22.Mel","D15.NCC","D22.NCC","D22.Mel","D15.NCC","D22.NCC","D22.Mel","D30.Mel","D15.NCC","D22.NCC","D30.NCC","D30.Mel")
  merged.regulons.NCC.Mels.temp$Path <- c("T1P1","T1P1","T1P1","T1P2","T1P2","T1P2","T2P1","T2P1","T2P1","T2P1","T2P2","T2P2","T2P2","T2P2")
  merged.regulons.NCC.Mels.temp$Ident <- c("S2 NCC1", "S2 NCC2", "S3 NCC2", "S2 NCC1", "S3 NCC1", "S3 NCC2", "S2 NCC1", "S3 NCC1", "S3 NCC3", "S4 NCC3", "S2 NCC1", "S3 NCC1", "S4 NCC1", "S4 NCC3")
  ggplot(merged.regulons.NCC.Mels.temp, aes(x=factor(Step, levels=c("D15.NCC", "D15.Mel", "D22.NCC", "D22.Mel", "D30.NCC", "D30.Mel")), y=Average, group=Path)) +
    geom_line(aes(color=Path), size = .5)+ geom_point(aes(color=Ident), size = 1) + scale_color_manual(values=c(indir_D15_subtype_cols, indir_D22_subtype_cols, indir_D30_subtype_cols[c(1,3)], paths)) + labs(x = "Step")+ylim(0,.82)+theme_classic()+NoLegend()
  ggsave(paste0(outputDir, "/TF.Average.Module",i,".pdf"), width = 2, height = 2, dpi = 320, limitsize = FALSE)
}

merged.regulons.NCC.Mel.Paths <- as.data.frame(merged.regulons.NCC.Mel.Paths)
merged.regulons.NCC.Mel.Paths$Step <- c("D15.NCC","D15.Mel","D22.Mel","D15.NCC","D22.NCC","D22.Mel","D15.NCC","D22.NCC","D22.Mel","D30.Mel","D15.NCC","D22.NCC","D30.NCC","D30.Mel")
merged.regulons.NCC.Mel.Paths$Path <- c("T1P1","T1P1","T1P1","T1P2","T1P2","T1P2","T2P1","T2P1","T2P1","T2P1","T2P2","T2P2","T2P2","T2P2")
merged.regulons.NCC.Mel.Paths$Ident <- c("S2 NCC1", "S2 NCC2", "S3 NCC2", "S2 NCC1", "S3 NCC1", "S3 NCC2", "S2 NCC1", "S3 NCC1", "S3 NCC3", "S4 NCC3", "S2 NCC1", "S3 NCC1", "S4 NCC1", "S4 NCC3")
merged.regulons.NCC.Mel.Paths.melted <- melt(merged.regulons.NCC.Mel.Paths)

for(x in 1:length(Regulon.grouping.list)){
  merged.regulons.NCC.Mel.Paths.melted.sub <- merged.regulons.NCC.Mel.Paths.melted[which(merged.regulons.NCC.Mel.Paths.melted$variable %in% Regulon.grouping.list[[x]]),]
  merged.regulons.NCC.Mel.Paths.melted.sub$paired <- NA
  for(i in 1:length(unique(merged.regulons.NCC.Mel.Paths.melted.sub$variable))){
    merged.regulons.NCC.Mel.Paths.melted.sub$paired[which(merged.regulons.NCC.Mel.Paths.melted.sub$variable == unique(merged.regulons.NCC.Mel.Paths.melted.sub$variable)[i] & merged.regulons.NCC.Mel.Paths.melted.sub$Path == "T1P1")] <-  
      paste0(unique(merged.regulons.NCC.Mel.Paths.melted.sub$variable)[i], "T1P1")
    merged.regulons.NCC.Mel.Paths.melted.sub$paired[which(merged.regulons.NCC.Mel.Paths.melted.sub$variable == unique(merged.regulons.NCC.Mel.Paths.melted.sub$variable)[i] & merged.regulons.NCC.Mel.Paths.melted.sub$Path == "T1P2")] <-
      paste0(unique(merged.regulons.NCC.Mel.Paths.melted.sub$variable)[i], "T1P2")
    merged.regulons.NCC.Mel.Paths.melted.sub$paired[which(merged.regulons.NCC.Mel.Paths.melted.sub$variable == unique(merged.regulons.NCC.Mel.Paths.melted.sub$variable)[i] & merged.regulons.NCC.Mel.Paths.melted.sub$Path == "T2P1")] <-
      paste0(unique(merged.regulons.NCC.Mel.Paths.melted.sub$variable)[i], "T2P1")
    merged.regulons.NCC.Mel.Paths.melted.sub$paired[which(merged.regulons.NCC.Mel.Paths.melted.sub$variable == unique(merged.regulons.NCC.Mel.Paths.melted.sub$variable)[i] & merged.regulons.NCC.Mel.Paths.melted.sub$Path == "T2P2")] <-
      paste0(unique(merged.regulons.NCC.Mel.Paths.melted.sub$variable)[i], "T2P2")
}
  ggplot(merged.regulons.NCC.Mel.Paths.melted.sub, aes(x=factor(Step, levels=c("D15.NCC", "D15.Mel", "D22.NCC", "D22.Mel", "D30.NCC", "D30.Mel")), y=value)) +
  geom_line(aes(group=paired, color = Path),size = .25, alpha = .2)+ labs(x = "Step")+ylim(0,.82)+theme_classic()+ scale_color_manual(values=paths)+NoLegend()
  ggsave(paste0(outputDir, "/TF.Individal.Module",x,".pdf"), width = 2, height = 2, dpi = 320, limitsize = FALSE)
}
```

```{r Figure 2G and S2D TF Activity Module Idenitfication and Heatmap}
indir_D15_cr_cellchat <- readRDS("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/indir_D15_cr_cellchat.RDS")
indir_D22_cr_cellchat <- readRDS("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/indir_D22_cr_cellchat.RDS")
indir_D30_cr_cellchat <- readRDS("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/indir_D30_cr_cellchat.RDS")

#Subset each subtypes incoming signaling
indir_D15_cr_cellchat_df.net <- subsetCommunication(indir_D15_cr_cellchat)
S2NCC1.incoming.signaling.by.LR <- indir_D15_cr_cellchat_df.net[indir_D15_cr_cellchat_df.net$target == "S2 NCC1",c(5,8)]
S2NCC2.incoming.signaling.by.LR <- indir_D15_cr_cellchat_df.net[indir_D15_cr_cellchat_df.net$target == "S2 NCC2",c(5,8)]

indir_D22_cr_cellchat_df.net <- subsetCommunication(indir_D22_cr_cellchat)
S3NCC1.incoming.signaling.by.LR <- indir_D22_cr_cellchat_df.net[indir_D22_cr_cellchat_df.net$target == "S3 NCC1",c(5,8)]
S3NCC2.incoming.signaling.by.LR <- indir_D22_cr_cellchat_df.net[indir_D22_cr_cellchat_df.net$target == "S3 NCC2",c(5,8)]
S3NCC3.incoming.signaling.by.LR <- indir_D22_cr_cellchat_df.net[indir_D22_cr_cellchat_df.net$target == "S3 NCC3",c(5,8)]

indir_D30_cr_cellchat_df.net <- subsetCommunication(indir_D30_cr_cellchat)
S4NCC1.incoming.signaling.by.LR <- indir_D30_cr_cellchat_df.net[indir_D30_cr_cellchat_df.net$target == "S4 NCC1",c(5,8)]
S4NCC2.incoming.signaling.by.LR <- indir_D30_cr_cellchat_df.net[indir_D30_cr_cellchat_df.net$target == "S4 NCC2",c(5,8)]
S4NCC3.incoming.signaling.by.LR <- indir_D30_cr_cellchat_df.net[indir_D30_cr_cellchat_df.net$target == "S4 NCC3",c(5,8)]

#for LRPs where the source was possible from two different cell types, average them
signaling.by.LR.dfs <- c("S2NCC1.incoming.signaling.by.LR", "S2NCC2.incoming.signaling.by.LR", "S3NCC1.incoming.signaling.by.LR", "S3NCC2.incoming.signaling.by.LR", "S3NCC3.incoming.signaling.by.LR", "S4NCC1.incoming.signaling.by.LR", "S4NCC2.incoming.signaling.by.LR", "S4NCC3.incoming.signaling.by.LR")
for(i in 1:length(signaling.by.LR.dfs)){
  df <- get(signaling.by.LR.dfs[i])
  interactions <- unique(df$interaction_name_2)
   temp <- data.frame(matrix(ncol = 1, nrow = length(interactions)))
   colnames(temp) <- gsub(".incoming.signaling.by.LR", "", signaling.by.LR.dfs[i])
   rownames(temp) <- interactions
   for(z in 1:length(interactions)){
    temp[z,1] <- sum(subset(df, subset =  interaction_name_2 == interactions[z])[,1])
   }
   assign(paste(signaling.by.LR.dfs[i]), temp)
}

#merge all together in one dataframe
signaling.by.LR <- merge(S2NCC1.incoming.signaling.by.LR, S2NCC2.incoming.signaling.by.LR, by = 0, all = T)
rownames(signaling.by.LR) <- signaling.by.LR$Row.names
signaling.by.LR <- signaling.by.LR[,-1]
for(i in 3:length(signaling.by.LR.dfs)){
  df <- get(signaling.by.LR.dfs[i])
  signaling.by.LR <- merge(signaling.by.LR, df, by = 0, all = T)
  rownames(signaling.by.LR) <- signaling.by.LR$Row.names
  signaling.by.LR <- signaling.by.LR[,-1]
}
signaling.by.LR[is.na(signaling.by.LR)] <- 0
signaling.by.LR <- signaling.by.LR[,-7]
signaling.by.LR <- as.data.frame(signaling.by.LR)
signaling.by.LR.NCC.Mel.Paths <- signaling.by.LR[,c(1,2,4,1,3,4,1,3,5,7,1,3,6,7)]

#transform data for module identification and plotting
signaling.by.LR.NCC.Mel.Paths.log <- log10((signaling.by.LR.NCC.Mel.Paths*1000000))
signaling.by.LR.NCC.Mel.Paths.log[signaling.by.LR.NCC.Mel.Paths.log == "-Inf"] <- 0

signaling.by.LR.log <- log10(signaling.by.LR*1000000)
signaling.by.LR.log[signaling.by.LR.log == "-Inf"] <- 0

LigRecPair.grouping.list <- vector(mode = "list", length = 0)
LigRecPair.grouping.list$Module1 <- rownames(subset(signaling.by.LR.log, subset = signaling.by.LR.log[,1] > 0 & signaling.by.LR.log[,4] > 0 & signaling.by.LR.log[,4] - signaling.by.LR.log[,5] > .7))
LigRecPair.grouping.list$Module2 <- rownames(subset(signaling.by.LR.log, subset = signaling.by.LR.log[,3] > 0 & signaling.by.LR.log[,1] == 0 & signaling.by.LR.log[,4] > 0 & signaling.by.LR.log[,4] - signaling.by.LR.log[,5] > .7))
LigRecPair.grouping.list$Module3 <- rownames(subset(signaling.by.LR.log, subset = signaling.by.LR.log[,1] == 0 & signaling.by.LR.log[,3] == 0 & signaling.by.LR.log[,4] > 0 & signaling.by.LR.log[,4] - signaling.by.LR.log[,5] > .7))
LigRecPair.grouping.list$Module4 <- rownames(subset(signaling.by.LR.log, subset = signaling.by.LR.log[,1] > 0 & signaling.by.LR.log[,5] > 0 & signaling.by.LR.log[,7] > 0 & signaling.by.LR.log[,5] - signaling.by.LR.log[,4] > .7 & signaling.by.LR.log[,7] - signaling.by.LR.log[,4] > .7))
LigRecPair.grouping.list$Module5 <- rownames(subset(signaling.by.LR.log, subset = signaling.by.LR.log[,3] > 0 & signaling.by.LR.log[,1] == 0 & signaling.by.LR.log[,5] > 0 & signaling.by.LR.log[,7] > 0 & signaling.by.LR.log[,5] - signaling.by.LR.log[,4] > .7 & signaling.by.LR.log[,7] - signaling.by.LR.log[,4] > .7))
LigRecPair.grouping.list$Module6 <- rownames(subset(signaling.by.LR.log, subset = signaling.by.LR.log[,1] == 0 & signaling.by.LR.log[,3] == 0 & signaling.by.LR.log[,5] > 0 & signaling.by.LR.log[,7] > 0 & signaling.by.LR.log[,5] - signaling.by.LR.log[,4] > .7 & signaling.by.LR.log[,7] - signaling.by.LR.log[,4] > .7))

signaling.by.LR_grouped <- signaling.by.LR.NCC.Mel.Paths.log[match(unlist(LigRecPair.grouping.list), rownames(signaling.by.LR.NCC.Mel.Paths.log)),]
colnames(signaling.by.LR_grouped) <- c("1S2 NCC1", "1S2 NCC2", "1S3 NCC2", "2S2 NCC1", "2S3 NCC1", "2S3 NCC2", "3S2 NCC1", "3S3 NCC1", "3S3 NCC3", "3S4 NCC3", "4S2 NCC1", "4S3 NCC1", "4S4 NCC3", "4S4 NCC3")
Heatmap(signaling.by.LR_grouped, name="log(Signaling Probability)", cluster_rows = F, cluster_columns = F, col = colorRamp2(c(0, max(signaling.by.LR_grouped)), c("white", "dodgerblue4")))
ht <- Heatmap(signaling.by.LR_grouped, name="log(Signaling Probability)", cluster_rows = F, cluster_columns = F, col = colorRamp2(c(0, max(signaling.by.LR_grouped)), c("white", "dodgerblue4")))
pdf("/Users/fattahilab/Desktop/Manuscript Figures/LRP.activity.modules.heatmap.pdf", width=8, height=10)
draw(ht)
dev.off()
```

```{r Figure F LRP Module Line Plots}
signaling.by.LR.NCC.Mel.Paths.log <- t(signaling.by.LR.NCC.Mel.Paths.log)
for(i in 1:length(LigRecPair.grouping.list)){
  signaling.by.LR.NCC.Mels.sub <- signaling.by.LR.NCC.Mel.Paths.log[,colnames(signaling.by.LR.NCC.Mel.Paths.log) %in% LigRecPair.grouping.list[[i]]]
  signaling.by.LR.NCC.Mels.temp <- as.data.frame(signaling.by.LR.NCC.Mels.sub)
  signaling.by.LR.NCC.Mels.temp$Average <- rowMeans(signaling.by.LR.NCC.Mels.temp)
  signaling.by.LR.NCC.Mels.temp$Step <- c("D15.NCC","D15.Mel","D22.Mel","D15.NCC","D22.NCC","D22.Mel","D15.NCC","D22.NCC","D22.Mel","D30.Mel","D15.NCC","D22.NCC","D30.NCC","D30.Mel")
  signaling.by.LR.NCC.Mels.temp$Path <- c("T1P1","T1P1","T1P1","T1P2","T1P2","T1P2","T2P1","T2P1","T2P1","T2P1","T2P2","T2P2","T2P2","T2P2")
  signaling.by.LR.NCC.Mels.temp$Ident <- c("S2 NCC1", "S2 NCC2", "S3 NCC2", "S2 NCC1", "S3 NCC1", "S3 NCC2", "S2 NCC1", "S3 NCC1", "S3 NCC3", "S4 NCC3", "S2 NCC1", "S3 NCC1", "S4 NCC1", "S4 NCC3")
  ggplot(signaling.by.LR.NCC.Mels.temp, aes(x=factor(Step, levels=c("D15.NCC", "D15.Mel", "D22.NCC", "D22.Mel", "D30.NCC", "D30.Mel")), y=Average, group=Path)) +
    geom_line(aes(color=Path), size = .5)+ geom_point(aes(color=Ident), size = 1) + scale_color_manual(values=c(indir_D15_subtype_cols, indir_D22_subtype_cols, indir_D30_subtype_cols[c(1,3)], paths)) + labs(x = "Step")+ylim(0,5.5)+theme_classic()+NoLegend()
  ggsave(paste0(outputDir, "/LRP.Average.Module",i,".pdf"), width = 2, height = 2, dpi = 320, limitsize = FALSE)
}

signaling.by.LR.NCC.Mel.Paths <- as.data.frame(signaling.by.LR.NCC.Mel.Paths.log)
signaling.by.LR.NCC.Mel.Paths$Step <- c("D15.NCC","D15.Mel","D22.Mel","D15.NCC","D22.NCC","D22.Mel","D15.NCC","D22.NCC","D22.Mel","D30.Mel","D15.NCC","D22.NCC","D30.NCC","D30.Mel")
signaling.by.LR.NCC.Mel.Paths$Path <- c("T1P1","T1P1","T1P1","T1P2","T1P2","T1P2","T2P1","T2P1","T2P1","T2P1","T2P2","T2P2","T2P2","T2P2")
signaling.by.LR.NCC.Mel.Paths$Ident <- c("S2 NCC1", "S2 NCC2", "S3 NCC2", "S2 NCC1", "S3 NCC1", "S3 NCC2", "S2 NCC1", "S3 NCC1", "S3 NCC3", "S4 NCC3", "S2 NCC1", "S3 NCC1", "S4 NCC1", "S4 NCC3")
signaling.by.LR.NCC.Mel.Paths.melted <- melt(signaling.by.LR.NCC.Mel.Paths)

for(x in 1:length(LigRecPair.grouping.list)){
  signaling.by.LR.NCC.Mel.Paths.melted.sub <- signaling.by.LR.NCC.Mel.Paths.melted[which(signaling.by.LR.NCC.Mel.Paths.melted$variable %in% LigRecPair.grouping.list[[x]]),]
  signaling.by.LR.NCC.Mel.Paths.melted.sub$paired <- NA
  for(i in 1:length(unique(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable))){
    signaling.by.LR.NCC.Mel.Paths.melted.sub$paired[which(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable == unique(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable)[i] & signaling.by.LR.NCC.Mel.Paths.melted.sub$Path == "T1P1")] <-  
      paste0(unique(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable)[i], "T1P1")
    signaling.by.LR.NCC.Mel.Paths.melted.sub$paired[which(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable == unique(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable)[i] & signaling.by.LR.NCC.Mel.Paths.melted.sub$Path == "T1P2")] <-
      paste0(unique(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable)[i], "T1P2")
    signaling.by.LR.NCC.Mel.Paths.melted.sub$paired[which(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable == unique(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable)[i] & signaling.by.LR.NCC.Mel.Paths.melted.sub$Path == "T2P1")] <-
      paste0(unique(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable)[i], "T2P1")
    signaling.by.LR.NCC.Mel.Paths.melted.sub$paired[which(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable == unique(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable)[i] & signaling.by.LR.NCC.Mel.Paths.melted.sub$Path == "T2P2")] <-
      paste0(unique(signaling.by.LR.NCC.Mel.Paths.melted.sub$variable)[i], "T2P2")
}
  ggplot(signaling.by.LR.NCC.Mel.Paths.melted.sub, aes(x=factor(Step, levels=c("D15.NCC", "D15.Mel", "D22.NCC", "D22.Mel", "D30.NCC", "D30.Mel")), y=value)) +
  geom_line(aes(group=paired, color = Path),size = .25, alpha = .2)+ labs(x = "Step")+ylim(0,5.5)+theme_classic()+ scale_color_manual(values=paths)+NoLegend()
  ggsave(paste0(outputDir, "/LRP.Individal.Module",x,".pdf"), width = 2, height = 2, dpi = 320, limitsize = FALSE)
}
```

```{r Figure 2H}
DefaultAssay(indir_D15_cr) <- "RNA"
DefaultAssay(indir_D22_cr) <- "RNA"
DefaultAssay(indir_D30_cr) <- "RNA"
NCC.progression <- merge(indir_D15_cr, c(indir_D22_cr, indir_D30_cr))
NCC.progression <- NormalizeData(NCC.progression)
NCC.progression@meta.data$Subtype.grouped <- factor(x = Idents(NCC.progression), levels = c("S2 NCC1", "S3 NCC1", "S4 NCC1", "S3 NCC3", "S4 NCC3", "S2 NCC2", "S3 NCC2", "S4 NCC2"))
DotPlot(NCC.progression, features = c("BHLHE40", "HES7", "ZNF467","EPHA5", "EPHB2", "BMPR1A", "ACVR1B", "RELB", "EPHB3", "EPHB4", "BMPR1B", "ACVR2B", "PLXNA1", "PLXNA2", "CD44"), assay = "alra", group.by = "Subtype.grouped", scale = T, idents = c("S2 NCC2", "S3 NCC2", "S3 NCC3", "S4 NCC3")) + theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust = 1, size = 18), axis.text.y = element_text(size = 18), axis.title.y = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 12), title = element_text(size = 12), legend.position = "top")+ scale_color_gradientn(colours = rev(brewer.pal(10, "Spectral")))
ggsave(paste0(outputDir, "/NCC_trajectory_specific_genes.pdf"), width = 5, height = 3, dpi = 320)
```

```{r Figure 2J}
swne.projection(indir_D22_cr.shared, dir_D15_cr.shared, genes.embed = c("MITF", "SOX10"), reference.group.by="Subtype", query.group.by="Subtype", reference.colors = indir_D22_subtype_cols, query.colors = dir_D15_subtype_cols)
```

```{r Figure S2A}
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures"
indir_D15_cr.cds <- readRDS(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/indir_D15_cr_monocle3.RDS")
indir_D22_cr.cds <- readRDS(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/indir_D22_cr_monocle3.RDS")
indir_D30_cr.cds <- readRDS(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/indir_D30_cr_monocle3.RDS")

datasets <- c("indir_D15_cr.cds", "indir_D22_cr.cds", "indir_D30_cr.cds")
for(i in 1:length(datasets)){
  cds <- get(datasets[i])
  plot_cells(cds, color_cells_by = "pseudotime",cell_size = 2, label_branch_points = F, label_roots = T, label_cell_groups = F, label_leaves = T, trajectory_graph_color = "black", trajectory_graph_segment_size = 1.5)+theme_void()+NoLegend()
  ggsave(paste0(outputDir, "/", datasets[i], "_pseudotime_monolce3.pdf"), width = 4, height = 4, dpi = 320)
}
```

```{r Figure S2B}
indir_D15_cr <- cluster.signature.scoring(query = indir_D15_cr, reference = indir_D22_cr, query.assay = "alra", reference.grouping = "Subtype", query.grouping = "Subtype")
indir_D22_cr <- cluster.signature.scoring(query = indir_D22_cr, reference = indir_D22_cr, query.assay = "alra", reference.grouping = "Subtype", query.grouping = "Subtype")
indir_D30_cr <- cluster.signature.scoring(query = indir_D30_cr, reference = indir_D22_cr, query.assay = "alra", reference.grouping = "Subtype", query.grouping = "Subtype")

FeaturePlot(indir_D22_cr, features = c("S3_NCC2", "S3_NCC3"), blend = TRUE, blend.threshold = 1, order = TRUE, cols = c("grey90", indir_D22_subtype_cols[2], indir_D22_subtype_cols[3]), pt.size = 3)
ggsave(paste0(outputDir, "/T2D22.S3NCC3_4.score.blend.jpeg"), width = 25, height = 5, dpi = 320)
FeaturePlot(indir_D15_cr, features = c("S3_NCC2", "S3_NCC3"), blend = TRUE, blend.threshold = 1, order = TRUE, cols = c("grey90", indir_D22_subtype_cols[2], indir_D22_subtype_cols[3]), pt.size =3)
ggsave(paste0(outputDir, "/T2D15.S3NCC3_4.score.blend.jpeg"), width = 25, height = 5, dpi = 320)
FeaturePlot(indir_D30_cr, features = c("S3_NCC2", "S3_NCC3"), blend = TRUE, blend.threshold = 1, order = TRUE, cols = c("grey90", indir_D22_subtype_cols[2], indir_D22_subtype_cols[3]), pt.size = 3)
ggsave(paste0(outputDir, "/T2D30.S3NCC3_4.score.blend.jpeg"), width = 25, height = 5, dpi = 320)
```

```{r Figure S2F}
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures/"

DimPlot(dir_D15_all, reduction = "umap", group.by = "Cell.Type", pt.size = 2, cols = dir_D15_celltype_cols) + theme_void() +theme(legend.position = "none")+ggtitle("")
ggsave(paste0(outputDir, "/UMAP.dir_D15_all.Cell.Type.jpeg"), width = 4, height = 4, dpi = 320)

DotPlot(dir_D15_all, features = "SOX10", assay = "alra", scale = F, scale.min = 0, scale.max = 100) + theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust = 1, size = 18), axis.text.y = element_text(size = 18), axis.title.y = element_blank(),axis.title.x = element_blank(), legend.text = element_text(size = 18), title = element_text(size = 18))+ scale_color_gradientn(colours = rev(brewer.pal(10, "Spectral")))
ggsave(paste0(outputDir, "/DotPlot.dir_D15_all.SOX10.pdf"), width = 6, height = 4, dpi = 320)

ggplot(dir_D15_all@meta.data, aes(x = sample, fill = factor(dir_D15_all$Cell.Type, levels=rev(levels(dir_D15_all$Cell.Type))))) + geom_bar(position = "fill") +theme_minimal()  + theme(axis.text.x = element_blank(), axis.title.x = element_text(size = 15),legend.text = element_text(size = 15), legend.title = element_text(size = 15), axis.text.y.left = element_text(size = 15, colour = "black"), axis.title.y = element_text(size = 15)) + ylab("Proportion") + scale_fill_manual(values = rev(dir_D15_celltype_cols)) +xlab("") +NoLegend()
ggsave(paste0(outputDir, "/BarGraph.dir_D15_all.cell.type.proportions" ,".pdf"), width = 2, height = 4, dpi = 320, limitsize = FALSE)
```

```{r Figure S2G}
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures/"
DimPlot(dir_D15_cr, reduction = "umap", group.by = "Subtype", pt.size = 2, cols = dir_D15_subtype_cols) + theme_void() +theme(legend.position = "none")+ggtitle("")
ggsave(paste0(outputDir, "/UMAP.dir_D15_cr.Subtype.jpeg"), width = 4, height = 4, dpi = 320)
```

```{r Figure S2H}
datasets <- c("indir_D15_cr", "dir_D15_cr")
color.palettes <- c("indir_D15_subtype_cols", "dir_D15_subtype_cols")
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures/"

#MITF Expression
for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  DotPlot(seur_obj, features = "MITF", assay = "alra", scale = F, scale.min = 0, scale.max = 100) + theme(axis.text.x = element_text(angle = 90, vjust = .5, hjust = 1, size = 18), axis.text.y = element_text(size = 18), axis.title.y = element_blank(), axis.title.x = element_blank(), legend.text = element_text(size = 18), title = element_text(size = 18))+ scale_color_gradientn(colours = rev(brewer.pal(10, "Spectral")))
  ggsave(paste0(outputDir, "/DotPlot.", datasets[i], ".SOX10.pdf"), width = 6, height = 4, dpi = 320)
}

#CellType Proportion Plots
for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  cols <- get(color.palettes[i])
  ggplot(seur_obj@meta.data, aes(x = sample, fill = factor(seur_obj$Subtype, levels=rev(levels(seur_obj$Subtype))))) + geom_bar(position = "fill") +theme_minimal()  + theme(axis.text.x = element_blank(), axis.title.x = element_text(size = 15),legend.text = element_text(size = 15), legend.title = element_text(size = 15), axis.text.y.left = element_text(size = 15, colour = "black"), axis.title.y = element_text(size = 15)) + ylab("Proportion") + scale_fill_manual(values = rev(cols)) +xlab("") +NoLegend()
  ggsave(paste0(outputDir, "/BarGraph.",datasets[i], ".cell.type.proportions" ,".pdf"), width = 2, height = 4, dpi = 320, limitsize = FALSE)
}
```

###Figure 3
```{r Figure 3E}
fsc.files <- list.files(path = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/FLOW Data/Mature Melanocytes MITF-SSC-CD44-T1 SM test 06012022/FCS files")
object.names <- sub('.fcs', '', fsc.files)
object.names <- sub(' ', '_', object.names)
for(i in 1:length(fsc.files)){
  temp <- read.FCS(filename = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/FLOW Data/Mature Melanocytes MITF-SSC-CD44-T1 SM test 06012022/FCS files/", fsc.files[i]))
  assign(object.names[i], temp)
}
SSC.list <- vector(mode = "list", length = 25)
names(SSC.list) <- object.names
for(i in 1:length(fsc.files)){
  temp <- get(object.names[i])
  raw.data <- as.data.frame(temp@exprs)
  SSC.list[[i]] <- raw.data$`SSC-A`
}

SSC.df <- plyr::ldply(SSC.list, rbind)
SSC.df <- t(SSC.df)
colnames(SSC.df) <- SSC.df[1,]
SSC.df <- SSC.df[-1,]

cells.dfs <- sub('$', '.cells.df', object.names)
for(i in 1:length(object.names)){
  temp <- get(object.names[i])
  temp.df <- subset(as.data.frame(temp@exprs), subset = `FSC-A` > 190000 & `SSC-A` > 50000)
  assign(cells.dfs[i], temp.df)
}

SSC.list <- vector(mode = "list", length = 25)
names(SSC.list) <- object.names
for(i in 1:length(cells.dfs)){
  temp <- get(cells.dfs[i])
  SSC.list[[i]] <- temp$`SSC-A`
}

SSC.df <- plyr::ldply(SSC.list, rbind)
SSC.df <- t(SSC.df)
colnames(SSC.df) <- SSC.df[1,]
SSC.df <- SSC.df[-1,]
SSC.df <- SSC.df[,-4]

cells.dfs <- cells.dfs[c(1:3, 5:length(cells.dfs))]
SSC.dfs <- sub('cells', 'SSC', cells.dfs)
Type <- c("T1", "T1", "T1", "T1", "T1", "T1", "T1", "T1", "T1", "T1", "T1", "T1", "T2",  "T2",  "T2",  "T2",  "T2",  "T2",  "T2",  "T2",  "T2",  "T2",  "T2",  "T2")
Samples <- c("T1.1", "T1.2", "T1.3", "T1.4", "T1.5", "T1.6", "T1.7", "T1.8", "T1.9", "T1.10", "T1.11", "T1.12", "T2.1", "T2.2", "T2.3", "T2.4", "T2.5", "T2.6", "T2.7", "T2.8", "T2.9", "T2.10", "T2.11", "T2.12")

for(i in 1:length(cells.dfs)){
  temp <- get(cells.dfs[i])
  temp.SSC.df <- as.data.frame(temp$`SSC-A`)
  colnames(temp.SSC.df) <- "SSC"
  temp.SSC.df$Type <- Type[i]
  temp.SSC.df$Samples <- Samples[i]
  assign(SSC.dfs[i], temp.SSC.df)
}

SSC.df.melt <- Experiment_T1_MITF_CD44_ADAM10_Sample.SSC.df
for(i in 2:length(SSC.dfs)){
  temp <- get(SSC.dfs[i])
  SSC.df.melt <- rbind(SSC.df.melt, temp)
}

g <- ggplot(SSC.df.melt, aes(SSC, color = Type, fill = Type))+xlim(0,1048574)+scale_color_manual(values=c("#7DB0DDFF", "#E093C3FF"))+scale_fill_manual(values=c("#7DB0DDFF", "#E093C3FF"))+geom_density(data=subset(SSC.df.melt,Samples == 'T1.1'),color = "#A2D1FF", alpha = 0)

p <- ggplot_build(g)

SSC.density.line.df.counts <- as.data.frame(matrix(nrow = 512, ncol = 25))
SSC.density.line.df.counts[,1] <- p$data[[1]]$x
colnames(SSC.density.line.df.counts) <- c("SSC", "T1.1", "T1.2", "T1.3", "T1.4", "T1.5", "T1.6", "T1.7", "T1.8", "T1.9", "T1.10", "T1.11", "T1.12", "T2.1", "T2.2", "T2.3", "T2.4", "T2.5", "T2.6", "T2.7", "T2.8", "T2.9", "T2.10", "T2.11", "T2.12")
SSC.df <- as.data.frame(SSC.df)
for(i in 1:24){
  SSC.df[,i] <- as.numeric(SSC.df[,i])
}

SSc.bin.values <- SSC.density.line.df.counts$SSC
SSC.density.line.df.counts[1,] <- 0
for(i in 1:24){
  for(x in 2:length(SSc.bin.values)){
    SSC.density.line.df.counts[x,i+1] <- length(subset(SSC.df[,i], subset = SSC.df[,i] > SSc.bin.values[x-1] & SSC.df[,i] < SSc.bin.values[x]))
  }
}

SSC.density.line.df.scaled <- as.data.frame(matrix(nrow = 512, ncol = 25))
SSC.density.line.df.scaled[,1] <- p$data[[1]]$x
colnames(SSC.density.line.df.scaled) <- c("SSC", "T1.1", "T1.2", "T1.3", "T1.4", "T1.5", "T1.6", "T1.7", "T1.8", "T1.9", "T1.10", "T1.11", "T1.12", "T2.1", "T2.2", "T2.3", "T2.4", "T2.5", "T2.6", "T2.7", "T2.8", "T2.9", "T2.10", "T2.11", "T2.12")
for(i in 2:length(SSC.density.line.df.counts)){
  SSC.density.line.df.scaled[,i] <- SSC.density.line.df.counts[,i]/max(SSC.density.line.df.counts[,i])
}

SSC.density.line.df.scaled$T1.average <- rowMeans(SSC.density.line.df.scaled[,c(2:13)])
SSC.density.line.df.scaled$T1.SD <- rowSds(as.matrix(SSC.density.line.df.scaled[,c(2:13)]))
SSC.density.line.df.scaled$T2.average <- rowMeans(SSC.density.line.df.scaled[,c(14:25)])
SSC.density.line.df.scaled$T2.SD <- rowSds(as.matrix(SSC.density.line.df.scaled[,c(14:25)]))

T1.SSC.density.line.df.scaled <- SSC.density.line.df.scaled[,c(1,26)]
T1.SSC.density.line.df.scaled$upper <- SSC.density.line.df.scaled$T1.average + SSC.density.line.df.scaled$T1.SD
T1.SSC.density.line.df.scaled$lower <- SSC.density.line.df.scaled$T1.average - SSC.density.line.df.scaled$T1.SD
T1.SSC.density.line.df.scaled$Type <- "T1"

T2.SSC.density.line.df.scaled <- SSC.density.line.df.scaled[,c(1,28)]
T2.SSC.density.line.df.scaled$upper <- SSC.density.line.df.scaled$T2.average + SSC.density.line.df.scaled$T2.SD
T2.SSC.density.line.df.scaled$lower <- SSC.density.line.df.scaled$T2.average - SSC.density.line.df.scaled$T2.SD
T2.SSC.density.line.df.scaled$Type <- "T2"
colnames(T1.SSC.density.line.df.scaled) <- c("SSC", "count", "upper", "lower", "Type")
colnames(T2.SSC.density.line.df.scaled) <- c("SSC", "count", "upper", "lower", "Type")

SSC.ribbon.plot.scaled.df <- rbind(T1.SSC.density.line.df.scaled, T2.SSC.density.line.df.scaled)

ggplot(SSC.ribbon.plot.scaled.df, aes(SSC, count, color = Type, fill = Type)) +
  geom_line(size = 1) +
  geom_ribbon(aes(ymin = lower, ymax = upper), alpha = .3, linetype = .5)+scale_color_manual(values=c("#7DB0DDFF", "#E093C3FF"))+scale_fill_manual(values=c("#7DB0DDFF", "#E093C3FF"))+ylab("Normalized Counts")+theme_classic()
ggsave(paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/FLOW Data/Mature Melanocytes MITF-SSC-CD44-T1 SM test 06012022", "/SSC.histogram.pdf"), width = 6, height = 4, dpi = 320)

options(contrasts = c("contr.sum","contr.poly"))
model<-lmer(SSC ~ Type + (1|Samples:Type), data = SSC.df.melt, REML = TRUE)
summary(model)
anova(model)
```

```{r Figure 3I}
DimPlot(mature.melanocytes, reduction = "umap", group.by = "Cell.Type", cols = c(dir_D30_celltype_cols, indir_D59_celltype_cols), pt.size = 1) + theme(legend.text = element_text(size = 20), title = element_blank())+theme_void()
ggsave(paste0(outputDir, "/UMAP","mature.melanocytes" ,".jpeg"), width = 7, height = 4, dpi = 320, limitsize = FALSE)
```

```{r Figure 3J-K}
GeneLists <- read.csv(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/Gene Lists.csv", na.strings = "")
LOI <- GeneLists[,c(1,9:11)]
source("/Users/fattahilab/Documents/Single Cell Database Repository/SynologyDrive/Useful R Things/Functions/Ryans_Functions.R")
melanocyte.Spec.genes <- filter.gene.df.for.markers(mature.melanocytes, LOI, group.by = "Cell.Type", assay = "alra")

VlnPlot(mature.melanocytes, melanocyte.Spec.genes$TFs.Full.List, stack = T, flip = T, fill.by = "ident", cols = c("#7DB0DDFF", "#E093C3FF")) + NoLegend() + ggtitle("Transcription Factors")+theme(axis.text.y = element_text(size = 9),strip.text  = element_text(size = 20, face = "plain", hjust = 0), axis.text.x = element_text(size = 15, angle = 0, hjust = .5), axis.title.x = element_blank())
ggsave(paste0(outputDir, "/VlnStack.","mature.melanocytes.TFs" ,".pdf"), width = 3, height = 10, dpi = 320, limitsize = FALSE)
VlnPlot(mature.melanocytes, melanocyte.Spec.genes$Secreted.Ligands.Full.List, stack = T, flip = T, fill.by = "ident", cols = c("#7DB0DDFF", "#E093C3FF")) + NoLegend() + ggtitle("Ligands")+theme(axis.text.y = element_text(size = 9),strip.text  = element_text(size = 20, face = "plain", hjust = 0), axis.text.x = element_text(size = 15, angle = 0, hjust = .5), axis.title.x = element_blank())
ggsave(paste0(outputDir, "/VlnStack.","mature.melanocytes.secreted" ,".pdf"), width = 3, height = 3, dpi = 320, limitsize = FALSE)
VlnPlot(mature.melanocytes, melanocyte.Spec.genes$Receptors.Full.List, stack = T, flip = T, fill.by = "ident", cols = c("#7DB0DDFF", "#E093C3FF")) + NoLegend() + ggtitle("Receptors")+theme(axis.text.y = element_text(size = 9),strip.text  = element_text(size = 20, face = "plain", hjust = 0), axis.text.x = element_text(size = 15, angle = 0, hjust = .5), axis.title.x = element_blank())
ggsave(paste0(outputDir, "/VlnStack.","mature.melanocytes.receptors" ,".pdf"), width = 3, height = 5.5, dpi = 320, limitsize = FALSE)
VlnPlot(mature.melanocytes, c(melanocyte.Spec.genes$Surface.markers.Full.List, "CD44"), stack = T, flip = T, fill.by = "ident", cols = c("#7DB0DDFF", "#E093C3FF")) + NoLegend() + ggtitle("Surface Markers")+theme(axis.text.y = element_text(size = 9),strip.text  = element_text(size = 20, face = "plain", hjust = 0), axis.text.x = element_text(size = 15, angle = 0, hjust = .5), axis.title.x = element_blank())
ggsave(paste0(outputDir, "/VlnStack.","mature.melanocytes.SMs" ,".pdf"), width = 3, height = 5.5, dpi = 320, limitsize = FALSE)
```

```{r Figure S3D}
datasets <- c("indir_D59_all", "dir_D30_all")
color.palettes <- c("indir_D59_celltype_cols", "dir_D30_celltype_cols")
for(i in 1:length(datasets)){
  seur_obj <- get(datasets[i])
  cols <- get(color.palettes[i])
  DimPlot(seur_obj, reduction = "umap", group.by = "Cell.Type", pt.size = 2, cols = cols) + theme_void() +theme(legend.position = "none")+ggtitle("")
  ggsave(paste0(outputDir, "/UMAP.", datasets[i], ".Cell.Type.jpeg"), width = 4, height = 4, dpi = 320)
}
```

```{r Figure S3E}
Idents(mature.melanocytes) <- "Cell.Type"
DefaultAssay(mature.melanocytes) <- "RNA"
mature.melanocytes.avg.expr <- as.data.frame(log1p(AverageExpression(mature.melanocytes, verbose = FALSE)$RNA))
colnames(mature.melanocytes.avg.expr) <- c("T1.Melanocyte", "T2.Melanocyte")
mat.mel.DE <- FindAllMarkers(mature.melanocytes, only.pos = TRUE)
T1.enriched <- mat.mel.DE$gene[mat.mel.DE$cluster == "T1 Melanocyte"]
T2.enriched <- mat.mel.DE$gene[mat.mel.DE$cluster == "T2 Melanocyte"]
mature.melanocytes.avg.expr$Sig.DE <- NA
mature.melanocytes.avg.expr$Sig.DE[which(rownames(mature.melanocytes.avg.expr) %in% T1.enriched)] <- "T1 Enriched"
mature.melanocytes.avg.expr$Sig.DE[which(rownames(mature.melanocytes.avg.expr) %in% T2.enriched)] <- "T2 Enriched"
mature.melanocytes.avg.expr$Sig.DE[is.na(mature.melanocytes.avg.expr$Sig.DE)] <- "Neither"
ggplot(mature.melanocytes.avg.expr, aes(x= T1.Melanocyte, y = T2.Melanocyte)) + geom_point(aes(size = Sig.DE, shape = Sig.DE, colour = factor(Sig.DE, levels = c("T1 Enriched", "T2 Enriched", "Neither"))))+ scale_shape_manual(values=c(1, 19, 19))+scale_size_manual(values=c(1,2,2)) + theme_minimal() + labs(x = "T1 Melanocyte Avg. Expression", y = "T2 Melanocyte Avg. Expression") +theme(legend.position = "none") + scale_colour_manual(values=c(indir_D59_celltype_cols, dir_D30_celltype_cols, "#000000"))+theme_cowplot()+NoLegend()
ggsave(paste0(outputDir, "/ScatterPLot","mature.melanocytes.DE.Genes" ,".jpeg"), width = 4, height = 4, dpi = 320, limitsize = FALSE)
```

###Figure 4
###Curation of Published Datasets
```{r Belote et al}
# object creation and metadata formatting 
Belote.counts <- read.csv("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Belote/GSE151091_raw_matrix.csv.gz")
rownames(Belote.counts) <- Belote.counts$X
Belote.counts <- Belote.counts[, 2:length(Belote.counts)]
Belote.all <- CreateSeuratObject(Belote.counts, min.cells = 3, min.features = 500)
samples <- colnames(Belote.all)
samples <- substr(samples,1,nchar(samples)-11)
samples <- gsub('_', "", samples)
Belote.all$Samples <- samples

Idents(Belote.all) <- "Samples"
cluster_levels <- levels(Belote.all)
new.cluster.ids <- c("Adult", "Fetal", "Neo","Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Neo", "Adult", "Adult", "Adult", "Adult", "Fetal", "Fetal", "Fetal", "Fetal", "Fetal", "Adult")
Idents(Belote.all) <- plyr::mapvalues(x = Idents(Belote.all), from = cluster_levels, to = new.cluster.ids)
Belote.all$Stage <- Idents(Belote.all)

Idents(Belote.all) <- "Samples"
cluster_levels <- levels(Belote.all)
new.cluster.ids <- c("Adult", "12wk", "Neo","Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Neo", "Adult", "Adult", "Adult", "Adult", "18wk", "12wk", "10wk", "16wk", "9.5wk", "Adult")
Idents(Belote.all) <- plyr::mapvalues(x = Idents(Belote.all), from = cluster_levels, to = new.cluster.ids)
Belote.all$Age <- Idents(Belote.all)

Idents(Belote.all) <- "Samples"
cluster_levels <- levels(Belote.all)
new.cluster.ids <- c("LM", "NA", "LM","L", "LM", "M", "M", "L", "LM", "L", "M", "LM", "LM", "L", "LM", "M", "NA", "NA", "NA", "NA", "NA", "M")
Idents(Belote.all) <- plyr::mapvalues(x = Idents(Belote.all), from = cluster_levels, to = new.cluster.ids)
Belote.all$SkinTone <- Idents(Belote.all)

Idents(Belote.all) <- "Samples"
cluster_levels <- c("A1015LM", "X12WKM01", "FS043LM", "A1011L", "A1020LM", "A1033M", "A1022M", "A1014L", "A1038LM", "A1026L", "A1021M", "FS030LM", "A1016LM", "A1025L", "A1017LM", "A1012M", "X18WKM06", "X12WK05", "X10WK03", "X16WKM04", "X9.5WK02", "A1046M")
new.cluster.ids <- c("A1015LM", "12WKM01", "FS043LM", "A1011L", "A1020LM", "A1033M", "A1022M", "A1014L", "A1038LM", "A1026L", "A1021M", "FS030LM", "A1016LM", "A1025L", "A1017LM", "A1012M", "18WKM06", "12WK05", "10WK03", "16WKM04", "9.5WK02", "A1046M")
Idents(Belote.all) <- plyr::mapvalues(x = Idents(Belote.all), from = cluster_levels, to = new.cluster.ids)
Belote.all$Samples <- Idents(Belote.all)
Idents(Belote.all) <- "Samples"
cluster_levels <- c("9.5WK02", "10WK03", "12WKM01", "12WK05", "16WKM04", "18WKM06", "FS030LM", "FS043LM", "A1021M", "A1038LM", "A1012M", "A1022M", "A1015LM", "A1025L", "A1016LM", "A1020LM", "A1033M", "A1011L", "A1026L", "A1014L", "A1046M", "A1017LM")
Belote.all@meta.data$Samples <- factor(x = Idents(Belote.all), levels = cluster_levels)

Idents(Belote.all) <- "Samples"
cluster_levels <- c("9.5WK02", "10WK03", "12WKM01", "12WK05", "16WKM04", "18WKM06", "FS030LM", "FS043LM", "A1021M", "A1038LM", "A1012M", "A1022M", "A1015LM", "A1025L", "A1016LM", "A1020LM", "A1033M", "A1011L", "A1026L", "A1014L", "A1046M", "A1017LM")
new.cluster.ids <- c("9.5WK02", "10WK03", "12WK01", "12WK05", "16WK04", "18WK06", "FS030", "FS043", "A1021", "A1038", "A1012", "A1022", "A1015", "A1025", "A1016", "A1020", "A1033", "A1011", "A1026", "A1014", "A1046", "A1017")
Idents(Belote.all) <- plyr::mapvalues(x = Idents(Belote.all), from = cluster_levels, to = new.cluster.ids)
Belote.all$Sample <- Idents(Belote.all)

#QC and Clustering
Idents(Belote.all) <- "orig.ident"
mt.genes <- rownames(Belote.all)[grep("^MT-", rownames(Belote.all))]
Belote.all <- PercentageFeatureSet(Belote.all, features = mt.genes, col.name = "percent.mt")
rb.genes <- rownames(Belote.all)[grep("^RP[SL]",rownames(Belote.all))]
Belote.all <- PercentageFeatureSet(Belote.all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(Belote.all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
Belote.all$novelty.score <- novelty$Score
Belote.all$orig.ident <- "SeuratProject"
Idents(Belote.all) <- "orig.ident"
VlnPlot(Belote.all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(Belote.all, features = c("percent.mt", "percent.ribo", "novelty.score"), ncol = 3)
VlnPlot(Belote.all, features = "nFeature_RNA") + geom_hline(yintercept = 500)
VlnPlot(Belote.all, features = "nCount_RNA") + geom_hline(yintercept = 7500)
FeatureScatter(Belote.all, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(Belote.all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
FeatureScatter(Belote.all, feature1 = "nFeature_RNA", feature2 = "percent.mt")
Belote.all <- subset(Belote.all, subset = nFeature_RNA > 500 & nFeature_RNA < 7500 & nCount_RNA > 50000)

Belote.all <- NormalizeData(Belote.all)
Belote.all <- FindVariableFeatures(Belote.all)
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
Belote.all <- CellCycleScoring(Belote.all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Belote.all <- ScaleData(Belote.all)
Belote.all <- RunPCA(Belote.all)
ElbowPlot(Belote.all, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(Belote.all@reductions[["pca"]]@stdev[Belote.all@reductions[["pca"]]@stdev > 2])
Belote.all <- RunUMAP(Belote.all, reduction = "pca", dims = 1:num.pcs)
Belote.all <- FindNeighbors(Belote.all, dims = 1:num.pcs)
Belote.all <- FindClusters(Belote.all, resolution = .1)
DimPlot(Belote.all, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(Belote.all, reduction = "umap", label = TRUE) + theme(legend.position = "none")
DimPlot(Belote.all, reduction = "umap", label = TRUE, group.by = "Cell.Type")
cluster_levels <- c(0,1,2,3,4,5,6,7,8,9,10)
new.cluster.ids <- c("Melanocyte", "Melanocyte", "Keratinocyte", "Melanocyte", "Eccrine", "Keratinocyte", "Immune", "Low Quality", "Eccrine", "Immune", "Keratinocyte")
Idents(Belote.all) <- plyr::mapvalues(x = Idents(Belote.all), from = cluster_levels, to = new.cluster.ids)
Belote.all$Cell.Type <- Idents(Belote.all)

#Subsetting melanocytes
Idents(Belote.all) <- "Cell.Type"
Belote.mel <- subset(Belote.all, idents = "Melanocyte")
Belote.mel <- NormalizeData(Belote.mel)
Belote.mel <- FindVariableFeatures(Belote.mel)
Belote.mel <- ScaleData(Belote.mel)
Belote.mel <- RunPCA(Belote.mel)
ElbowPlot(Belote.mel, ndims = 30) + geom_hline(yintercept = 2)
num.pcs <- length(Belote.mel@reductions[["pca"]]@stdev[Belote.mel@reductions[["pca"]]@stdev > 2])
Belote.mel <- RunUMAP(Belote.mel, reduction = "pca", dims = 1:num.pcs)
Belote.mel <- FindNeighbors(Belote.mel, dims = 1:num.pcs)
Belote.mel <- FindClusters(Belote.mel, resolution = .5)
DimPlot(Belote.mel, reduction = "pca", label = TRUE) + theme(legend.position = "none")
DimPlot(Belote.mel, reduction = "umap", label = TRUE) + theme(legend.position = "none")

save(Belote.all, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Belote/Belote.all.RData")
save(Belote.mel, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Belote/Belote.mel.RData")
```

```{r Lyko et al}
# object creation and metadata formatting 
lyko <- readRDS("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Lyko/raw/GSE130973_seurat_analysis_lyko.rds")
lyko.meta <- lyko@meta.data[,c(4,6:8)]
lyko.counts <- lyko@assays$RNA@counts
Lyko.all <- CreateSeuratObject(lyko.counts, meta.data = lyko.meta, min.cells = 3, min.features = 250)

#QC and Clustering
mt.genes <- rownames(Lyko.all)[grep("^MT-", rownames(Lyko.all))]
Lyko.all <- PercentageFeatureSet(Lyko.all, features = mt.genes, col.name = "percent.mt")
rb.genes <- rownames(Lyko.all)[grep("^RP[SL]",rownames(Lyko.all))]
Lyko.all <- PercentageFeatureSet(Lyko.all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(Lyko.all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
Lyko.all$novelty.score <- novelty$Score
Idents(Lyko.all) <- "orig.ident"
VlnPlot(Lyko.all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(Lyko.all, features = c("percent.mt", "percent.ribo"), ncol = 3)
VlnPlot(Lyko.all, features = "nFeature_RNA") + geom_hline(yintercept = 200)
VlnPlot(Lyko.all, features = "nFeature_RNA") + geom_hline(yintercept = 6500)
FeatureScatter(Lyko.all, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(Lyko.all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
FeatureScatter(Lyko.all, feature1 = "nFeature_RNA", feature2 = "percent.mt")

Lyko.all <- NormalizeData(Lyko.all, verbose = FALSE)
lyko.all <- FindVariableFeatures(lyko.all, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
lyko.all <- CellCycleScoring(lyko.all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
lyko.all <- RunFastMNN(object.list = SplitObject(lyko.all, split.by = "subj"))
lyko.all <- RunUMAP(lyko.all, reduction = "mnn", dims = 1:30)
lyko.all <- FindNeighbors(lyko.all, dims = 1:30, reduction = "mnn")
lyko.all <- FindClusters(lyko.all, resolution = .1)
DimPlot(lyko.all, reduction = "umap", label = TRUE)
DimPlot(lyko.all, reduction = "umap", group.by = "integrated_snn_res.0.4", label = TRUE)
DimPlot(lyko.all, reduction = "umap", group.by = "integrated_snn_res.0.4", pt.size = 1) +theme_void()+theme(legend.text = element_text(size = 20), title = element_blank())

#Subsetting melanocytes
Idents(lLyko.all) <- "integrated_snn_res.0.4"
Lyko.mel <- subset(lyko.all, idents = 14)
Lyko.mel <- NormalizeData(Lyko.mel, verbose = FALSE)
Lyko.mel <- FindVariableFeatures(Lyko.mel, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Lyko.mel <- CellCycleScoring(Lyko.mel, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Lyko.mel <- RunFastMNN(object.list = SplitObject(Lyko.mel, split.by = "subj"))
Lyko.mel <- RunUMAP(Lyko.mel, reduction = "mnn", dims = 1:30)
Lyko.mel <- FindNeighbors(Lyko.mel, dims = 1:30, reduction = "mnn")
Lyko.mel <- FindClusters(Lyko.mel, resolution = .1)
DimPlot(Lyko.mel, reduction = "umap", label = TRUE)
DimPlot(Lyko.mel, reduction = "umap", group.by = "subj", label = FALSE)
DimPlot(Lyko.mel, reduction = "umap", group.by = "age", pt.size = 4) +theme_void()+theme(legend.text = element_text(size = 20), title = element_blank())

save(Lyko.all, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Lyko/Processed/lyko.all.RData")
save(Lyko.mel, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Lyko/Processed/lyko.mel.RData")
```

```{r Haniffa et al Adult}
# object creation and metadata formatting 
Convert("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa/submission.h5ad", dest = "h5seurat", overwrite = TRUE)
Haniffa <- LoadH5Seurat("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa/submission.h5seurat")
Idents(Haniffa) <- "final_clustering"
table(Idents(Haniffa))
Haniffa.meta <- Haniffa@meta.data
Haniffa.counts <- Haniffa@assays$RNA@counts
Haniffa.all <- CreateSeuratObject(Haniffa.counts, meta.data = Haniffa.meta, min.cells = 3, min.features = 250)

#QC and Clustering
mt.genes <- rownames(Haniffa.all)[grep("^MT-", rownames(Haniffa.all))]
Haniffa.all <- PercentageFeatureSet(Haniffa.all, features = mt.genes, col.name = "percent.mt")
rb.genes <- rownames(Haniffa.all)[grep("^RP[SL]",rownames(Haniffa.all))]
Haniffa.all <- PercentageFeatureSet(Haniffa.all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(Haniffa.all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
Haniffa.all$novelty.score <- novelty$Score
Idents(Haniffa.all) <- "orig.ident"
VlnPlot(Haniffa.all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(Haniffa.all, features = c("percent.mt", "percent.ribo"), ncol = 3)
VlnPlot(Haniffa.all, features = "nFeature_RNA") + geom_hline(yintercept = 200)
VlnPlot(Haniffa.all, features = "nFeature_RNA") + geom_hline(yintercept = 6500)
FeatureScatter(Haniffa.all, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(Haniffa.all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
FeatureScatter(Haniffa.all, feature1 = "nFeature_RNA", feature2 = "percent.mt")

Haniffa.all <- NormalizeData(Haniffa.all, verbose = FALSE)
Haniffa.all <- FindVariableFeatures(Haniffa.all, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Haniffa.all <- CellCycleScoring(Haniffa.all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Haniffa.all <- RunFastMNN(object.list = SplitObject(Haniffa.all, split.by = "donor_id"))
Haniffa.all <- RunUMAP(Haniffa.all, reduction = "mnn", dims = 1:30)
Haniffa.all <- FindNeighbors(Haniffa.all, dims = 1:30, reduction = "mnn")
Haniffa.all <- FindClusters(Haniffa.all, resolution = .2)
DimPlot(Haniffa.all, reduction = "umap", label = TRUE)
DimPlot(Haniffa.all, reduction = "umap", group.by = "final_clustering", label = TRUE, repel = TRUE)+NoLegend()
DimPlot(Haniffa.all, reduction = "umap", group.by = "Status", label = FALSE)
DimPlot(Haniffa.all, reduction = "umap", group.by = "Site", label = FALSE)
DimPlot(Haniffa.all, reduction = "umap", group.by = "Tissue", label = FALSE)
DimPlot(Haniffa.all, reduction = "umap", group.by = "Location", label = FALSE)
DimPlot(Haniffa.all, reduction = "umap", group.by = "Sex", label = FALSE)

#Subsetting melanocytes
Idents(Haniffa.all) <- "final_clustering"
Haniffa.mel <- subset(Haniffa.all, idents = "Melanocyte")
Haniffa.mel <- NormalizeData(Haniffa.mel, verbose = FALSE)
Haniffa.mel <- FindVariableFeatures(Haniffa.mel, selection.method = "vst", nfeatures = 2000, verbose = FALSE)
Haniffa.mel <- CellCycleScoring(Haniffa.mel, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Haniffa.mel <- RunFastMNN(object.list = SplitObject(Haniffa.mel, split.by = "donor_id"))
Haniffa.mel <- RunUMAP(Haniffa.mel, reduction = "mnn", dims = 1:30)
Haniffa.mel <- FindNeighbors(Haniffa.mel, dims = 1:30, reduction = "mnn")
Haniffa.mel <- FindClusters(Haniffa.mel, resolution = .2)
DimPlot(Haniffa.mel, reduction = "umap", label = TRUE)

save(Haniffa.all, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa/Haniffa.all.RData")
save(Haniffa.mel, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa/Haniffa.mel.RData")
```

```{r Guttman et al}
# object creation and metadata formatting 
guttman1 <- read.csv(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Guttman/Raw/GSE147424_RAW/GSM4430459_MS.sample1.clean.data.txt.gz", row.names = 1)
files <- list.files(path= "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Guttman/Raw/GSE147424_RAW/")
sample.names <- c("S1_LS", "S2_LS", "S3_NL", "S4_H", "S5_LS", "S6_H", "S7_LS", "S8_H", "S9_H", "S10_H", "S11_NL", "S12_H", "S13_H", "S14_NL", "S15_NL", "S16_NL", "S17_H")
for(i in 1:length(files)){
  temp <- read.csv(file = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Guttman/Raw/GSE147424_RAW/", files[i]), row.names = 1)
  temp <- as.matrix(temp)
  temp <- as(temp, "sparseMatrix")
  temp <- CreateSeuratObject(counts = temp)
  if(grepl("LS", sample.names[i])){
    temp$Status <- "LS"
  }else if(grepl("NL", sample.names[i])){
    temp$Status <- "NL"
  }else if(grepl("H", sample.names[i])){
    temp$Status <- "H"}
  assign(paste(sample.names[i]), temp)
}
Guttman.all <- merge(S1_LS, S2_LS)
for(i in 3:length(sample.names)){
  temp <- get(sample.names[i]) 
  Guttman.all<- merge(Guttman.all, temp)
}
#QC and Clustering
mt.genes <- rownames(Guttman.all)[grep("^MT-", rownames(Guttman.all))]
Guttman.all <- PercentageFeatureSet(Guttman.all, features = mt.genes, col.name = "percent.mt")
rb.genes <- rownames(Guttman.all)[grep("^RP[SL]",rownames(Guttman.all))]
Guttman.all <- PercentageFeatureSet(Guttman.all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(Guttman.all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
Guttman.all$novelty.score <- novelty$Score
Idents(Guttman.all) <- "orig.ident"
VlnPlot(Guttman.all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(Guttman.all, features = c("percent.mt", "percent.ribo"), ncol = 3)
VlnPlot(Guttman.all, features = "nFeature_RNA") + geom_hline(yintercept = 200)
VlnPlot(Guttman.all, features = "nCount_RNA") + geom_hline(yintercept = 6500)
FeatureScatter(Guttman.all, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(Guttman.all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
FeatureScatter(Guttman.all, feature1 = "nFeature_RNA", feature2 = "percent.mt")

Guttman.all$sample <- Idents(Guttman.all)
Guttman.all <- NormalizeData(Guttman.all, verbose = FALSE)
Guttman.all <- FindVariableFeatures(Guttman.all, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Guttman.all <- CellCycleScoring(Guttman.all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Guttman.all <- RunFastMNN(object.list = SplitObject(Guttman.all, split.by = "sample"))
Guttman.all <- RunUMAP(Guttman.all, reduction = "mnn", dims = 1:30)
Guttman.all <- FindNeighbors(Guttman.all, dims = 1:30, reduction = "mnn")
Guttman.all <- FindClusters(Guttman.all, resolution = .4)
DimPlot(Guttman.all, reduction = "umap", label = TRUE)
DimPlot(Guttman.all, reduction = "umap", group.by = "sample", label = TRUE, repel = TRUE)+NoLegend()
DimPlot(Guttman.all, reduction = "umap", group.by = "Status", label = FALSE)
annotation.genes <- c("KRT15", "KRT14", "KRT5", "KRT1", "KRT10", "KRT6", "TOP2A", "UBE2C", "CFD", "CFH", "MFAP5", "FBN1", "COL1A1", "COL6A3", "TAGLN", "ACTA2", "DCT", "MLANA", "DCD", "AQP5", "ACKR1", "VWF", "PECAM1", "FABP4", "CD36", "CCL21", "LYVE1", "LYZ", "FCER1G", "CD52", "CD3D")
DotPlot(Guttman.all, features = rev(annotation.genes), cols = "RdBu")+coord_flip()

DE.13 <- FindMarkers(Guttman.all, ident.1 = 13)
#remove cluster 13 likely doublets
Guttman.all <- subset(Guttman.all, idents = 13, invert = TRUE)
Guttman.all <- NormalizeData(Guttman.all, verbose = FALSE)
Guttman.all <- FindVariableFeatures(Guttman.all, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Guttman.all <- CellCycleScoring(Guttman.all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Guttman.all <- RunFastMNN(object.list = SplitObject(Guttman.all, split.by = "sample"))
Guttman.all <- RunUMAP(Guttman.all, reduction = "mnn", dims = 1:30)
Guttman.all <- FindNeighbors(Guttman.all, dims = 1:30, reduction = "mnn")
Guttman.all <- FindClusters(Guttman.all, resolution = .4)
DimPlot(Guttman.all, reduction = "umap", label = TRUE)
DimPlot(Guttman.all, reduction = "umap", group.by = "sample", label = TRUE, repel = TRUE)+NoLegend()
DimPlot(Guttman.all, reduction = "umap", group.by = "Status", label = FALSE)
annotation.genes <- c("KRT15", "KRT14", "KRT5", "KRT1", "KRT10", "KRT6", "TOP2A", "UBE2C", "CFD", "CFH", "MFAP5", "FBN1", "COL1A1", "COL6A3", "TAGLN", "ACTA2", "DCT", "MLANA", "DCD", "AQP5", "ACKR1", "VWF", "PECAM1", "FABP4", "CD36", "CCL21", "LYVE1", "LYZ", "FCER1G", "CD52", "CD3D")
DotPlot(Guttman.all, features = rev(annotation.genes), cols = "RdBu")+coord_flip()

Idents(Guttman.all) <- "seurat_clusters"
cluster_levels <- c(9,3,5,8,0,13,6,10,12,1,2,11,7,4)
new.cluster.ids <- c("KC", "KC", "KC", "FB", "FB", "FB", "PC-vSMC", "MEL/NC/SC", "SGC", "VEC", "VEC", "LEC", "MAC-DC", "TC-NKT")
Idents(Guttman.all) <- plyr::mapvalues(x = Idents(Guttman.all), from = cluster_levels, to = new.cluster.ids)
Guttman.all$Cell.Type <- Idents(Guttman.all)
Idents(Guttman.all) <- "Cell.Type"
cluster_levels <- c("KC", "FB", "PC-vSMC", "MEL/NC/SC", "SGC", "VEC", "LEC", "MAC-DC", "TC-NKT")
Guttman.all@meta.data$Cell.Type <- factor(x = Idents(Guttman.all), levels = cluster_levels)
DimPlot(Guttman.all, reduction = "umap", group.by = "Cell.Type", label = TRUE, repel = TRUE)+NoLegend()
DotPlot(Guttman.all, features = rev(annotation.genes), cols = "RdBu", group.by = "Cell.Type")+coord_flip()
save(Guttman.all, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Guttman/Processed/Guttman.all.RData")

#Subsetting melanocytes
Guttman.mel <- subset(Guttman.all, idents = "MEL/NC/SC")
Guttman.mel <- NormalizeData(Guttman.mel, verbose = FALSE)
Guttman.mel <- FindVariableFeatures(Guttman.mel, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Guttman.mel <- RunFastMNN(object.list = SplitObject(Guttman.mel, split.by = "sample"))
Guttman.mel <- RunUMAP(Guttman.mel, reduction = "mnn", dims = 1:30)
Guttman.mel <- FindNeighbors(Guttman.mel, dims = 1:30, reduction = "mnn")
Guttman.mel <- FindClusters(Guttman.mel, resolution = .1)
DimPlot(Guttman.mel, reduction = "umap", label = TRUE)
DimPlot(Guttman.mel, reduction = "umap", group.by = "sample", label = TRUE, repel = TRUE)+NoLegend()
DimPlot(Guttman.mel, reduction = "umap", group.by = "Status", label = FALSE)
annotation.genes <- c("MITF", "DCT", "TYR", "MLANA", "NRXN1", "SCN7A", "GLDN", "TJP1")
DotPlot(Guttman.mel, features = rev(annotation.genes), cols = "RdBu")+coord_flip()

Guttman.mel <- subset(Guttman.mel, idents = 1)
Guttman.mel <- NormalizeData(Guttman.mel, verbose = FALSE)
Guttman.mel <- FindVariableFeatures(Guttman.mel, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Guttman.mel <- RunFastMNN(object.list = SplitObject(Guttman.mel, split.by = "sample"))
Guttman.mel <- RunUMAP(Guttman.mel, reduction = "mnn", dims = 1:30)
Guttman.mel <- FindNeighbors(Guttman.mel, dims = 1:30, reduction = "mnn")
Guttman.mel <- FindClusters(Guttman.mel, resolution = .1)
DimPlot(Guttman.mel, reduction = "umap", label = TRUE)
DimPlot(Guttman.mel, reduction = "umap", group.by = "sample", label = TRUE, repel = TRUE)+NoLegend()
DimPlot(Guttman.mel, reduction = "umap", group.by = "Status", label = FALSE)
annotation.genes <- c("MITF", "DCT", "TYR", "MLANA")
DotPlot(Guttman.mel, features = rev(annotation.genes), cols = "RdBu")+coord_flip()

save(Guttman.all, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Guttman/Processed/Guttman.all.RData")
save(Guttman.mel, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Guttman/Processed/Guttman.mel.RData")
```

```{r Cho et al.}
# object creation and metadata formatting 
cho.counts <- fread(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Cho/Raw/Cell Reports Processed Data/counts_filt.csv.gz")
genes <- cho.counts$V1
cho.counts <- cho.counts[,2:length(cho.counts)]
cho.counts <- as.matrix(cho.counts)
cho.counts <- as(cho.counts, "sparseMatrix")
rownames(cho.counts) <- genes
cho.meta <- read.csv(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Cho/Raw/Cell Reports Processed Data/publication_celldata.csv", row.names = 1)
Cho.all <- CreateSeuratObject(counts = cho.counts, meta.data = cho.meta)

#QC and Clustering
mt.genes <- rownames(Cho.all)[grep("^MT-", rownames(Cho.all))]
Cho.all <- PercentageFeatureSet(Cho.all, features = mt.genes, col.name = "percent.mt")
rb.genes <- rownames(Cho.all)[grep("^RP[SL]",rownames(Cho.all))]
Cho.all <- PercentageFeatureSet(Cho.all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(Cho.all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
Cho.all$novelty.score <- novelty$Score
Idents(Cho.all) <- "orig.ident"
VlnPlot(Cho.all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(Cho.all, features = c("percent.mt", "percent.ribo"), ncol = 3)
VlnPlot(Cho.all, features = "nFeature_RNA") + geom_hline(yintercept = 200)
VlnPlot(Cho.all, features = "nCount_RNA") + geom_hline(yintercept = 6500)
FeatureScatter(Cho.all, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(Cho.all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
FeatureScatter(Cho.all, feature1 = "nFeature_RNA", feature2 = "percent.mt")

Cho.all <- NormalizeData(Cho.all, verbose = FALSE)
Cho.all <- FindVariableFeatures(Cho.all, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Cho.all <- CellCycleScoring(Cho.all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Cho.all <- RunFastMNN(object.list = SplitObject(Cho.all, split.by = "sample"))
Cho.all <- RunUMAP(Cho.all, reduction = "mnn", dims = 1:30)
Cho.all <- FindNeighbors(Cho.all, dims = 1:30, reduction = "mnn")
Cho.all <- FindClusters(Cho.all, resolution = .4)
DimPlot(Cho.all, reduction = "umap", label = TRUE)
DimPlot(Cho.all, reduction = "umap", group.by = "sample", label = FALSE)
DimPlot(Cho.all, reduction = "umap", group.by = "tissue", label = FALSE)
DimPlot(Cho.all, reduction = "umap", group.by = "cluster.name", label = FALSE)
VlnPlot(Cho.all, features = c("MITF", "DCT", "TYR", "MLANA"), pt.size = 0)

#Subsetting melanocytes
Cho.mel <- subset(Cho.all, idents = c(7,9,14))
Cho.mel <- NormalizeData(Cho.mel, verbose = FALSE)
Cho.mel <- FindVariableFeatures(Cho.mel, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Cho.mel <- CellCycleScoring(Cho.mel, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Cho.mel <- RunFastMNN(object.list = SplitObject(Cho.mel, split.by = "sample"))
Cho.mel <- RunUMAP(Cho.mel, reduction = "mnn", dims = 1:30)
Cho.mel <- FindNeighbors(Cho.mel, dims = 1:30, reduction = "mnn")
Cho.mel <- FindClusters(Cho.mel, resolution = .4)
DimPlot(Cho.mel, reduction = "umap", label = TRUE)
DimPlot(Cho.mel, reduction = "umap", group.by = "sample", label = FALSE)
DimPlot(Cho.mel, reduction = "umap", group.by = "tissue", label = FALSE)
DimPlot(Cho.mel, reduction = "umap", group.by = "Phase", label = FALSE)
VlnPlot(Cho.mel, features = c("MITF", "DCT", "TYR", "MLANA"), pt.size = 0)
FeaturePlot(Cho.mel, features = "MLANA")

save(Cho.all, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Cho/Processed/Cho.all.RData")
save(Cho.mel, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Cho/Processed/Cho.mel.RData")
```

```{r Hughes et al.}
# object creation and metadata formatting 
hughes.counts <- fread(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Hughes/raw1/ExpressionFile.txt.gz")
hughes.meta1 <- fread(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Hughes/raw1/MetadataUploadFinal.txt")
hughes.meta2 <- fread(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Hughes/raw1/alexandria_structured_metadata.txt")
hughes.meta1 <- hughes.meta1[2:length(rownames(hughes.meta1)),]
hughes.meta2 <- hughes.meta2[2:length(rownames(hughes.meta2)),]
hughes.meta <- cbind(hughes.meta1, hughes.meta2)
colnames(hughes.counts)
genes <- hughes.counts$GENE
hughes.counts <- hughes.counts[,2:length(hughes.counts)]
hughes.counts <- as.matrix(hughes.counts)
hughes.counts <- as(hughes.counts, "sparseMatrix")
rownames(hughes.counts) <- genes
cells <- as.character(hughes.meta$NAME)
hughes.meta <- hughes.meta[,2:length(hughes.meta)]
hughes.meta <- as.data.frame(hughes.meta)
rownames(hughes.meta) <- cells
hughes.meta <- hughes.meta[,c(1:4,12:19,21,23,25,26,29,30)]
Hughes.all <- CreateSeuratObject(counts = hughes.counts, meta.data = hughes.meta)

#QC and Clustering
mt.genes <- rownames(Hughes.all)[grep("^MT-", rownames(Hughes.all))]
Hughes.all <- PercentageFeatureSet(Hughes.all, features = mt.genes, col.name = "percent.mt")
rb.genes <- rownames(Hughes.all)[grep("^RP[SL]",rownames(Hughes.all))]
Hughes.all <- PercentageFeatureSet(Hughes.all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(Hughes.all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
Hughes.all$novelty.score <- novelty$Score

Idents(Hughes.all) <- "orig.ident"
VlnPlot(Hughes.all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(Hughes.all, features = c("percent.mt", "percent.ribo"), ncol = 3)
VlnPlot(Hughes.all, features = "nFeature_RNA") + geom_hline(yintercept = 200)
VlnPlot(Hughes.all, features = "nCount_RNA") + geom_hline(yintercept = 6500)
FeatureScatter(Hughes.all, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(Hughes.all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
FeatureScatter(Hughes.all, feature1 = "nFeature_RNA", feature2 = "percent.mt")

Hughes.all <- NormalizeData(Hughes.all, verbose = FALSE)
Hughes.all <- FindVariableFeatures(Hughes.all, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Hughes.all <- CellCycleScoring(Hughes.all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Hughes.all <- RunFastMNN(object.list = SplitObject(Hughes.all, split.by = "donor_id"))
Hughes.all <- RunUMAP(Hughes.all, reduction = "mnn", dims = 1:30)
Hughes.all <- FindNeighbors(Hughes.all, dims = 1:30, reduction = "mnn")
Hughes.all <- FindClusters(Hughes.all, resolution = .4)
DimPlot(Hughes.all, reduction = "umap", label = TRUE)
DimPlot(Hughes.all, reduction = "umap", group.by = "Condition", label = FALSE)
DimPlot(Hughes.all, reduction = "umap", group.by = "CellType1.50", label = FALSE)
DimPlot(Hughes.all, reduction = "umap", group.by = "Specific", label = FALSE)
DimPlot(Hughes.all, reduction = "umap", group.by = "sex", label = FALSE)
DimPlot(Hughes.all, reduction = "umap", group.by = "race__ontology_label", label = FALSE)
DimPlot(Hughes.all, reduction = "umap", group.by = "Age", label = FALSE)
VlnPlot(Hughes.all, features = c("MITF", "DCT", "TYR", "MLANA"), pt.size = 0)

#Subsetting melanocytes
Idents(Hughes.all) <- "CellType1.50"
Hughes.mel <- subset(Hughes.all, idents = "Melanocyte")
Hughes.mel <- NormalizeData(Hughes.mel, verbose = FALSE)
Hughes.mel <- FindVariableFeatures(Hughes.mel, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Hughes.mel <- CellCycleScoring(Hughes.mel, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Hughes.mel <- RunFastMNN(object.list = SplitObject(Hughes.mel, split.by = "donor_id"))
Hughes.mel <- RunUMAP(Hughes.mel, reduction = "mnn", dims = 1:30)
Hughes.mel <- FindNeighbors(Hughes.mel, dims = 1:30, reduction = "mnn")
Hughes.mel <- FindClusters(Hughes.mel, resolution = .4)
DimPlot(Hughes.mel, reduction = "umap", label = TRUE)
DimPlot(Hughes.mel, reduction = "umap", group.by = "Condition", label = FALSE)
DimPlot(Hughes.mel, reduction = "umap", group.by = "CellType1.50", label = FALSE)
DimPlot(Hughes.mel, reduction = "umap", group.by = "Specific", label = FALSE)
DimPlot(Hughes.mel, reduction = "umap", group.by = "sex", label = FALSE)
DimPlot(Hughes.mel, reduction = "umap", group.by = "race__ontology_label", label = FALSE)
DimPlot(Hughes.mel, reduction = "umap", group.by = "Age", label = FALSE)

save(Hughes.all, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Hughes/Processed/Hughes.all.RData")
save(Hughes.mel, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Hughes/Processed/Hughes.mel.RData")
```

```{r Haniffa et al. fetal}
# object creation and metadata formatting 
folders <- list.files(path = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa Fetal/Raw")
new.names <- sub("^", "HF_", folders)
meta.data <- fread(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa Fetal/supp/E-MTAB-7407.sdrf.txt")

for(i in 1:length(folders)){
  temp.counts <- Read10X(data.dir = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa Fetal/Raw/", folders[i], "/GRCh38"))
  temp.meta <- read.csv(file = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa Fetal/Raw/", folders[i], "/",folders[i], ".csv"), row.names = 1)
  rownames(temp.meta) <- sub('$', '-1', rownames(temp.meta))
  temp <- CreateSeuratObject(counts = temp.counts, meta.data = temp.meta, min.cells = 3, min.features = 200)
  rownum <- which(meta.data$`Source Name` == folders[i])
  temp$Sample <- meta.data$`Comment[ENA_SAMPLE]`[rownum]
  temp$Age <- meta.data$`Characteristics[age]`[rownum]
  temp$Sex <- meta.data$`Characteristics[sex]`[rownum]
  temp$Tissue <- meta.data$`Characteristics[organism part]`[rownum]
  temp$Donor_id <- meta.data$`Characteristics[individual]`[rownum]
  temp$Trimester <- meta.data$`Characteristics[clinical information]`[rownum]
  temp$Sorting <- meta.data$`Characteristics[facs sorting]`[rownum]
  assign(paste(new.names[i]), temp)
}

Haniffa.fetal.all <- merge(HF_4834STDY7002875, HF_4834STDY7002876)
for(i in 3:length(new.names)){
  temp <- get(new.names[i])
  Haniffa.fetal.all <- merge(Haniffa.fetal.all, temp)
}

#QC and Clustering
mt.genes <- rownames(Haniffa.fetal.all)[grep("^MT-", rownames(Haniffa.fetal.all))]
Haniffa.fetal.all <- PercentageFeatureSet(Haniffa.fetal.all, features = mt.genes, col.name = "percent.mt")
rb.genes <- rownames(Haniffa.fetal.all)[grep("^RP[SL]",rownames(Haniffa.fetal.all))]
Haniffa.fetal.all <- PercentageFeatureSet(Haniffa.fetal.all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(Haniffa.fetal.all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
Haniffa.fetal.all$novelty.score <- novelty$Score

Haniffa.fetal.all <- subset(Haniffa.fetal.all, subset = percent.mt < 20)

Idents(Haniffa.fetal.all) <- "orig.ident"
VlnPlot(Haniffa.fetal.all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(Haniffa.fetal.all, features = c("percent.mt", "percent.ribo"), ncol = 3)
VlnPlot(Haniffa.fetal.all, features = "nFeature_RNA") + geom_hline(yintercept = 200)
VlnPlot(Haniffa.fetal.all, features = "nCount_RNA") + geom_hline(yintercept = 6500)
FeatureScatter(Haniffa.fetal.all, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(Haniffa.fetal.all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
FeatureScatter(Haniffa.fetal.all, feature1 = "nFeature_RNA", feature2 = "percent.mt")

Haniffa.fetal.all <- NormalizeData(Haniffa.fetal.all, verbose = FALSE)
Haniffa.fetal.all <- FindVariableFeatures(Haniffa.fetal.all, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Haniffa.fetal.all <- CellCycleScoring(Haniffa.fetal.all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Haniffa.fetal.all <- RunFastMNN(object.list = SplitObject(Haniffa.fetal.all, split.by = "Sample"))
Haniffa.fetal.all <- RunUMAP(Haniffa.fetal.all, reduction = "mnn", dims = 1:30)
Haniffa.fetal.all <- FindNeighbors(Haniffa.fetal.all, dims = 1:30, reduction = "mnn")
Haniffa.fetal.all <- FindClusters(Haniffa.fetal.all, resolution = .1)
DimPlot(Haniffa.fetal.all, reduction = "umap", label = TRUE)
DimPlot(Haniffa.fetal.all, reduction = "umap", group.by = "Sample", label = FALSE)
DimPlot(Haniffa.fetal.all, reduction = "umap", group.by = "Age", label = FALSE)
DimPlot(Haniffa.fetal.all, reduction = "umap", group.by = "Sex", label = FALSE)
DimPlot(Haniffa.fetal.all, reduction = "umap", group.by = "Tissue", label = FALSE)
DimPlot(Haniffa.fetal.all, reduction = "umap", group.by = "Donor_id", label = FALSE)
DimPlot(Haniffa.fetal.all, reduction = "umap", group.by = "Trimester", label = FALSE)
DimPlot(Haniffa.fetal.all, reduction = "umap", group.by = "Cell.Labels", label = TRUE, repel = TRUE)+NoLegend()
FeaturePlot(Haniffa.fetal.all, features = "MITF", order = TRUE)
VlnPlot(Haniffa.fetal.all, features = c("MITF"), pt.size = 0)
VlnPlot(Haniffa.fetal.all, features = c("KIT"), pt.size = 0)
VlnPlot(Haniffa.fetal.all, features = c("EDNRB"), pt.size = 0)
VlnPlot(Haniffa.fetal.all, features = c("DHH"), pt.size = 0)
VlnPlot(Haniffa.fetal.all, features = c("PLP1"), pt.size = 0)
VlnPlot(Haniffa.fetal.all, features = c("SOX10"), pt.size = 0)
DimPlot(Haniffa.fetal.all, reduction = "umap", label = TRUE)

Idents(Haniffa.fetal.all) <- "Cell.Labels"
Haniffa.fetal.no_immune <- subset(Haniffa.fetal.all, idents = c("Non-immune", "Fibroblast", "Hepatocyte", "Endothelial cell"))
Haniffa.fetal.no_immune <- NormalizeData(Haniffa.fetal.no_immune, verbose = FALSE)
Haniffa.fetal.no_immune <- FindVariableFeatures(Haniffa.fetal.no_immune, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Haniffa.fetal.no_immune <- CellCycleScoring(Haniffa.fetal.no_immune, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Haniffa.fetal.no_immune <- RunFastMNN(object.list = SplitObject(Haniffa.fetal.no_immune, split.by = "Sample"))
Haniffa.fetal.no_immune <- RunUMAP(Haniffa.fetal.no_immune, reduction = "mnn", dims = 1:30)
Haniffa.fetal.no_immune <- FindNeighbors(Haniffa.fetal.no_immune, dims = 1:30, reduction = "mnn")
Haniffa.fetal.no_immune <- FindClusters(Haniffa.fetal.no_immune, resolution = .1)
DimPlot(Haniffa.fetal.no_immune, reduction = "umap", label = TRUE)
DimPlot(Haniffa.fetal.no_immune, reduction = "umap", group.by = "Sample", label = FALSE)
DimPlot(Haniffa.fetal.no_immune, reduction = "umap", group.by = "Age", label = FALSE)
DimPlot(Haniffa.fetal.no_immune, reduction = "umap", group.by = "Sex", label = FALSE)
DimPlot(Haniffa.fetal.no_immune, reduction = "umap", group.by = "Tissue", label = FALSE)
DimPlot(Haniffa.fetal.no_immune, reduction = "umap", group.by = "Donor_id", label = FALSE)
DimPlot(Haniffa.fetal.no_immune, reduction = "umap", group.by = "Trimester", label = FALSE)
DimPlot(Haniffa.fetal.no_immune, reduction = "umap", group.by = "Cell.Labels", label = TRUE, repel = TRUE)+NoLegend()
VlnPlot(Haniffa.fetal.no_immune, features = c("MITF"), pt.size = 0)
VlnPlot(Haniffa.fetal.no_immune, features = c("KIT"), pt.size = 0)
VlnPlot(Haniffa.fetal.no_immune, features = c("EDNRB"), pt.size = 0)
VlnPlot(Haniffa.fetal.no_immune, features = c("DHH"), pt.size = 0)
VlnPlot(Haniffa.fetal.no_immune, features = c("PLP1"), pt.size = 0)
VlnPlot(Haniffa.fetal.no_immune, features = c("SOX10"), pt.size = 0)
DimPlot(Haniffa.fetal.no_immune, reduction = "umap", label = TRUE)

#Subsetting melanocytes
Haniffa.fetal.mel <- subset(Haniffa.fetal.no_immune, idents = 8)
Haniffa.fetal.mel <- NormalizeData(Haniffa.fetal.mel, verbose = FALSE)
Haniffa.fetal.mel <- FindVariableFeatures(Haniffa.fetal.mel, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Haniffa.fetal.mel <- CellCycleScoring(Haniffa.fetal.mel, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Haniffa.fetal.mel <- RunFastMNN(object.list = SplitObject(Haniffa.fetal.mel, split.by = "Sample"))
Haniffa.fetal.mel <- RunUMAP(Haniffa.fetal.mel, reduction = "mnn", dims = 1:30)
Haniffa.fetal.mel <- FindNeighbors(Haniffa.fetal.mel, dims = 1:30, reduction = "mnn")
Haniffa.fetal.mel <- FindClusters(Haniffa.fetal.mel, resolution = .1)
DimPlot(Haniffa.fetal.mel, reduction = "umap", label = TRUE)
DimPlot(Haniffa.fetal.mel, reduction = "umap", group.by = "Sample", label = FALSE)
DimPlot(Haniffa.fetal.mel, reduction = "umap", group.by = "Age", label = FALSE)
DimPlot(Haniffa.fetal.mel, reduction = "umap", group.by = "Sex", label = FALSE)
DimPlot(Haniffa.fetal.mel, reduction = "umap", group.by = "Tissue", label = FALSE)
DimPlot(Haniffa.fetal.mel, reduction = "umap", group.by = "Donor_id", label = FALSE)
DimPlot(Haniffa.fetal.mel, reduction = "umap", group.by = "Trimester", label = FALSE)
DimPlot(Haniffa.fetal.mel, reduction = "umap", group.by = "Cell.Labels", label = TRUE, repel = TRUE)+NoLegend()

save(Haniffa.fetal.all, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa Fetal/Processed/Haniffa.fetal.all.RData")
save(Haniffa.fetal.no_immune, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa Fetal/Processed/Haniffa.fetal.no_immune.RData")
save(Haniffa.fetal.mel, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Haniffa Fetal/Processed/Haniffa.fetal.mel.RData")
```

```{r Brunner et al}
# object creation and metadata formatting 
files <- list.files(path= "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Brunner/Raw/")
for(i in 1:length(files)){
  temp.counts <- Read10X(data.dir = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Brunner/Raw/", files[i]))
  temp <- CreateSeuratObject(temp.counts)
  temp$Race <- "White"
  temp$Sample <- paste(files[i])
  if(files[i] %in% c("AD20", "AD24", "HC5", "HC6", "HC7", "AD1", "AD3")){
    temp$Sex <- "Male"
  } else if(files[i] %in% c("AD5", "AD6", "AD7", "AD8")){
    temp$Sex <- NA
  }else {
    temp$Sex <- "Female"
  }
  
  if(files[i] %in% c("AD20", "AD21", "AD24", "AD25")){
    temp$Status <- "Healed AD"
  } else if(files[i] %in% c("AD1", "AD2", "AD3", "AD4", "AD5", "AD6", "AD7", "AD8")){
    temp$Status <- "AD"
  }else{
    temp$Status <- "Healthy"
  }
  
  if(files[i] %in% c("AD20")){
    temp$Age <- 22
  } else if(files[i] %in% c("AD21")){
    temp$Age <- 45
  }else if(files[i] %in% c("AD24", "AD25")){
    temp$Age <- 41
  }else if(files[i] %in% c("HC1", "HC2")){
    temp$Age <- 42
  }else if(files[i] %in% c("HC3")){
    temp$Age <- 47
  }else if(files[i] %in% c("HC4")){
    temp$Age <- 49
  }else if(files[i] %in% c("HC5")){
    temp$Age <- 39
  }else if(files[i] %in% c("HC6", "HC7")){
    temp$Age <- 40
  }else if(files[i] %in% c("AD1")){
    temp$Age <- 18
  }else if(files[i] %in% c("AD2")){
    temp$Age <- 19
  }else if(files[i] %in% c("AD3")){
    temp$Age <- 34
  }else if(files[i] %in% c("AD4")){
    temp$Age <- 33
  }else{
    temp$Age <- NA
  }
  assign(paste(files[i]), temp)
}
Brunner.all <- merge(AD1, AD2)
for(i in 3:length(files)){
  temp <- get(files[i])
  Brunner.all <- merge(Brunner.all, temp)
}

#QC and Clustering
mt.genes <- rownames(Brunner.all)[grep("^MT-", rownames(Brunner.all))]
Brunner.all <- PercentageFeatureSet(Brunner.all, features = mt.genes, col.name = "percent.mt")
rb.genes <- rownames(Brunner.all)[grep("^RP[SL]",rownames(Brunner.all))]
Brunner.all <- PercentageFeatureSet(Brunner.all, features = rb.genes, col.name = "percent.ribo")
novelty <- as.data.frame(Brunner.all@meta.data[,2:3])
novelty$nCount_RNA <- log10(novelty$nCount_RNA)
novelty$nFeature_RNA <- log10(novelty$nFeature_RNA)
novelty$Score <- novelty$nFeature_RNA/novelty$nCount_RNA
Brunner.all$novelty.score <- novelty$Score
Brunner.all <- subset(Brunner.all, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & percent.mt < 15)
Idents(Brunner.all) <- "orig.ident"
VlnPlot(Brunner.all, features = c("nFeature_RNA", "nCount_RNA", "novelty.score"), ncol = 3)
VlnPlot(Brunner.all, features = c("percent.mt", "percent.ribo"), ncol = 3)
VlnPlot(Brunner.all, features = "nFeature_RNA") + geom_hline(yintercept = 200)
VlnPlot(Brunner.all, features = "nCount_RNA") + geom_hline(yintercept = 6500)
FeatureScatter(Brunner.all, feature1 = "nCount_RNA", feature2 = "percent.mt")
FeatureScatter(Brunner.all, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
FeatureScatter(Brunner.all, feature1 = "nFeature_RNA", feature2 = "percent.mt")

Brunner.all <- NormalizeData(Brunner.all, verbose = FALSE)
Brunner.all <- FindVariableFeatures(Brunner.all, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Brunner.all <- CellCycleScoring(Brunner.all, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Brunner.all <- RunFastMNN(object.list = SplitObject(Brunner.all, split.by = "Sample"))
Brunner.all <- RunUMAP(Brunner.all, reduction = "mnn", dims = 1:30)
Brunner.all <- FindNeighbors(Brunner.all, dims = 1:30, reduction = "mnn")
Brunner.all <- FindClusters(Brunner.all, resolution = .4)
DimPlot(Brunner.all, reduction = "umap", label = TRUE)
DimPlot(Brunner.all, reduction = "umap", group.by = "Sample", label = FALSE)
DimPlot(Brunner.all, reduction = "umap", group.by = "Sex", label = FALSE)
DimPlot(Brunner.all, reduction = "umap", group.by = "Status", label = FALSE)
DimPlot(Brunner.all, reduction = "umap", group.by = "Age", label = FALSE)
VlnPlot(Brunner.all, features = c("TYR"), pt.size = 0)

#Subsetting melanocytes
Brunner.mel <- subset(Brunner.all, idents = 7)
Brunner.mel <- NormalizeData(Brunner.mel, verbose = FALSE)
Brunner.mel <- FindVariableFeatures(Brunner.mel, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Brunner.mel <- CellCycleScoring(Brunner.mel, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
Brunner.mel <- RunFastMNN(object.list = SplitObject(Brunner.mel, split.by = "Sample"))
Brunner.mel <- RunUMAP(Brunner.mel, reduction = "mnn", dims = 1:30)
Brunner.mel <- FindNeighbors(Brunner.mel, dims = 1:30, reduction = "mnn")
Brunner.mel <- FindClusters(Brunner.mel, resolution = .4)
DimPlot(Brunner.mel, reduction = "umap", label = TRUE)
DimPlot(Brunner.mel, reduction = "umap", group.by = "Sample", label = FALSE)
DimPlot(Brunner.mel, reduction = "umap", group.by = "Sex", label = FALSE)
DimPlot(Brunner.mel, reduction = "umap", group.by = "Status", label = FALSE)
DimPlot(Brunner.mel, reduction = "umap", group.by = "Age", label = FALSE)
VlnPlot(Brunner.mel, features = c("MITF", "DCT", "TYR", "MLANA"), pt.size = 0)
FeaturePlot(Brunner.mel, features = "MLANA")

save(Brunner.all, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Brunner/Processed/Brunner.all.RData")
save(Brunner.mel, file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/Brunner/Processed/Brunner.mel.RData")
```

```{r Make metadata labeling consistent across all datasets}
melanocyte.datasets <- c("Yang.mel", "Lyko.mel", "Haniffa.mel", "Guttman.mel", "Cho.mel", "Hughes.mel", "Haniffa.fetal.mel", "Brunner.mel", "Belote.mel")
drops <- c("orig.ident", "nCount_RNA", "nFeature_RNA", "percent.mt", "percent.ribo", "novelty.score", "S.Score", "G2M.Score", "Phase")

for(x in 1:length(melanocyte.datasets)){
  data <- get(melanocyte.datasets[x])
  temp.meta <- data@meta.data[ , !(names(data@meta.data) %in% drops)]
  meta.list <- vector(mode = "list", length = length(colnames(temp.meta)))
  names(meta.list) <- colnames(temp.meta)
  for(i in 1:length(meta.list)){
    meta.list[[i]] <- names(table(temp.meta[,i])) 
    }
  capture.output(meta.list, file = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/", melanocyte.datasets[x], ".txt"))
}

files <- list.files(path = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/patient.meta.data")
file.names <- sub('.csv', '', files)
for(i in 1:length(files)){
  patient.meta <- read.csv(file = paste0("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/Published Datasets/melanocytes/patient.meta.data/", files[i]))
  assign(paste(file.names[i]), patient.meta)
}

melanocyte.datasets <- c("Yang.mel", "Lyko.mel", "Haniffa.mel", "Guttman.mel", "Cho.mel", "Hughes.mel", "Haniffa.fetal.mel", "Brunner.mel", "Belote.mel")
patient.meta <- sub('.mel', '.patient.meta', melanocyte.datasets)

Yang.mel$Sample_ID <- Yang.mel$dataset
Lyko.mel$Sample_ID <- Lyko.mel$subj
Haniffa.mel$Sample_ID <- Haniffa.mel$donor_id
Guttman.mel$Sample_ID <- Guttman.mel$sample
Cho.mel$Sample_ID <- Cho.mel$sample
Hughes.mel$Sample_ID <- Hughes.mel$SampleCondition
Haniffa.fetal.mel$Sample_ID <- Haniffa.fetal.mel$Donor_id
Brunner.mel$Sample_ID <- Brunner.mel$Sample
Belote.mel$Sample_ID <- Belote.mel$Sample

for(i in 1:length(melanocyte.datasets)){
temp.ds <- get(melanocyte.datasets[i])
temp.meta <- get(patient.meta[i])
samples <- names(table(temp.ds$Sample_ID))
temp.ds$Dataset <- melanocyte.datasets[i]
temp.ds$Dataset_Sample <- sub("^", paste0(melanocyte.datasets[i], "_"), as.character(temp.ds$Sample_ID))
ds.meta <- temp.ds@meta.data
drops <- colnames(temp.meta)
ds.meta <- ds.meta[ , !(names(ds.meta) %in% drops)]
ds.meta[,(length(ds.meta)+1):(length(ds.meta)+length(temp.meta))] <- NA
colnames(ds.meta) <- c(colnames(ds.meta)[1:(length(ds.meta)-length(colnames(temp.meta)))], colnames(temp.meta))
for(x in 1:length(samples)){
rownums <- which(ds.meta$Sample_ID == samples[x])
meta.row <- which(temp.meta$Sample == samples[x])
ds.meta$Sample[rownums] <- temp.meta$Sample[meta.row]
ds.meta$Disease_Status[rownums] <- temp.meta$Disease_Status[meta.row]
ds.meta$Sex[rownums] <- temp.meta$Sex[meta.row]
if("Age" %in% colnames(temp.meta)){
  ds.meta$Age[rownums] <- temp.meta$Age[meta.row]
}else{
  Sys.sleep(0)
}
ds.meta$Stage[rownums] <- temp.meta$Stage[meta.row]
ds.meta$Race[rownums] <- temp.meta$Race[meta.row]
if("Site" %in% colnames(temp.meta)){
  ds.meta$Site[rownums] <- temp.meta$Site[meta.row]
}else{
  Sys.sleep(0)
}
ds.meta$Tissue[rownums] <- temp.meta$Tissue[meta.row]
ds.meta$Tissue_Site[rownums] <- temp.meta$Tissue_Site[meta.row]
}
temp.ds@meta.data <- ds.meta
assign(paste(melanocyte.datasets[i]), temp.ds)
}

Idents(Brunner.mel) <- "Sample_ID"
###remove "NA" samples from Brunner
Brunner.mel <- subset(Brunner.mel, idents = as.character(names(table(Brunner.mel@meta.data$Sample_ID))))
```

```{r Merge all primary melanocyte datasets}
Primary.Melanocytes.All <- merge(Yang.mel, Lyko.mel)
for(i in 3:length(melanocyte.datasets)){
  temp <- get(melanocyte.datasets[i])
  Primary.Melanocytes.All <- merge(Primary.Melanocytes.All, temp)
}

Primary.Melanocytes.All <- NormalizeData(Primary.Melanocytes.All, verbose = FALSE)
Primary.Melanocytes.All <- FindVariableFeatures(Primary.Melanocytes.All, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Primary.Melanocytes.All <- RunFastMNN(object.list = SplitObject(Primary.Melanocytes.All, split.by = "Dataset"))
Primary.Melanocytes.All <- RunUMAP(Primary.Melanocytes.All, reduction = "mnn", dims = 1:30)
Primary.Melanocytes.All <- FindNeighbors(Primary.Melanocytes.All, dims = 1:30, reduction = "mnn")
Primary.Melanocytes.All <- FindClusters(Primary.Melanocytes.All, resolution = .2)
DimPlot(Primary.Melanocytes.All, reduction = "umap", label = TRUE)

#Subset Healthy melanocytes
Idents(Primary.Melanocytes.All) <- "Disease_Status"
Primary.Melanocytes.Healthy <- subset(Primary.Melanocytes.All, idents = "Healthy")
Primary.Melanocytes.Healthy <- NormalizeData(Primary.Melanocytes.Healthy, verbose = FALSE)
Primary.Melanocytes.Healthy <- FindVariableFeatures(Primary.Melanocytes.Healthy, selection.method = "vst", 
        nfeatures = 2000, verbose = FALSE)
Primary.Melanocytes.Healthy <- RunFastMNN(object.list = SplitObject(Primary.Melanocytes.Healthy, split.by = "Dataset"))
Primary.Melanocytes.Healthy <- RunUMAP(Primary.Melanocytes.Healthy, reduction = "mnn", dims = 1:30)
Primary.Melanocytes.Healthy <- FindNeighbors(Primary.Melanocytes.Healthy, dims = 1:30, reduction = "mnn")
Primary.Melanocytes.Healthy <- FindClusters(Primary.Melanocytes.Healthy, resolution = .1)
DimPlot(Primary.Melanocytes.Healthy, reduction = "umap", label = FALSE)
DimPlot(Primary.Melanocytes.Healthy, reduction = "umap", group.by = "Subtype", label = FALSE)
DimPlot(Primary.Melanocytes.Healthy, reduction = "umap", group.by = "Dataset", label = FALSE)

#format metadata
Idents(Primary.Melanocytes.Healthy) <- "seurat_clusters"
cluster_levels <- names(table(Primary.Melanocytes.Healthy$seurat_clusters))
new.cluster.ids <- c("pMelanocyte 1", "pMelanocyte 2", "pMelanocyte 3", "pMelanocyte 4")
Idents(Primary.Melanocytes.Healthy) <- plyr::mapvalues(x = Idents(Primary.Melanocytes.Healthy), from = cluster_levels, to = new.cluster.ids)
Primary.Melanocytes.Healthy$Subtype <- Idents(Primary.Melanocytes.Healthy)

Idents(Primary.Melanocytes.Healthy) <- "Tissue_Site"
cluster_levels <- names(table(Primary.Melanocytes.Healthy$Tissue_Site))
new.cluster.ids <- c("Trunk", "Mix", "Mix", "Mix", "Trunk", "Mix", "Posterior", "Posterior", "Posterior", "Mix", "Posterior", "Trunk", "Anterior", "Trunk", "Undisclosed")
Idents(Primary.Melanocytes.Healthy) <- plyr::mapvalues(x = Idents(Primary.Melanocytes.Healthy), from = cluster_levels, to = new.cluster.ids)
Primary.Melanocytes.Healthy$APaxis <- Idents(Primary.Melanocytes.Healthy)
cluster_levels <- c("Anterior", "Trunk", "Posterior", "Mix", "Undisclosed")
Primary.Melanocytes.Healthy@meta.data$APaxis <- factor(x = Idents(Primary.Melanocytes.Healthy), levels = cluster_levels)

Idents(Primary.Melanocytes.Healthy) <- "Tissue_Site"
cluster_levels <- names(table(Primary.Melanocytes.Healthy$Tissue_Site))
new.cluster.ids <- c("Ambigious", "Ambigious", "Mix", "Mix", "Ventral", "Ambigious", "Ventral", "Ventral", "Ambigious", "Mix", "Mix", "Ventral", "Ambigious", "Ventral", "Undisclosed")
Idents(Primary.Melanocytes.Healthy) <- plyr::mapvalues(x = Idents(Primary.Melanocytes.Healthy), from = cluster_levels, to = new.cluster.ids)
Primary.Melanocytes.Healthy$DVaxis <- Idents(Primary.Melanocytes.Healthy)
cluster_levels <- c("Ventral", "Mix", "Ambigious", "Undisclosed")
Primary.Melanocytes.Healthy@meta.data$DVaxis <- factor(x = Idents(Primary.Melanocytes.Healthy), levels = cluster_levels)

Idents(Primary.Melanocytes.Healthy) <- "Tissue_Site"
cluster_levels <- names(table(Primary.Melanocytes.Healthy$Tissue_Site))
new.cluster.ids <- c("Peripheral", "Peripheral", "Peripheral", "Peripheral", "Central", "Peripheral", "Central", "Central", "Peripheral", "Peripheral", "Peripheral", "Peripheral", "Central", "Central", "Undisclosed")
Idents(Primary.Melanocytes.Healthy) <- plyr::mapvalues(x = Idents(Primary.Melanocytes.Healthy), from = cluster_levels, to = new.cluster.ids)
Primary.Melanocytes.Healthy$CPaxis <- Idents(Primary.Melanocytes.Healthy)
cluster_levels <- c("Central", "Peripheral", "Undisclosed")
Primary.Melanocytes.Healthy@meta.data$CPaxis <- factor(x = Idents(Primary.Melanocytes.Healthy), levels = cluster_levels)

Idents(Primary.Melanocytes.Healthy) <- "Age"
cluster_levels <- c("7fw", "8fw", "9fw", "9.5fw", "10fw", "12fw", "16fw", "18fw", "0", "24", "25", "27", "28", "35", "37", "39", "40", "42", "47", "49", "52", "53", "56", "58", "60", "61", "63", "65", "66", "68", "69", "70", "77", "81", "Unknown")
Primary.Melanocytes.Healthy@meta.data$Age <- factor(x = Idents(Primary.Melanocytes.Healthy), levels = cluster_levels)

Idents(Primary.Melanocytes.Healthy) <- "Stage"
cluster_levels <- c("Fetal", "Neo", "Adult")
Primary.Melanocytes.Healthy@meta.data$Stage <- factor(x = Idents(Primary.Melanocytes.Healthy), levels = cluster_levels)

Idents(Primary.Melanocytes.Healthy) <- "Age"
cluster_levels <- names(table(Primary.Melanocytes.Healthy$Age))
new.cluster.ids <- c("1st Trimester", "1st Trimester", "1st Trimester", "1st Trimester", "1st Trimester", "2nd Trimester", "2nd Trimester", "2nd Trimester", "Neonatal", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Adult", "Senior", "Senior", "Senior", "Senior", "Senior", "Senior", "Senior", "Senior", "Senior", "Senior", "Undisclosed")
Idents(Primary.Melanocytes.Healthy) <- plyr::mapvalues(x = Idents(Primary.Melanocytes.Healthy), from = cluster_levels, to = new.cluster.ids)
Primary.Melanocytes.Healthy$Age_Group <- Idents(Primary.Melanocytes.Healthy)
cluster_levels <- c("1st Trimester", "2nd Trimester", "Neonatal", "Adult", "Senior","Undisclosed")
Primary.Melanocytes.Healthy@meta.data$Age_Group <- factor(x = Idents(Primary.Melanocytes.Healthy), levels = cluster_levels)

Idents(Primary.Melanocytes.Healthy) <- "Age"
cluster_levels <- names(table(Primary.Melanocytes.Healthy$Age))
new.cluster.ids <- c("1st Trimester", "1st Trimester", "1st Trimester", "1st Trimester", "1st Trimester", "2nd Trimester", "2nd Trimester", "2nd Trimester", "Neonatal", "20s", "20s", "20s", "20s", "30s", "30s", "30s", "40s", "40s", "40s", "40s", "50s", "50s", "50s", "50s", "60s", "60s", "60s", "60s", "60s", "60s", "60s", "70+", "70+", "70+", "Undisclosed")
Idents(Primary.Melanocytes.Healthy) <- plyr::mapvalues(x = Idents(Primary.Melanocytes.Healthy), from = cluster_levels, to = new.cluster.ids)
Primary.Melanocytes.Healthy$Age_Range <- Idents(Primary.Melanocytes.Healthy)
cluster_levels <- c("1st Trimester", "2nd Trimester", "Neonatal", "20s", "30s", "40s", "50s", "60s", "70+", "Undisclosed")
Primary.Melanocytes.Healthy@meta.data$Age_Range <- factor(x = Idents(Primary.Melanocytes.Healthy), levels = cluster_levels)
```

```{r Figure S4A}
DimPlot(Primary.Melanocytes.Healthy, reduction = "umap", group.by = "Dataset", label = FALSE, pt.size = .5)+theme_void()+theme(legend.position = "right")
ggsave(paste0(outputDir, "/UMAP","Primary.Melanocytes.Healthy.Subtypes" ,".jpeg"), width = 5, height = 4, dpi = 320, limitsize = FALSE)
```

```{r Scoring primary melanocytes and Figure 4A}
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures/"
Idents(mature.melanocytes) <- "Cell.Type"
DefaultAssay(mature.melanocytes) <- "RNA"
mat.mel.DE <- FindAllMarkers(mature.melanocytes, only.pos = T)
T1.enriched <- mat.mel.DE$gene[mat.mel.DE$cluster == "T1 Melanocyte"]
T2.enriched <- mat.mel.DE$gene[mat.mel.DE$cluster == "T2 Melanocyte"]
T1.mel.markers <- T1.enriched[T1.enriched %in% rownames(Primary.Melanocytes.Healthy)][1:100]
T2.mel.markers <- T2.enriched[T2.enriched %in% rownames(Primary.Melanocytes.Healthy)][1:100]
Primary.Melanocytes.Healthy <- AddModuleScore(Primary.Melanocytes.Healthy, features = list(T1.mel.markers), name = "T1.Melanocyte.signture")
Primary.Melanocytes.Healthy <- AddModuleScore(Primary.Melanocytes.Healthy, features = list(T2.mel.markers), name = "T2.Melanocyte.signture")

FeaturePlot(Primary.Melanocytes.Healthy, features = c("T1.Melanocyte.signture1", "T2.Melanocyte.signture1"), blend = TRUE, cols = c("#009FD8", "#D169D0"), pt.size = 1, order = T, blend.threshold = .5)
ggsave(paste0(outputDir, "/Primary.Melanocytes.All.T1.T2.Score.blend.jpeg"), width = 25, height = 5, dpi = 320)
```

```{r Figure 4B}
Primary.Melanocytes.Healthy$T2minusnegT1 <- Primary.Melanocytes.Healthy$T2.Melanocyte.signture1 + -(Primary.Melanocytes.Healthy$T1.Melanocyte.signture1)
T1like.Cells <- WhichCells(Primary.Melanocytes.Healthy, expression = T2minusnegT1 < 0)
T2like.Cells <- WhichCells(Primary.Melanocytes.Healthy, expression = T2minusnegT1 > 0)
Idents(object = Primary.Melanocytes.Healthy, cells = T1like.Cells) <- "T1-like"
Idents(object = Primary.Melanocytes.Healthy, cells = T2like.Cells) <- "T2-like"
Primary.Melanocytes.Healthy <- AddMetaData(Primary.Melanocytes.Healthy, col.name = "T_prediction", metadata = Idents(Primary.Melanocytes.Healthy))
levels <- c("T1-like", "T2-like")
Primary.Melanocytes.Healthy@meta.data$T_prediction <- factor(x = Idents(Primary.Melanocytes.Healthy), levels = levels)

ggplot(Primary.Melanocytes.Healthy@meta.data, aes(x = Stage, fill = T_prediction)) + geom_bar(position = "fill") +theme_light() + theme(axis.text.x = element_text(angle = 90, vjust = .5, size = 15, colour = "black", hjust = 1), axis.title.x = element_blank(),legend.text = element_text(size = 15), legend.title = element_text(size = 15), axis.text.y.left = element_text(size = 15, colour = "black"), axis.title.y = element_text(size = 15))+ scale_fill_manual(values=c(dir_D30_celltype_cols, indir_D59_celltype_cols))
ggsave(paste0(outputDir, "/BarPlot","Primary.Melanocytes.Healthy.T_prediction", "Stage" ,".pdf"), width = 4, height = 4, dpi = 320, limitsize = FALSE)

ggplot(Primary.Melanocytes.Healthy@meta.data, aes(x = APaxis, fill = T_prediction)) + geom_bar(position = "fill") +theme_light() + theme(axis.text.x = element_text(angle = 90, vjust = .5, size = 15, colour = "black", hjust = 1), axis.title.x = element_blank(),legend.text = element_text(size = 15), legend.title = element_text(size = 15), axis.text.y.left = element_text(size = 15, colour = "black"), axis.title.y = element_text(size = 15))+ scale_fill_manual(values=c(dir_D30_celltype_cols, indir_D59_celltype_cols))
ggsave(paste0(outputDir, "/BarPlot","Primary.Melanocytes.Healthy.T_prediction", "APaxis" ,".pdf"), width = 5, height = 4, dpi = 320, limitsize = FALSE)
```

###Figure 5
```{r TCGA data curation and GSEA}
setwd("/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/TCGA_Data_Sets/Melanoma/Sample FPKM files/")
files <- list.files(path= "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/TCGA_Data_Sets/Melanoma/Sample FPKM files/")
gene.name.ref <- read.delim(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/TCGA_Data_Sets/Melanoma/Sample FPKM Files/0a3f430a-7e5e-4e68-92cc-f767da7e361c.FPKM.txt", header = FALSE)
genes <- as.character(gene.name.ref[,1])
genes <- str_replace(genes,pattern = ".[0-9]+$",replacement = "")
df    <- do.call(cbind,lapply(files,function(fn)read.table(fn,header=FALSE, sep="\t")[,2]))
row.names(df) <- genes
colnames(df) <- gsub('\\.', '-', files)
df <- df[which(rowMeans(df, dims = 1)!=0),]

mart <- useMart(biomart = "ENSEMBL_MART_ENSEMBL", dataset = "hsapiens_gene_ensembl")
Gene_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id", "hgnc_symbol", "hgnc_id"),values=row.names(df),mart= mart)
Gene_list <- Gene_list[which(!duplicated(Gene_list$ensembl_gene_id)),]
#Gene_list.sub <- Gene_list[Gene_list$hgnc_symbol != "",]

rownames(Gene_list) <- Gene_list$ensembl_gene_id
df <- merge(df, Gene_list, by = 0)
df <- df[df$hgnc_symbol != "",]
df <- df[which(!duplicated(df$hgnc_symbol)),]
rownames(df) <- df$hgnc_symbol
df <- df[,-c(1,474:476)]

TCGA.metadata <- read.csv(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/TCGA_Data_Sets/Melanoma/skcm_tcga_clinical_metadata_allExprSamples.csv")
rownames(TCGA.metadata) <- as.character(TCGA.metadata$Sample.ID)
colnames(df) <- as.character(TCGA.metadata$Sample.ID)
df <- df[,colnames(df) != ""]
TCGA.metadata <- TCGA.metadata[rownames(TCGA.metadata) != "",]

##breaking here
df.scaled <- t(scale(t(df)))
samples <- colnames(df.scaled)

for(i in 1:length(samples)) {
  x <- as.numeric(df.scaled[,i])
  names(x) <- rownames(df.scaled)
  x <- sort(x, decreasing = T)
  new.name <- paste(samples[i])
  assign(new.name, x)
}

mat.mel.DE <- read.csv(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/data/10x_full/mat.mel.DE.csv")
top100.mel.DE <- list(mat.mel.DE$gene[which(mat.mel.DE$cluster == "T1 Melanocyte" & mat.mel.DE$gene %in% rownames(df.scaled))][1:100], mat.mel.DE$gene[which(mat.mel.DE$cluster == "T2 Melanocyte" & mat.mel.DE$gene %in% rownames(df.scaled))][1:100])
names(top100.mel.DE) <- c("T1_enriched", "T2_enriched")

Sample.Pvalues <- data.frame(NA, matrix(nrow = length(top100.mel.DE), ncol = length(samples)-1))
Sample.EnrichmentScores <- data.frame(NA, matrix(nrow = length(top100.mel.DE), ncol = length(samples)-1))
colnames(Sample.Pvalues) <- samples
colnames(Sample.EnrichmentScores) <- samples
rownames(Sample.Pvalues) <- names(top100.mel.DE)
rownames(Sample.EnrichmentScores) <- names(top100.mel.DE)

for(i in 1:length(samples)) {
  sample.object <- get(samples[i])
  fgseaRes <- fgsea(pathways = top100.mel.DE, 
                  stats = sample.object,
                  minSize=1,
                  maxSize=500, 
                  nperm = 1000)

  Sample.Pvalues[,i] <- fgseaRes$padj
  Sample.EnrichmentScores[,i] <- fgseaRes$NES
}

TCGA.metadata$T1.enrichment.NES <- as.numeric(Sample.EnrichmentScores[1,])
TCGA.metadata$T1.enrichment.pval <- as.numeric(Sample.Pvalues[1,])
TCGA.metadata$T2.enrichment.NES <- as.numeric(Sample.EnrichmentScores[2,])
TCGA.metadata$T2.enrichment.pval <- as.numeric(Sample.Pvalues[2,])
TCGA.metadata$T1.Enrichment.Status <- NA
TCGA.metadata$T1.Enrichment.Status[which(TCGA.metadata$T1.enrichment.NES > 0 & TCGA.metadata$T1.enrichment.pval < 0.05)] <- "Enriched"
TCGA.metadata$T1.Enrichment.Status[which(TCGA.metadata$T1.enrichment.NES < 0 & TCGA.metadata$T1.enrichment.pval < 0.05)] <- "Reduced"
TCGA.metadata$T1.Enrichment.Status[which(TCGA.metadata$T1.enrichment.pval > 0.05)] <- "Non Sig"
TCGA.metadata$T2.Enrichment.Status <- NA
TCGA.metadata$T2.Enrichment.Status[which(TCGA.metadata$T2.enrichment.NES > 0 & TCGA.metadata$T2.enrichment.pval < 0.05)] <- "Enriched"
TCGA.metadata$T2.Enrichment.Status[which(TCGA.metadata$T2.enrichment.NES < 0 & TCGA.metadata$T2.enrichment.pval < 0.05)] <- "Reduced"
TCGA.metadata$T2.Enrichment.Status[which(TCGA.metadata$T2.enrichment.pval > 0.05)] <- "Non Sig"
TCGA.metadata$T.Enrichment.Label <- NA
TCGA.metadata$T.Enrichment.Label[which(TCGA.metadata$T1.enrichment.NES > 0 & TCGA.metadata$T1.enrichment.pval < 0.05 & !(TCGA.metadata$T2.enrichment.NES > 0 & TCGA.metadata$T2.enrichment.pval < 0.05))] <- "T1 Enriched"
TCGA.metadata$T.Enrichment.Label[which(TCGA.metadata$T2.enrichment.NES > 0 & TCGA.metadata$T2.enrichment.pval < 0.05 & !(TCGA.metadata$T1.enrichment.NES > 0 & TCGA.metadata$T1.enrichment.pval < 0.05))] <- "T2 Enriched"
```

```{r Figure S5A}
outputDir <- "/Users/fattahilab/Desktop/Manuscript Figures/"
ggplot(data=TCGA.metadata[!is.na(TCGA.metadata$T1.enrichment.NES),], aes(x=T1.enrichment.NES, y=-log10(T1.enrichment.pval))) + geom_point(aes(col=T1.Enrichment.Status), shape=16, size = 3) + theme_test() +scale_color_manual(values = c("#7DB0DDFF", "gainsboro", "#666464")) +theme(legend.position = "none", panel.border = element_rect(size = 2, colour = "black"), axis.ticks = element_line(size = 2, colour = "black"))+geom_hline(yintercept = -log10(.05), size = 2, linetype = "dashed")
ggsave(paste0(outputDir, "/T1.Mel.GSEA.Enrichment.Volcano.jpeg"), width = 4, height = 4, dpi = 320)

ggplot(data=TCGA.metadata[!is.na(TCGA.metadata$T2.enrichment.NES),], aes(x=T2.enrichment.NES, y=-log10(T2.enrichment.pval))) + geom_point(aes(col=T2.Enrichment.Status), shape=16, size = 3) + theme_test() +scale_color_manual(values = c("#E093C3FF", "gainsboro", "#666464")) +theme(legend.position = "none", panel.border = element_rect(size = 2, colour = "black"), axis.ticks = element_line(size = 2, colour = "black"))+geom_hline(yintercept = -log10(.05), size = 2, linetype = "dashed")
ggsave(paste0(outputDir, "/T2.Mel.GSEA.Enrichment.Volcano.jpeg"), width = 4, height = 4, dpi = 320)
```

```{r Survival Analysis of T1 and T2 TCGA Figures 5B and S5B}
#remove samples with missing values
DFS <- subset(TCGA.metadata, subset = !is.na(Overall.Survival.Status) & !is.na(Overall.Survival..Months.) & Overall.Survival..Months.>0)

#add column of 1/0 status for living or deceased
DFS$Survival.Status <- NA
DFS$Survival.Status[which(DFS$Overall.Survival.Status=="LIVING")] <- 1
DFS$Survival.Status[which(DFS$Overall.Survival.Status=="DECEASED")] <- 0

#not censoring, convert months to years
DFS$Survial.Time.Years <- DFS$Overall.Survival..Months./12

#create survival object
surv<-Surv(DFS$Survial.Time.Years, DFS$Survival.Status)

T1.DFS.sub <- subset(DFS, subset = T1.Enrichment.Status %in% c("Enriched", "Reduced"))

km_fit <- survfit(Surv(Survial.Time.Years, Survival.Status) ~ T1.Enrichment.Status, data=T1.DFS.sub)
ggsurvplot(km_fit, pval = TRUE, palette = c("#7DB0DDFF", "#666464"))
ggsave(paste0(outputDir, "/T1.Mel.GSEA.Enrichment.survcurve.pdf"), width = 4, height = 3, dpi = 320, limitsize = FALSE)

T2.DFS.sub <- subset(DFS, subset = T2.Enrichment.Status %in% c("Enriched", "Reduced"))

km_fit <- survfit(Surv(Survial.Time.Years, Survival.Status) ~ T2.Enrichment.Status, data=T2.DFS.sub)
ggsurvplot(km_fit, pval = TRUE, palette = c("#E093C3FF", "#666464"))
ggsave(paste0(outputDir, "/T2.Mel.GSEA.Enrichment.survcurve.pdf"), width = 4, height = 3, dpi = 320, limitsize = FALSE)

DFS.sub <- subset(DFS, subset = T.Enrichment.Label %in% c("T1 Enriched", "T2 Enriched"))

km_fit <- survfit(Surv(Survial.Time.Years, Survival.Status) ~ T.Enrichment.Label, data=DFS.sub)
ggsurvplot(km_fit, pval = TRUE, palette = c("#7DB0DDFF", "#E093C3FF"))
ggsave(paste0(outputDir, "/T1vT2.Mel.GSEA.Enrichment.survcurve.pdf"), width = 4, height = 3, dpi = 320, limitsize = FALSE)
```

```{r Significant gRNA Depletion analysis Figures S5C and D}
D0.Met <- read.csv(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/CRISPRi/20200913/D0.to.Met.by.guide.csv")
guide.labels <- read.csv(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/CRISPRi/20200913/guide.labels.csv")
# add a column of NAs
D0.Met$Significant <- "Not Significant"
D0.Met$Significant[D0.Met$logFC < 0 & D0.Met$p.value < 0.05] <- "Significantly Depleted"

D0.Met <- merge(D0.Met, guide.labels, by = "Guide", all = TRUE)

ggplot(data=D0.Met, aes(x=logFC, y=-log10(p.value)))+geom_hline(yintercept = -log10(.05), size = 1, linetype = "dashed") + geom_point(aes(col=Mel.Enrich, shape = Mel.Enrich, size = Mel.Enrich)) + theme_test() +scale_color_manual(values = c("#7DB0DDFF", "#E093C3FF"), na.value = "black")+scale_size_manual(values=c(5,5), na.value = 2)+ scale_shape_manual(values=c(16, 16), na.value = 1) +theme(legend.position = "none", panel.border = element_rect(size = 2, colour = "black"))
ggsave(paste0(outputDir, "/gRNA.volcano.jpeg"), width = 4, height = 4, dpi = 320, limitsize = FALSE)

D0.Met.hits <- D0.Met[!(is.na(D0.Met$Gene)),c(2,7,8)]
D0.Met.hits$Gene <- make.unique(D0.Met.hits$Gene)

D0.Met.hits <- D0.Met.hits[order(D0.Met.hits$logFC, decreasing = F),]
D0.Met.hits$logFC <- as.numeric(D0.Met.hits$logFC)

ggplot(data=D0.Met.hits, aes(x=logFC, y=reorder(Gene, -logFC), fill=Mel.Enrich)) + geom_bar(stat = "identity") + theme_cowplot() +scale_fill_manual(values = c("#7DB0DDFF", "#E093C3FF")) +theme(legend.position = "none")+xlim(-5,0)+scale_y_discrete(position = "left")+scale_x_reverse(position = "top")+ylab("sgRNA Target Gene")
ggsave(paste0(outputDir, "/D0toMet.hits.barplot.logFC.pdf"), width = 4, height = 8, dpi = 320, limitsize = FALSE)
```

```{r COX Proportional hazard multivariate analysis of CRISPRi hits Figure 5D}
guide.labels <- read.csv(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/CRISPRi/20200913/guide.labels.csv")
#make dataframe of scaled expression for for hit genes of each sample
T2.CRISPRi.hits <- unique(guide.labels$Gene[which(guide.labels$Mel.Enrich == "T2")])

exp <- data.frame(NA, matrix(nrow = length((colnames(df.scaled))), ncol = length(T2.CRISPRi.hits)))
colnames(exp) <- c("Sample.ID", T2.CRISPRi.hits)
exp$Sample.ID <- TCGA.metadata$Sample.ID
for(i in 1:length(T2.CRISPRi.hits)){
  exp[,1+i] <- df.scaled[which(rownames(df.scaled) == T2.CRISPRi.hits[i]),]
}

#join metadata and scaled expression of hit genes
DFS <- DFS %>% inner_join(exp, by="Sample.ID")

#create survival object
surv<-Surv(DFS$Survial.Time.Years, DFS$Survival.Status)

HR.df <- data.frame(NA, matrix(nrow = length(T2.CRISPRi.hits), ncol = 1))
rownames(HR.df) <- T2.CRISPRi.hits
colnames(HR.df) <- c("HR", "pvalue")

for(i in 1:length(rownames(HR.df))){
r <- sort(DFS[,i+74])[50:(length(DFS[,i+74])-50)]

pvalues <- data.frame(threshold= numeric(0), pv=numeric(0), HR=numeric(0), stringsAsFactors = FALSE)
for (x in r){
  updn <- as.numeric(DFS[,i+74]>x)
  diff <- survdiff(surv~updn)
  pv <- (pchisq(diff$chisq, length(diff$n)-1, lower.tail = FALSE))
  HR <- (diff$obs[2]/diff$exp[2])/(diff$obs[1]/diff$exp[1])
  pvalues[nrow(pvalues)+1,] <- c(x, pv, HR)
}

th <-  pvalues$threshold[which.max(-log10(pvalues$pv))]
DFS <- DFS %>% mutate(exp_t = ifelse(DFS[,i+74]>th, "UP", "DOWN"))

cox2 <- coxph(Surv(Survial.Time.Years, Survival.Status) ~ exp_t + Diagnosis.Age + American.Joint.Committee.on.Cancer.Metastasis.Stage.Code + Neoplasm.Disease.Lymph.Node.Stage.American.Joint.Committee.on.Cancer.Code + Neoplasm.Disease.Stage.American.Joint.Committee.on.Cancer.Code + American.Joint.Committee.on.Cancer.Tumor.Stage.Code + Neoadjuvant.Therapy.Type.Administered.Prior.To.Resection.Text + Adjuvant.Postoperative.Pharmaceutical.Therapy.Administered.Indicator + Did.patient.start.adjuvant.postoperative.radiotherapy., data = DFS)
results <- summary(cox2)
HR.df[i,] <- results$coefficients[1,c(2,5)]
}

HR.df$Multi.Significant <- "Not Significant"
HR.df$Multi.Significant[HR.df$HR > 1 & HR.df$pvalue < 0.05] <- "Significant"
HR.df$Sig.Gene <- NA
HR.df$Sig.Gene[which(HR.df$Multi.Significant == "Significant")] <- rownames(HR.df)[which(HR.df$Multi.Significant == "Significant")]

ggplot(data=HR.df, aes(x=HR, y=-log10(pvalue), label = Sig.Gene))+geom_hline(yintercept = -log10(.05), size = 1, linetype = "dashed") + geom_point(aes(col=Multi.Significant, size = Multi.Significant)) + theme_test() +scale_color_manual(values = c("black", "firebrick"))+scale_size_manual(values=c(4,5))+geom_label_repel(box.padding = 1, label.size = .5, size = 8, nudge_y = .5, nudge_x = -.1) +theme(legend.position = "none", panel.border = element_rect(size = 2, colour = "black"), axis.ticks = element_line(size = 1.5))
ggsave(paste0(outputDir, "/T2.Mel.GSEA.Enrichment.Volcano.pdf"), width = 6, height = 3.5, dpi = 320, limitsize = FALSE)
```

```{r Survival Curves Mann Whitney U Tests of selected 3 Hits, Figure 5E}
r <- sort(DFS$HMG20B)[50:(length(DFS$HMG20B)-50)]
pvalues <- data.frame(threshold= numeric(0), pv=numeric(0), HR=numeric(0), stringsAsFactors = FALSE)
for (i in r){
#for (i in r[r>-0.1 & r<0.1]){
  updn <- as.numeric(DFS$HMG20B>i)
  diff <- survdiff(surv~updn)
  pv <- (pchisq(diff$chisq, length(diff$n)-1, lower.tail = FALSE))
  HR <- (diff$obs[2]/diff$exp[2])/(diff$obs[1]/diff$exp[1])
  pvalues[nrow(pvalues)+1,] <- c(i, pv, HR)
}
plot(pvalues$threshold, -log10(pvalues$pv))
th <-  pvalues$threshold[which.max(-log10(pvalues$pv))]
DFS <- DFS %>% mutate(exp_t = ifelse(HMG20B>th, "UP", "DOWN"))
km_fit <- survfit(Surv(Survial.Time.Years, Survival.Status) ~ exp_t, data=DFS)
#autoplot(km_fit, conf.int = FALSE,censor = FALSE)
ggsurvplot(km_fit, pval = TRUE, palette = c("#666464", "#F48F23"))
ggsave(paste0(outputDir, "/HMG20B.survcurve.pdf"), width = 4, height = 3, dpi = 320, limitsize = FALSE)

r <- sort(DFS$MMP17)[50:(length(DFS$MMP17)-50)]
pvalues <- data.frame(threshold= numeric(0), pv=numeric(0), HR=numeric(0), stringsAsFactors = FALSE)
for (i in r){
#for (i in r[r>-0.1 & r<0.1]){
  updn <- as.numeric(DFS$MMP17>i)
  diff <- survdiff(surv~updn)
  pv <- (pchisq(diff$chisq, length(diff$n)-1, lower.tail = FALSE))
  HR <- (diff$obs[2]/diff$exp[2])/(diff$obs[1]/diff$exp[1])
  pvalues[nrow(pvalues)+1,] <- c(i, pv, HR)
}
plot(pvalues$threshold, -log10(pvalues$pv))
th <-  pvalues$threshold[which.max(-log10(pvalues$pv))]
DFS <- DFS %>% mutate(exp_t = ifelse(MMP17>th, "UP", "DOWN"))
km_fit <- survfit(Surv(Survial.Time.Years, Survival.Status) ~ exp_t, data=DFS)
#autoplot(km_fit, conf.int = FALSE,censor = FALSE)
ggsurvplot(km_fit, pval = TRUE, palette = c("#666464", "#64509A"))
ggsave(paste0(outputDir, "/MMP17.survcurve.pdf"), width = 4, height = 3, dpi = 320, limitsize = FALSE)

r <- sort(DFS$SNRPB)[50:(length(DFS$SNRPB)-50)]
pvalues <- data.frame(threshold= numeric(0), pv=numeric(0), HR=numeric(0), stringsAsFactors = FALSE)
for (i in r){
#for (i in r[r>-0.1 & r<0.1]){
  updn <- as.numeric(DFS$SNRPB>i)
  diff <- survdiff(surv~updn)
  pv <- (pchisq(diff$chisq, length(diff$n)-1, lower.tail = FALSE))
  HR <- (diff$obs[2]/diff$exp[2])/(diff$obs[1]/diff$exp[1])
  pvalues[nrow(pvalues)+1,] <- c(i, pv, HR)
}
plot(pvalues$threshold, -log10(pvalues$pv))
th <-  pvalues$threshold[which.max(-log10(pvalues$pv))]
DFS <- DFS %>% mutate(exp_t = ifelse(SNRPB>th, "UP", "DOWN"))
km_fit <- survfit(Surv(Survial.Time.Years, Survival.Status) ~ exp_t, data=DFS)
#autoplot(km_fit, conf.int = FALSE,censor = FALSE)
ggsurvplot(km_fit, pval = TRUE, palette = c("#666464", "#EB4142"))
ggsave(paste0(outputDir, "/SNRPB.survcurve.pdf"), width = 4, height = 3, dpi = 320, limitsize = FALSE)
```

```{r Analysis of SNRPB Spliceofrom MISO outputs}
SNRPB.Bed <- read.table(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/CRISPRi/SNRPB KD Data/SE.hg38.full.SNRPB.bed.txt", col.names = c("Chromosome", "Start_BP", "Stop_BP", "Event", "Num", "Strand"))
SNRPB.clip <- read.table(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/CRISPRi/SNRPB KD Data/SE.hg38.full.SNRPB.list.txt", col.names = c("Event"))
map <- read.table(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/CRISPRi/SNRPB KD Data/SE.hg38.map.txt", col.names = c("Event", "Chromosome_Locations"))
gene.event.map <- read.table(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/CRISPRi/SNRPB KD Data/SE.hg38.gene.event.map.txt", col.names = c("Gene", "Event"))
gene.event.map <- gene.event.map[-which(duplicated(gene.event.map$Event)),]
gene.event.map$genes.unique <- make.unique(gene.event.map$Gene)

MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff <- read.table(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/CRISPRi/SNRPB KD Data/miso_comp/MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso/bayes-factors/MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff.txt", header = T)
MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$Gene <- NA
MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$genes.unique <- NA

A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff <- read.table(file = "/Volumes/Ryan_EHD/UCSF/Faranak_Lab/Data/Melanocyte Timecourse/CRISPRi/SNRPB KD Data/miso_comp/A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso/bayes-factors/A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff.txt", header = T)
A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$Gene <- NA
A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$genes.unique <- NA

for(i in 1:length(rownames(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff))){
  gene <- gene.event.map$Gene[which(gene.event.map$Event == MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff[i,1])]
  MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff[i,4] <- paste(unique(gene), collapse = "/")
  gene.unique <- gene.event.map$genes.unique[which(gene.event.map$Event == MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff[i,1])]
  MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff[i,5] <- paste(unique(gene.unique), collapse = "/")
}
MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$genes.unique[which(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$genes.unique == "")] <- MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$eid[which(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$genes.unique == "")]

for(i in 1:length(rownames(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff))){
  gene <- gene.event.map$Gene[which(gene.event.map$Event == A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff[i,1])]
  A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff[i,4] <- paste(unique(gene), collapse = "/")
  gene.unique <- gene.event.map$genes.unique[which(gene.event.map$Event == A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff[i,1])]
  A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff[i,5] <- paste(unique(gene.unique), collapse = "/")
}
A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$genes.unique[which(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$genes.unique == "")] <- A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$eid[which(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$genes.unique == "")]

MeWo_ctrl.sig <- subset(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff, subset = diff < -.1 & bayes_factor > 10)
MeWo_SNRPB.KD.sig <- subset(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff, subset = diff > .1 & bayes_factor > 10)

A375_ctrl.sig <- subset(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff, subset = diff < -.1 & bayes_factor > 10)
A375_SNRPB.KD.sig <- subset(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff, subset = diff > .1 & bayes_factor > 10)

MeWo_ctrl.all <- MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff
MeWo_ctrl.all$diff <- -MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$diff
MeWo_SNRPB.KD.all <- MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff

A375_ctrl.all <- A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff
A375_ctrl.all$diff <- -A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$diff
A375_SNRPB.KD.all <- A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff

A375.sig <- subset(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff, subset = diff < -.1 & bayes_factor > 10 | diff > .1 & bayes_factor > 10)
MeWo.sig <- subset(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff, subset = diff < -.1 & bayes_factor > 10 | diff > .1 & bayes_factor > 10)

melanoma.shared <- A375.sig$genes.unique[which(A375.sig$genes.unique %in% MeWo.sig$genes.unique)]
```

```{r Spliceoform Volcano Plot Figure 5K}
MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$Significant <- "Not Significant"
MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$Significant[which(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$diff > .1  & MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$bayes_factor > 10 | MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$diff < -.1 & MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$bayes_factor > 10)] <- "Significant"
MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$Significant[which(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$eid %in% SNRPB.clip$Event & MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$Significant == "Significant")] <- "Significant Target"
MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$Shared.event <- NA
MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$Shared.event[which(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$genes.unique %in% melanoma.shared)] <- MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$genes.unique[which(MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff$genes.unique %in% melanoma.shared)]

ggplot(data=MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff, aes(x=diff, y=log(bayes_factor), color = Significant, label = Shared.event))+geom_point(shape=16, size = 3) + theme_test() +geom_hline(yintercept = log(10), size = 1, linetype = "dashed")+geom_vline(xintercept = c(-.1,.1), size = 1, linetype = "dashed")+scale_colour_manual(values=c("grey", "steelblue", "#EB4142"))+NoLegend()
ggsave(paste0(outputDir, "/MeWo.KD.v.CNTL.Diff.Splicing.with.SNRPBtarget.jpeg"), width = 5, height = 4, dpi = 320)

ggplot(data=MeWo_gSNRPB_se_miso_vs_MeWo_gCTRL_se_miso_diff, aes(x=diff, y=log(bayes_factor), color = Significant, label = Shared.event))+geom_label_repel(box.padding = .5, label.size = .5, size = 4, na.rm = T, nudge_y = 15) + theme_test() +geom_hline(yintercept = log(10), size = 1, linetype = "dashed")+geom_vline(xintercept = c(-.1,.1), size = 1, linetype = "dashed")+scale_colour_manual(values=c("grey", "steelblue", "#EB4142"))+NoLegend()
ggsave(paste0(outputDir, "/MeWo.KD.v.CNTL.Diff.Splicing.with.SNRPBtarget.just.labels.pdf"), width = 5, height = 4, dpi = 320)

A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$Significant <- "Not Significant"
A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$Significant[which(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$diff > .1  & A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$bayes_factor > 10 | A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$diff < -.1 & A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$bayes_factor > 10)] <- "Significant"
A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$Significant[which(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$eid %in% SNRPB.clip$Event & A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$Significant == "Significant")] <- "Significant Target"
A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$Shared.event <- NA
A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$Shared.event[which(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$genes.unique %in% melanoma.shared)] <- A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$genes.unique[which(A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff$genes.unique %in% melanoma.shared)]

ggplot(data=A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff, aes(x=diff, y=log(bayes_factor), color = Significant, label = Shared.event))+geom_point(shape=16, size = 3) + theme_test() +geom_hline(yintercept = log(10), size = 1, linetype = "dashed")+geom_vline(xintercept = c(-.1,.1), size = 1, linetype = "dashed")+scale_colour_manual(values=c("grey", "steelblue", "#EB4142"))+NoLegend()
ggsave(paste0(outputDir, "/A375.KD.v.CNTL.Diff.Splicing.with.SNRPBtarget.jpeg"), width = 5, height = 4, dpi = 320)

ggplot(data=A375_gSNRPB_se_miso_vs_A375_gCTRL_se_miso_diff, aes(x=diff, y=log(bayes_factor), color = Significant, label = Shared.event))+geom_label_repel(box.padding = .5, label.size = .5, size = 4, na.rm = T, nudge_y = -3) + theme_test() +geom_hline(yintercept = log(10), size = 1, linetype = "dashed")+geom_vline(xintercept = c(-.1,.1), size = 1, linetype = "dashed")+scale_colour_manual(values=c("grey", "steelblue", "#EB4142"))+NoLegend()
ggsave(paste0(outputDir, "/A375.KD.v.CNTL.Diff.Splicing.with.SNRPBtarget.just.labels.pdf"), width = 5, height = 4, dpi = 320)
```

```{r Spliced Genes GSEA Figure 5L}
Human_gene_sets = msigdbr(species = "Homo sapiens", category = "C5")
Human_gene_sets = split(x = Human_gene_sets$gene_symbol, f = Human_gene_sets$gs_name)
Human_gene_sets <- Human_gene_sets[grepl("^GOBP", names(Human_gene_sets))]

A375_ctrl.sig.for.gsea <- as.numeric(-A375_ctrl.sig$diff[which(A375_ctrl.sig$Gene != "")])
names(A375_ctrl.sig.for.gsea) <- A375_ctrl.sig$Gene[which(A375_ctrl.sig$Gene != "")]
A375_ctrl.sig.for.gsea <- sort(A375_ctrl.sig.for.gsea, decreasing = T)

A375_SNRPB.KD.sig.for.gsea <- as.numeric(A375_SNRPB.KD.sig$diff[which(A375_SNRPB.KD.sig$Gene != "")])
names(A375_SNRPB.KD.sig.for.gsea) <- A375_SNRPB.KD.sig$Gene[which(A375_SNRPB.KD.sig$Gene != "")]
A375_SNRPB.KD.sig.for.gsea <- sort(A375_SNRPB.KD.sig.for.gsea, decreasing = T)

MeWo_ctrl.sig.for.gsea <- as.numeric(-MeWo_ctrl.sig$diff[which(MeWo_ctrl.sig$Gene != "")])
names(MeWo_ctrl.sig.for.gsea) <- MeWo_ctrl.sig$Gene[which(MeWo_ctrl.sig$Gene != "")]
MeWo_ctrl.sig.for.gsea <- sort(MeWo_ctrl.sig.for.gsea, decreasing = T)

MeWo_SNRPB.KD.sig.for.gsea <- as.numeric(MeWo_SNRPB.KD.sig$diff[which(MeWo_SNRPB.KD.sig$Gene != "")])
names(MeWo_SNRPB.KD.sig.for.gsea) <- MeWo_SNRPB.KD.sig$Gene[which(MeWo_SNRPB.KD.sig$Gene != "")]
MeWo_SNRPB.KD.sig.for.gsea <- sort(MeWo_SNRPB.KD.sig.for.gsea, decreasing = T)

A375.sig.for.gsea <- c(A375_ctrl.sig.for.gsea, A375_SNRPB.KD.sig.for.gsea)
A375.sig.for.gsea <- sort(A375.sig.for.gsea, decreasing = T)

A375.sig.gsea.df <- fgseaMultilevel(pathways = Human_gene_sets, 
                         stats = A375.sig.for.gsea,
                         nPermSimple=1000, scoreType = "pos", minSize = 5)

MeWo.sig.for.gsea <- c(MeWo_ctrl.sig.for.gsea, MeWo_SNRPB.KD.sig.for.gsea)
MeWo.sig.for.gsea <- sort(MeWo.sig.for.gsea, decreasing = T)

MeWo.sig.gsea.df <- fgseaMultilevel(pathways = Human_gene_sets, 
                         stats = MeWo.sig.for.gsea,
                         nPermSimple=1000, scoreType = "pos", minSize = 5)

pathways.of.interest <- A375.sig.gsea.df$pathway[A375.sig.gsea.df$pathway %in% MeWo.sig.gsea.df$pathway]
rna.paths <- pathways.of.interest[grepl("RNA_METABOLIC", pathways.of.interest)]
migration.paths <- c(pathways.of.interest[grepl("LOCOMOTION", pathways.of.interest)], pathways.of.interest[grepl("MIGRATION", pathways.of.interest)])
adhesion.paths <- pathways.of.interest[grepl("ADHESION", pathways.of.interest)][!grepl("LEUKOCYTE", pathways.of.interest[grepl("ADHESION", pathways.of.interest)])]
proliferation.paths <- c(pathways.of.interest[grepl("CELL_CYCLE", pathways.of.interest)], pathways.of.interest[grepl("PROLIFERATION", pathways.of.interest)], pathways.of.interest[grepl("GROWTH", pathways.of.interest)])[c(1,3:7)]

paths.to.plot <- c(rna.paths, migration.paths, adhesion.paths, proliferation.paths)

spliced.paths.df <- as.data.frame(matrix(nrow = length(paths.to.plot), ncol = 2), row.names = paths.to.plot)
colnames(spliced.paths.df) <- c("MeWo", "A375")

for(x in 1:length(paths.to.plot)){
    spliced.paths.df[x,1] <- MeWo.sig.gsea.df$size[which(MeWo.sig.gsea.df$pathway == rownames(spliced.paths.df)[x])]
    spliced.paths.df[x,2] <- A375.sig.gsea.df$size[which(A375.sig.gsea.df$pathway == rownames(spliced.paths.df)[x])]
}

spliced.paths.df$Paths <- gsub("GOBP_", "", rownames(spliced.paths.df))
spliced.paths.df$Paths <- gsub("_", " ", spliced.paths.df$Paths)
spliced.paths.df.melt <- melt(spliced.paths.df)
colnames(spliced.paths.df.melt) <- c("Paths", "Cell_Line", "value")

ggplot(spliced.paths.df.melt, aes(fill=Cell_Line, y=factor(spliced.paths.df.melt$Paths, levels = rev(gsub("_", " ", gsub("GOBP_", "", paths.to.plot)))), x=value)) + 
    geom_bar(position="dodge", stat="identity", col = "black") +theme_cowplot()+theme(axis.text.x = element_text(size = 10,angle = 90, hjust = 1, vjust = .5), axis.text.y = element_text(size = 10))+scale_fill_manual(values=c("plum4", "dodgerblue4"))+xlab("Pathway")+ylab("#Genes in Pathway")
ggsave(paste0(outputDir, "/Shared.spliced.paths.barplot.pdf"), width = 8, height = 5, dpi = 320)
```